<!DOCTYPE HTML>
<html lang="en-US">
<head>
  <meta charset="UTF-8">
  <title>News</title>
  <meta name="viewport" content="width=device-width,initial-scale=1">
  <meta name="generator" content="Jekyll v4.2.1">
  <link rel="alternate" type="application/rss+xml" title="Global Names Architecture" href="/feed.xml">
  <link rel="alternate" type="application/atom+xml" title="Recent commits to the site" href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io/commits/master.atom">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic,900">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/screen.css">
  <link rel="icon" type="image/x-icon" href="/favicon.ico">
  <!--[if lt IE 9]>
  <script src="/js/html5shiv.min.js"></script>
  <script src="/js/respond.min.js"></script>
  <![endif]-->
</head>


<body class="wrap">
  <header role="banner">
  <nav class="mobile-nav show-on-mobiles">
    <ul>
  <li class="">
    <a href="/">Home</a>
  </li>
  <li class="current">
    <a href="/news/">News</a>
  </li>
  <li class="">
    <a href="/apps/">Apps</a>
  </li>
  <li class="">
    <a href="/workshops/">Workshops</a>
  </li>
  <li class="">
    <a href="/bibliography/">Papers</a>
  </li>
  <li class="">
    <a href="/docs/home/">Docs</a>
  </li>
  <li>
    <a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io"><span class="hide-on-mobiles"></span>Source</a>
  </li>
</ul>

  </nav>
  <div class="grid">
    <div class="unit one-quarter center-on-mobiles">
      <div class="logo">
        <a href="/">
          <span class="sr-only">GlobalNames</span>
          <img src="/img/gna-logo.svg" width="100" alt="GNA Logo">
        </a>
      </div>
    </div>
    <nav class="main-nav unit three-quarters hide-on-mobiles">
      <ul>
  <li class="">
    <a href="/">Home</a>
  </li>
  <li class="current">
    <a href="/news/">News</a>
  </li>
  <li class="">
    <a href="/apps/">Apps</a>
  </li>
  <li class="">
    <a href="/workshops/">Workshops</a>
  </li>
  <li class="">
    <a href="/bibliography/">Papers</a>
  </li>
  <li class="">
    <a href="/docs/home/">Docs</a>
  </li>
  <li>
    <a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io"><span class="hide-on-mobiles"></span>Source</a>
  </li>
</ul>

    </nav>
  </div>
</header>


    <section class="news">
    <div class="grid">

      <div class="docs-nav-mobile unit whole show-on-mobiles">
  <select onchange="if (this.value) window.location.href=this.value">
    <option value="">Navigate the blog…</option>
    <option value="/news/">Home</option>
    <optgroup label="v1.x">
      
      <option value="/news/2021/12/12/gnverifier-0.6.2/">GNverifier release v0.6.2, Advanced Search</option>
      
      <option value="/news/2021/10/28/gnparser-1.5.0/">GNparser (Go language) release 1.5.0</option>
      
      <option value="/news/2021/08/13/gnparser-1.3.3/">GNparser (Go language) release 1.3.3</option>
      
      <option value="/news/2021/07/14/go/">Why Go is a great language for biodiversity informatics</option>
      
      <option value="/news/2021/07/05/gnparser-1.3.0/">GNparser (Go language) release 1.3.0</option>
      
      <option value="/news/2021/06/19/gnfinder-0.14.1/">GNfinder release v0.14.1</option>
      
      <option value="/news/2021/04/10/gnverifier-0.3.0/">GNverifier release v0.3.0</option>
      
      <option value="/news/2021/04/10/gnarser-1.2.0/">GNparser (Go language) release 1.2.0</option>
      
      <option value="/news/2021/03/23/gnparser/">GNparser (Go language) release 1.1.0</option>
      
      <option value="/news/2019/12/11/gnfinder/">gnfinder release 0.9.1 -- bug fixes</option>
      
      <option value="/news/2019/12/10/gnrd/">GNRD release (v0.9.0) switched to gnfinder from TaxonFinder/NetiNeti and became 25 times faster.</option>
      
      <option value="/news/2019/11/21/nsf1645959-outcomes/">Outcomes draft (Final Report for NSF award 1645959)</option>
      
      <option value="/news/2019/11/20/col-gbif-bhl/">Biodiversity Heritage Library (BHL) link to Catalogue of Life Plus (CoL+)</option>
      
      <option value="/news/2019/11/19/gnparser-clib/">gnparser release (v0.12.0) can be used in most modern languages</option>
      
      <option value="/news/2019/11/13/biodiversity-next/">Biodiversity-Next Conference</option>
      
      <option value="/news/2017/05/19/blackout/">Power down</option>
      
      <option value="/news/2016/11/15/names-in-november/">Names in November meeting</option>
      
      <option value="/news/2016/10/06/gnparser-0-3-3/">gnparser release v0.3.3</option>
      
      <option value="/news/2016/05/23/gnparser-0-3-0/">Scala-based gnparser v.0.3.0</option>
      
      <option value="/news/2015/11/30/ubio-is-back/">uBio and Nomenclator Zoologicus are back</option>
      
      <option value="/news/2015/11/30/gnparser-0-2-0/">Scala-based gnparser v.0.2.0</option>
      
      <option value="/news/2015/10/30/open-source-papers2/">Writing Papers as Open Source -- Solutions</option>
      
      <option value="/news/2015/10/19/open-source-papers/">Writing Papers as Open Source</option>
      
      <option value="/news/2015/10/14/bishop-museum-workshop/">Honolulu Workshop</option>
      
      <option value="/news/2015/10/13/BioGUID-Wins-Award/">BioGUID Wins Award!</option>
      
      <option value="/news/2015/10/08/BioGUID-Moved/">BioGUID merges with GN</option>
      
      <option value="/news/2015/10/06/gnub-proposal/">GNUB proposal submitted to NSF</option>
      
      <option value="/news/2015/10/05/ui-mid-november/">Global Names Gains Stable Funding</option>
      
      <option value="/news/2015/09/22/otol-eol/">Crossmapping Names by IDs</option>
      
      <option value="/news/2015/09/09/alex-myltsev/">Alexander Myltsev</option>
      
      <option value="/news/2015/09/03/sysopia/">Sysopia release v0.5.0</option>
      
      <option value="/news/2015/09/03/globalnames-site/">Site globalnames.org merged with GN blog</option>
      
      <option value="/news/2015/08/27/parser-3-4-1/">GN Parser v.3.4.1</option>
      
      <option value="/news/2015/08/20/parser-3-4-0/">GN Parser v.3.4.0</option>
      
      <option value="/news/2015/08/14/parser-3-3-0/">GlobalNames Parser v.3.3.0 adds host option for socket server</option>
      
      <option value="/news/2015/08/13/col-meeting/">Catalogue of Life Meeting in Champaign/Urbana</option>
      
      <option value="/news/2015/08/12/gn-resolver-vernaculars/">Global Names Resolver now shows vernacular names</option>
      
      <option value="/news/2015/06/01/biodiversity-3-2-1/">Global Names Parser v3.2.1</option>
      
      <option value="/news/2015/05/31/gn-uuid-0-5-0/">New UUID v5 Generation Tool -- gn_uuid v0.5.0</option>
      
      <option value="/news/2015/05/31/gn-crossmap-0-1-6/">Crossmap Tool v0.1.6</option>
      
      <option value="/news/2015/05/28/gn-crossmap-0-1-5/">Crossmap Tool v0.1.5</option>
      
      <option value="/news/2015/05/20/idigbio-client/">iDigBio API Client v0.1.1</option>
      
      <option value="/news/2015/05/14/parser-3-1-10/">Scientific Name Parser v3.1.10</option>
      
      <option value="/news/2015/05/12/gn-moves-to-illinois/">Global Names Grant Goes to Illinois</option>
      
      <option value="/news/2015/05/11/gn-crossmap-gem/">New tool to crossmap checklists</option>
      
      <option value="/news/2015/05/06/phylo-trees/">Kickoff  Meeting for Disseminating Phylogenetic Knowledge Project</option>
      
      <option value="/news/2015/05/03/higher-level-classification/">New Higher Level Classification for Catalogue of Life</option>
      
      <option value="/news/2015/05/03/sysopia-starts/">Starting GSOC 2015 Sysopia Project</option>
      
      <option value="/news/2015/05/02/species-file-group/">Species File Group -- new home for GN?</option>
      
      <option value="/news/2015/04/29/good-bad-names/">Good names, bad names?</option>
      
      <option value="/news/2015/04/28/idigbio-api-hackathon-meet/">iDigBio hackathon preparation</option>
      
      <option value="/news/2015/04/28/gsoc-netineti/">Launching NetiNeti Google Summer of Code Project</option>
      
      <option value="/news/2015/04/27/gsoc-2015-starts/">Google Summer of Code 2015 starts</option>
      
      <option value="/news/2015/04/22/thank_you_eol/">Gracious gift from Encylopedia of Life</option>
      
      <option value="/news/2015/04/17/gna_is_back/">GNA work continues</option>
      
    </optgroup>
  </select>
</div>


      <div class="unit four-fifths">
        
  <article>
  <h2>
    <a href="/news/2021/12/12/gnverifier-0.6.2/">
      GNverifier release v0.6.2, Advanced Search
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNverifier and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      12 Dec 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="https://github.com/gnames/gnverifier">GNverifier</a> can help to answer the following questions:</p>

<ul>
  <li>Is a name-string a real name?</li>
  <li>Is it spelled correctly, and if not, what might be the correct spelling?</li>
  <li>Is the name currently in use?</li>
  <li>If it is a synonym, what data sources consider to be currently accepted names?</li>
  <li>Is a name a homonym?</li>
  <li>What is a taxon that the name points to, and where is it placed in various
classifications?</li>
</ul>

<p>Biannual <a href="https://github.com/gnames/gnverifier">GNverifier</a> database update is done for 14 datasets, new datasets are
pending to be added.</p>

<p>The <a href="https://github.com/gnames/gnverifier/releases/tag/v0.6.2">v0.6.2</a> version of <a href="https://github.com/gnames/gnverifier">GNverifier</a> is out. It brings new features and some
changes in API, input and output format. The main change is an ability to
perform a search by name details, such as an abbreviated genus, author, year.</p>

<p>For example <code class="language-plaintext highlighter-rouge">g:M. sp:galloprovincialis au:Oliv. y:-1800</code> will search for a
name with genus starting with <code class="language-plaintext highlighter-rouge">M</code>, <code class="language-plaintext highlighter-rouge">galloprovincialis</code> as a specific epithet,
an author starting with <code class="language-plaintext highlighter-rouge">Oliv</code> and a year earlier or equal to <code class="language-plaintext highlighter-rouge">1800</code>.</p>

<p>Both name-verification and search return the same format of results. Because of
that we needed to change the name of some fields in the output, so their
meaning would correspond to both verification and search outputs.</p>

<ul>
  <li>
<code class="language-plaintext highlighter-rouge">inputId</code> changed to <code class="language-plaintext highlighter-rouge">id</code>
</li>
  <li>
<code class="language-plaintext highlighter-rouge">input</code> changed to <code class="language-plaintext highlighter-rouge">name</code>
</li>
  <li>
<code class="language-plaintext highlighter-rouge">preferredResults changed to </code>results`</li>
</ul>

<p>GNverifier’s <a href="https://apidoc.globalnames.org/gnames">API v1</a> still can be used
(it did not undrgo any format changes), but the <a href="https://verifier.globalnames.org">web-page</a> and command line app
<code class="language-plaintext highlighter-rouge">gnverifier</code> moved to use new
<a href="https://apidoc.globalnames.org/gnames-beta">API v0</a>. When the new API
stabilizes, it will be renamed to API v2.</p>

<h2 id="install-gnverifier-with-homebrew">Install GNverifier with Homebrew</h2>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>brew tap gnames/gn
brew <span class="nb">install </span>gnverifier
</code></pre></div></div>

<p>or</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>brew upgrade gnverifier
</code></pre></div></div>

<h2 id="changes-in-functionality-since-v030">Changes in functionality since v0.3.0</h2>

<ul>
  <li>
<a href="https://verifier.globalnames.org">web-page</a> shows the date when a name was imported into <a href="https://github.com/gnames/gnverifier">GNverifier</a> database.</li>
  <li>to make it easier to cite <a href="https://github.com/gnames/gnverifier">GNverifier</a>, there is a <a href="https://zenodo.org/record/5774421#.YbXkDYpMFOQ">DOI information</a> on
its <a href="https://github.com/gnames/gnverifier">GitHub page</a>.</li>
  <li>tab-delimited values (TSV) format is now supported.</li>
  <li>added <a href="https://www.algaebase.org">AlgaeBase</a> to data-sources.</li>
  <li>there are now options to return all matched results (use with caution, as
the output might be excessively big).</li>
  <li>show score details in the output and on the <a href="https://verifier.globalnames.org">web-page</a>.</li>
  <li>
<a href="https://github.com/gnames/gnverifier#advanced-search">advanced search</a> is added to both command line and web-based user interfaces.</li>
</ul>

<h2 id="deprecation-of-services">Deprecation of services.</h2>

<p><a href="http://gni.globalnames.org">GNI</a> is the oldest version of GN name-verification algorithms. Most of its
functionality now exists in <a href="https://github.com/gnames/gnverifier">GNverifier</a>, so the <a href="http://gni.globalnames.org">GNI web-site</a> is
going to be removed in the beginning of 2022.</p>

<p>The <a href="https://index.globalnames.org">Scala version of GNI</a> will also be
scheduled for removal.</p>

<p><a href="https://resolver.globalnames.org">GNresolver</a> will continue to run the longest.
It will not be deprecated until <a href="https://github.com/gnames/gnverifier">GNverifier</a> API v2 will be released.</p>

<p>If you use old systems, consider switching to <a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a>
because older systems will eventually be deprecated and stopped.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/10/28/gnparser-1.5.0/">
      GNparser (Go language) release 1.5.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      28 Oct 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="https://github.com/gnames/gnparser/releases/tag/v1.5.0">GNparser v1.5.0</a> is out. The following changes happened since 1.3.3:</p>

<h2 id="v150"><a href="https://github.com/gnames/gnparser/compare/v1.4.2...v1.5.0">v1.5.0</a></h2>

<p>Courtesy of Toby Marsden (<a href="https://github.com/tobymarsden" class="user-mention">@tobymarsden</a>) GNparser in ‘cultivars mode’ is able to parse graft-chymeras.
An example: “Cytisus purpureus + Laburnum anagyroides”.
Note that cultivar-specific names are not recognized outside of the cultivars mode.</p>

<h2 id="v142"><a href="https://github.com/gnames/gnparser/compare/v1.4.1...v1.4.2">v1.4.2</a></h2>

<p>Add support for authors with prefix ‘ver’.
An example: “Cryptopleura farlowiana (J.Agardh) ver Steeg &amp; Jossly”.</p>

<h2 id="v141"><a href="https://github.com/gnames/gnparser/compare/v1.4.0...v1.4.1">v1.4.1</a></h2>

<p>Fixed parsing of multinomials where authorship is not separated by space.
An example: “Paeonia daurica coriifolia(Rupr.) D.Y.Hong”.</p>

<h2 id="v140"><a href="https://github.com/gnames/gnparser/compare/v1.3.3...v1.4.0">v1.4.0</a></h2>

<p>Support an output in tab-separated values format. Quite often TSV format is much easier to parse than CSV.
Tab character is much less common inside of scientific names than <code class="language-plaintext highlighter-rouge">,</code> character.
Therefore just splitting by <code class="language-plaintext highlighter-rouge">\t</code> breaks a row into its components in many cases.
It is still recommended to use CSV libraries for any given language to avoid unexpected problems.</p>

<p>Authors that contain prefixes <code class="language-plaintext highlighter-rouge">do</code> and <code class="language-plaintext highlighter-rouge">de los</code> are now parsed correctly.
An example: “… de Cássia Silva do Nascimento …”</p>

<p>Authors with suffex <code class="language-plaintext highlighter-rouge">ter</code> are parsed correctly.
An example: “Dematiocladium celtidicola Crous, M.J. Wingf. &amp; Y. Zhang ter”.</p>

<p>Added support for non-ASCII apostrophes in Authors’ names.
An example: “Galega officinalis (L.) L`Hèr.”.</p>

<h2 id="new-gnparser-c-binding-package-for-nodejs-by-toby-marsden">New GNparser C-binding package for Node.js by Toby Marsden</h2>

<p>Toby Marsden also <a href="https://github.com/amazingplants/node-gnparser">created GNparser wrapper for Node.js</a>.</p>

<h2 id="update-of-c-binding-for-ruby-based-parser">Update of C-binding for Ruby-based parser</h2>

<p>The new v5.3.4 [<code class="language-plaintext highlighter-rouge">biodiversity</code> Ruby gem][biodiversity] is released using C-binding to GNparser v1.4.2</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/08/13/gnparser-1.3.3/">
      GNparser (Go language) release 1.3.3
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      13 Aug 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="https://github.com/gnames/gnparser/releases/tag/v1.3.3">GNparser v1.3.3</a> is out. The following changes happened since 1.3.0:</p>

<ul>
  <li>
    <p>GNparser received a <a href="https://zenodo.org/badge/DOI/10.5281/zenodo.5180412.svg">citable DOI</a> (v1.3.1)</p>
  </li>
  <li>
    <p><a href="https://github.com/gnames/gnparser/issues/53">Names-exceptions</a> that are hard to parse because they use some nomenclautral,
or biochemical terms as specific epithets are [now covered]. Some examples:</p>
  </li>
</ul>

<pre><code class="language-txt">Navicula bacterium
Xestia cfuscum
Bolivina prion
Bembidion satellites
Acrostichum nudum
Gnathopleustes den
</code></pre>

<ul>
  <li>2-letter generic names are appended with <a href="https://github.com/gnames/gnparser/issues/182">3 more genera</a> (<code class="language-plaintext highlighter-rouge">Do</code>, <code class="language-plaintext highlighter-rouge">Oo</code>, <code class="language-plaintext highlighter-rouge">Du</code>).</li>
</ul>

<pre><code class="language-txt">Do holotrichius (beetle)
Oo spinosum (arachnid)
Nu aakhu (annelid)
</code></pre>

<ul>
  <li>Known prefixes in authorships are appended with <a href="https://github.com/gnames/gnparser/issues/183">3 more prefixes</a> adding
support for authors like:</li>
</ul>

<pre><code class="language-txt">delle Chiaje
dos Santos
ten Broeke
ten Hove
</code></pre>

<ul>
  <li>
    <p>Parsing of names with <a href="https://github.com/gnames/gnparser/issues/184">ms in</a> like <code class="language-plaintext highlighter-rouge">Crisia eburneodenticulata Smitt ms in Busk, 1875</code> is 
supported (normalized to <code class="language-plaintext highlighter-rouge">Crisia eburneodenticulata Smitt ex Busk, 1875</code>).</p>
  </li>
  <li>
    <p>More <a href="https://github.com/gnames/gnparser/issues/184">annotation ‘stop’ words are added</a>, fixing parsing for names like:</p>
  </li>
</ul>

<pre><code class="language-txt">Crisina excavata (d'Orbigny, 1853) non (d'Orbigny, 1853)
Eulima excellens Verkrüzen fide Paetel, 1887
Porina reussi Meneghini in De Amicis, 1885 vide Neviani (1900)
</code></pre>

<p>Many thanks to <a href="https://github.com/diatomsRcool" class="user-mention">@diatomsRcool</a>, <a href="https://github.com/KatjaSchulz" class="user-mention">@KatjaSchulz</a> and <a href="https://github.com/joelnitta" class="user-mention">@joelnitta</a> for feature requests
and bug reports!</p>

<h2 id="clib-libraries-are-now-provided-with-each-new-release">Clib libraries are now provided with each new release</h2>

<p>GNparser can be incorportated via C-binding into many other languages. To make
such incorporation easier, the clib files for MacOS, Linux and MS Windows are
now provided with <a href="https://github.com/gnames/gnparser/releases/tag/v1.3.3">every new release</a>.</p>

<pre><code class="language-txt">macos-latest-clib.zip
ubuntu-latest-clib.zip
windows-latest-clib.zip
</code></pre>

<h2 id="update-of-c-binding-for-ruby-based-parser">Update of C-binding for Ruby-based parser</h2>

<p>The new v5.3.3 <a href="https://rubygems.org/gems/biodiversity"><code class="language-plaintext highlighter-rouge">biodiversity</code> Ruby gem</a> is released using C-binding to
GNparser v1.3.3</p>

<h2 id="gnparser-for-javascript">GNparser for JavaScript</h2>

<p><a href="https://github.com/tobymarsden" class="user-mention">@tobymarsden</a> incorporated GNparser C-binding into a <a href="https://github.com/gnames/gnparser/issues/181">Node.js package</a>. He plans to release the new package for NPM.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/07/14/go/">
      Why Go is a great language for biodiversity informatics
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      review
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      14 Jul 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>We, as a biodiversity informatics community, use quite a few programming
languages. Python, Java, R, Ruby are probably the most popular ones. After
working for the last few years with a language Go I think it would be very
beneficial to add it to the biodiversity informatics toolset.</p>

<p>When I started writing in Go, the language seemed to be a bit clunky, did not
have much of cool shiny language constructs that other modern languages have.  I
felt there were no “cute” syntax gold nuggets that are common in languages like
Ruby or Python. However, I think Go is amasing for solving many biodiversity
informatics problems.</p>

<p>I tried about 10 different languages for the last 12 years working on
Encyclopedia Of Life and Global Names projects, and for the last 5 years Go
is the language of choice for the vast majority of projects I make.</p>

<p>Lets look at the key features of the language that influenced my choice.</p>

<h2 id="feature-1-amazing-code-readability">Feature 1: Amazing code readability</h2>

<p>Biodiversity informatics runs mostly in academia. In academia projects
blossom and wither depending on availability of funds. Funds often appear and
dissapear, students and postdocs come and go, and quite often people have to
continue development of projects written by someone else.</p>

<p>Therefore, an ability to understand a code written by another person is crucial
for longevity of academia projects. So many important and interesting
developments become stale or died out only because their code was too hard to
understand by newcomers.</p>

<p>I found that code written in Go is one of the easiest to understand. The
simplicity of syntax was one of major design goals in Go, and most of the common
programming tasks can be solved only in one way. Go designers decided on a
brutally minimalistic approach, as a result Go has very little bloat and in vast
majority of cases has no duplication of features between its syntactic
constructs.</p>

<p>As a result a programming style of novice and experienced programmers does not
differ that much. It is great for supporting projects written by others, and for
learning from Go code itself how to solve common problems.</p>

<h2 id="feature-2-go-is-easy-to-learn">Feature 2: Go is easy to learn</h2>

<p>Minimalistic approach to the language design makes it possible to learn Go with
5-20 times less effort than other languages. Specification of the language is
tiny and just going through an <a href="https://tour.golang.org/welcome/1">‘official’ Go tutorial</a> is enough to
become productive after a few hours.</p>

<h2 id="feature-3-go-is-easy-to-maintain">Feature 3: Go is easy to maintain</h2>

<p>Developers of the language released Go v1.0 in 2012 and plan to support backward
compatibility for many years to come. Starting with Go v1.11 there is also
support for versioning of community packages as dependencies. As a result
it is possible to write a program or library and use it without any changes
inspite of new versions of Go appearing regularly. If a library or a program
depends on Open Source packages, the specific version of each package can be
set in a <code class="language-plaintext highlighter-rouge">go.mod</code> file.</p>

<p>New versions of Go appear regularly, and to use recently added features a
developer can provide Go version in <code class="language-plaintext highlighter-rouge">go.mod</code>.</p>

<p>Go team is very careful when they consider new features, and such features are
implemented only after much thought and discussion. As a result even massive new
functionalities do not create backward incompatibilities.</p>

<h2 id="feature-4-go-is-fast">Feature 4: Go is fast</h2>

<p>Go is much faster than such languages like Python, Ruby, Perl or R. In addition
concurrency and parallel execution of the code is a core concept of Go.
Writing concurrent code in Go is orders of magnitude easier than
in C, and after some practice becomes a second nature of a developer.</p>

<p>The combination of the language speed and parallel execution of the code on
multi-core CPUs allows to make Go programs up to 100 times faster than programs
written in interpreted languages. Go is somewhat slower than C, but Go programs
are often faster than analogous C programs, because of the ease of developing
concurrency and parallelism in Go.</p>

<h2 id="feature-5-speed-of-programming-is-great">Feature 5: Speed of programming is great</h2>

<p>Go is quite expressive, and, in my experience, the speed of developing in Go
is comparable with speed of development in Python or Ruby.</p>

<p>Also, running tests in Go is very fast, even big programs
can be tested without a long wait. It allows to run tests often, or run
them on each save.</p>

<h2 id="feature-6-very-fast-compilation">Feature 6: Very fast compilation</h2>

<p>Go is a compiled language, and requires compilation of its code before
execution. The speed of compiling the code is usually almost instantaneous,
and the language developers spend a lot of effort to keep it this way. The
speed of compiling is only slightly slower than executing Python or Ruby
code.</p>

<h2 id="feature-7-convenient-executable-files">Feature 7: Convenient executable files</h2>

<p>As a rule, Go compiles a program into a self-sufficient single executable file.
Most of Go programs have no external dependencies and, as a result, are very
easy to distribute and install. Downloading one file and running it is all what
is required. Go supports cross-compilation. It means that it can create
executable files for any supported OS on one computer. For example, a computer
running Linux can create executables for MS Windows, Mac OS, and Linux in one
go.</p>

<p>The size of the executable files is really tiny. For example, the size of a
name-finding project written in Ruby (GNRD) is about a gigabyte, while the
analogous code in Go is only 50 megabytes.</p>

<p>Small size of executables and lack of external dependencies make Go fantastic
for publishing projects as Docker images, or deplying such images on Kubernetes.</p>

<h2 id="feature-8-go-is-great-for-remote-apis-and-web-applications">Feature 8: Go is great for remote APIs and web-applications</h2>

<p>Writing an extemely fast web-server in Go is a trivial task. Developing
web-applications is quite easy with provided template methods. Distributing
web-applications is also easy, because all the static files are usually
included into a single binary. Besides a traditional REST approach to APIs
there is a very fast streaming gRPC approach.</p>

<h2 id="feature-9-rich-ecosystem-of-open-source-libraries">Feature 9: Rich ecosystem of Open Source libraries</h2>

<p>Inspite of being a relatively young language (v1.0.0 was released in 2012) Go
has a very active community and a large number of libraries for many development
needs. Specific libraries for biodiversity informatics are scarse so far, but
they start to appear. For example Global Names provides libraries for <a href="https://github.com/gnames/gnfinder">finding</a>,
<a href="https://github.com/gnames/gnparser">parsing</a> and <a href="https://github.com/gnames/gnverifier">verifying</a> of scientific names.</p>

<h2 id="feature-10-go-code-can-be-used-in-many-other-languages-via-c-bindings">Feature 10: Go code can be used in many other languages via C-bindings</h2>

<p>It is possible to compile Go into a C library, and use it via C-binding with
many other languages (C, R, Ruby, Python, Java for example).</p>

<p>It is also possible to incorporate C libraries into Go, however most of
functionality usually provided by C libraries is already implemented in pure Go
and, most of the time, introducing such depencencies is not required. For
example there are fantastic Go drivers for most popular databases.</p>

<h2 id="feature-11-there-are-very-good-tools-for-go-developers">Feature 11: There are very good tools for Go developers</h2>

<p>A lot of tools exist for Go that simplify development. Go creators pioneered an
idea of auto-formatting of a written code. There are fantastic plugins for Go
development in VS code, Vim, Emacs etc. JetBrains releases a standalone Go
development platform. Go plugins incorporate linting, formatting,
debugging, refactoring tools. Most of these tools can also be used as a
stand-alone command line applications. Go has powerful profiling and tracing
tools as well.</p>

<h2 id="conclusion">Conclusion</h2>

<p>I hope this post gave you an idea why Go is good for biodiversity informatics,
and you can try to start solving problems that appear in your work using Go
language. I would suggest to pick a small well-defined task, that requires fast
execution, read the <a href="https://tour.golang.org/welcome/1">tour of Go</a> tutotiral, install Go and its tools for your
favorite editor and start hacking the code!</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/07/05/gnparser-1.3.0/">
      GNparser (Go language) release 1.3.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      05 Jul 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="https://github.com/gnames/gnparser/releases/tag/v1.3.0">GNparser v1.3.0</a> is out. The major new functionality is an ability
to recognize and parse botanical cultivar names. This ability was added to
<a href="https://github.com/gnames/gnparser">GNparser</a> by <a href="https://github.com/tobymarsden">Toby Marsden</a>, thanks Toby for a great patch!</p>

<p>In addition to <a href="https://www.iapt-taxon.org/nomen/main.php">ICN</a> nomenclatural code for botanical scientific names there is
an <a href="https://www.ishs.org/sites/default/files/static/ScriptaHorticulturae_18.pdf">ICNCP</a> nomenclatural code for cultivated plants. <a href="https://www.ishs.org/sites/default/files/static/ScriptaHorticulturae_18.pdf">ICNCP</a> supports names
like:</p>

<div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Dahlia ‘Doris Day’
Fragaria 'Cambridge Favourite'
Rosa multiflora cv. 'Crimson Rambler'
</code></pre></div></div>

<p>Now, if these names are parsed as cultivars, the cultivar epithet is included
into a canonical form like <code class="language-plaintext highlighter-rouge">Rosa multiflora ‘Crimson Rambler’</code>. However such
addition would create problems for users who are more interested in the
canonical form according to <a href="https://www.iapt-taxon.org/nomen/main.php">ICN</a>: <code class="language-plaintext highlighter-rouge">Rosa multiflora</code>. Therefore, by default
<a href="https://github.com/gnames/gnparser">GNparser</a> will process such names according to <a href="https://www.iapt-taxon.org/nomen/main.php">ICN</a> code, providing a
warning:</p>

<div class="language-json highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">{</span><span class="w">
  </span><span class="nl">"quality"</span><span class="p">:</span><span class="w"> </span><span class="mi">2</span><span class="p">,</span><span class="w">
  </span><span class="nl">"warning"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Cultivar epithet"</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>If a user does need to treat such names as cultivars, there is a flag in the
command line app: <code class="language-plaintext highlighter-rouge">gnparser "Rosa multiflora cv. 'Crimson Rambler'" -C</code>. When
parsed with this flag, warning will disappear and canonical forms will include
cultivar information. <a href="https://parser.globalnames.org">GNparser web-interface</a> has a “cultivar checkbox” now,
as well as there is a “cultivar” option in <a href="https://apidoc.globalnames.org/gnparser">GNparser RESTful API</a>.</p>

<p>Parsed detailed data for cultivars:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnparser <span class="s2">"Rosa multiflora cv. 'Crimson Rambler'"</span> <span class="nt">-C</span> <span class="nt">-d</span> <span class="nt">-f</span> pretty
</code></pre></div></div>

<div class="language-json highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">{</span><span class="w">
  </span><span class="nl">"parsed"</span><span class="p">:</span><span class="w"> </span><span class="kc">true</span><span class="p">,</span><span class="w">
  </span><span class="nl">"quality"</span><span class="p">:</span><span class="w"> </span><span class="mi">1</span><span class="p">,</span><span class="w">
  </span><span class="nl">"verbatim"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa multiflora cv. 'Crimson Rambler'"</span><span class="p">,</span><span class="w">
  </span><span class="nl">"normalized"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa multiflora ‘Crimson Rambler’"</span><span class="p">,</span><span class="w">
  </span><span class="nl">"canonical"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="nl">"stemmed"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa multiflor ‘Crimson Rambler’"</span><span class="p">,</span><span class="w">
    </span><span class="nl">"simple"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa multiflora ‘Crimson Rambler’"</span><span class="p">,</span><span class="w">
    </span><span class="nl">"full"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa multiflora ‘Crimson Rambler’"</span><span class="w">
  </span><span class="p">},</span><span class="w">
  </span><span class="nl">"cardinality"</span><span class="p">:</span><span class="w"> </span><span class="mi">3</span><span class="p">,</span><span class="w">
  </span><span class="nl">"details"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="nl">"species"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
      </span><span class="nl">"genus"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"species"</span><span class="p">:</span><span class="w"> </span><span class="s2">"multiflora"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"cultivar"</span><span class="p">:</span><span class="w"> </span><span class="s2">"‘Crimson Rambler’"</span><span class="w">
    </span><span class="p">}</span><span class="w">
  </span><span class="p">},</span><span class="w">
  </span><span class="nl">"words"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="w">
    </span><span class="p">{</span><span class="w">
      </span><span class="nl">"verbatim"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"normalized"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Rosa"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"wordType"</span><span class="p">:</span><span class="w"> </span><span class="s2">"GENUS"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"start"</span><span class="p">:</span><span class="w"> </span><span class="mi">0</span><span class="p">,</span><span class="w">
      </span><span class="nl">"end"</span><span class="p">:</span><span class="w"> </span><span class="mi">4</span><span class="w">
    </span><span class="p">},</span><span class="w">
    </span><span class="p">{</span><span class="w">
      </span><span class="nl">"verbatim"</span><span class="p">:</span><span class="w"> </span><span class="s2">"multiflora"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"normalized"</span><span class="p">:</span><span class="w"> </span><span class="s2">"multiflora"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"wordType"</span><span class="p">:</span><span class="w"> </span><span class="s2">"SPECIES"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"start"</span><span class="p">:</span><span class="w"> </span><span class="mi">5</span><span class="p">,</span><span class="w">
      </span><span class="nl">"end"</span><span class="p">:</span><span class="w"> </span><span class="mi">15</span><span class="w">
    </span><span class="p">},</span><span class="w">
    </span><span class="p">{</span><span class="w">
      </span><span class="nl">"verbatim"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Crimson Rambler"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"normalized"</span><span class="p">:</span><span class="w"> </span><span class="s2">"‘Crimson Rambler’"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"wordType"</span><span class="p">:</span><span class="w"> </span><span class="s2">"CULTIVAR"</span><span class="p">,</span><span class="w">
      </span><span class="nl">"start"</span><span class="p">:</span><span class="w"> </span><span class="mi">21</span><span class="p">,</span><span class="w">
      </span><span class="nl">"end"</span><span class="p">:</span><span class="w"> </span><span class="mi">36</span><span class="w">
    </span><span class="p">}</span><span class="w">
  </span><span class="p">],</span><span class="w">
  </span><span class="nl">"id"</span><span class="p">:</span><span class="w"> </span><span class="s2">"38ff69c4-7e1a-5a26-bfc4-ee641fed6ba7"</span><span class="p">,</span><span class="w">
  </span><span class="nl">"parserVersion"</span><span class="p">:</span><span class="w"> </span><span class="s2">"nightly"</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>In addition Toby found and helped to fix problems with <a href="https://github.com/amazingplants/gnparser/blob/378a4d1d4907a3d696b2fe02a0336ca6445d2bc7/ent/stemmer/stemmer.go">stemming of hybrid
formulas</a> and with providing correct <a href="https://github.com/gnames/gnparser/issues/174">output for hybrid signs</a>
in the “details:words” section. Again, thanks for this contribution <a href="https://github.com/tobymarsden">Toby
Marsden</a>!</p>

<p>You can grab <a href="https://github.com/gnames/gnparser/releases/tag/v1.3.0">GNparser v1.3.0</a> binaries and follow <a href="https://github.com/gnames/gnparser#installation">installation instructions</a>,
or use <a href="https://brew.sh/">Homebrew</a> to install it on operating systems that support it:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>brew tap gnames/gn
brew <span class="nb">install </span>gnparser
</code></pre></div></div>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/06/19/gnfinder-0.14.1/">
      GNfinder release v0.14.1
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNfinder and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      19 Jun 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Find scientific names in plain texts, PDF files, MS documents etc.</p>

<p><a href="https://github.com/gnames/gnfinder"><strong>GNfinder</strong></a> is a program for finding scientific names in texts.
GNfinder exists for several years now and is responsible for creation of name
indices for Biodiversity Heritage Library (BHL) and HathiTrust Digital Library.
The program is fast enough to process 5 million pages of BHL in just a couple
of hours.</p>

<p>The new <a href="https://github.com/gnames/gnfinder/releases/v0.14.1"><strong>GNfinder</strong> v0.14.1</a> can find names not only
in plain UTF8-encoded texts, but also in a large variety of files including
PDF, MS Word, MS Excel, and images. In this blog post we describe how it
can be used.</p>

<p><a href="https://github.com/gnames/gnfinder"><strong>GNfinder</strong></a> code follows <a href="https://semver.org">Semantic Versioning</a> practices.  So
users need to be aware that for versions 0.x.x backward incompatible changes
might happen.</p>

<h2 id="summary">Summary</h2>

<p><a href="https://github.com/gnames/gnfinder"><strong>GNfinder</strong></a> is a command line application, that can also be used
as a RESTful service. In the near future it will also have a web-based user
interface and will run at <code class="language-plaintext highlighter-rouge">https://finder.globalnames.org</code></p>

<p>The program uses heuristic and Natural Language Processing (NLP) algorithms for
name finding.</p>

<h2 id="performance">Performance</h2>

<p>For a test we used 4MB PDF file that contains ~2000 unique names. These names
are mentioned in text ~13000 times.</p>

<p>Time for conversion to UTF8-encoded plain text: 2.5 sec</p>

<p>Time for name-finding: 0.4 sec</p>

<p>Time for name-verification of 2000 uniquely found names: 2.5 sec</p>

<h2 id="installation">Installation</h2>

<p>The program consists of one stand-alone file, so it is easy to install. The
binaries for MS WIndows, Mac OS or Linux can be downloaded from
<a href="https://github.com/gnames/gnfinder/releases/v0.14.1">GitHub</a>. In addition <a href="https://github.com/gnames/gnfinder"><strong>GNfinder</strong></a> can be
installed using a Homebrew package manager with the following terminal commands:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>brew tap gnames/gn
brew <span class="nb">install </span>gnfinder
</code></pre></div></div>

<p>For more detailed installation instructions see the <a href="https://github.com/gnames/gnfinder">documentation</a>
on GitHub.</p>

<h2 id="usages">Usages</h2>

<p><a href="https://github.com/gnames/gnfinder"><strong>GNfinder</strong></a> is a command line application. It requires an internet
connection for converting files to UTF8-encoded text and for name-verification.</p>

<p>To get help use:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnfinder <span class="nt">-h</span>
</code></pre></div></div>

<p>To get names from a UTF8-encoded text file (with -U flag no internet connection
is required):</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnfinder <span class="nt">-U</span> file-with-names.txt
</code></pre></div></div>

<p>To get names from any other kind of file:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnfinder file.pdf
</code></pre></div></div>

<p>To find names, to verify them and to output results in JSON format:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnfinder file.pdf <span class="nt">-v</span> <span class="nt">-f</span> pretty <span class="o">&gt;</span> file-names.csv
</code></pre></div></div>

<p>To convert PDF file into text:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnfinder <span class="nt">-I</span> file.pdf <span class="o">&gt;</span> file.txt
</code></pre></div></div>

<h2 id="tutorial">Tutorial</h2>

<p>I wrote a <a href="https://globalnames.org/docs/tut-gnfinder/">tutorial</a> how to exctract scientific names in parallel from a
large number of PDF files.</p>

<p>For more information read the <a href="https://github.com/gnames/gnfinder">documentation</a> on GitHub.</p>

<h2 id="restful-api">RESTful (API)</h2>

<p>You can run <a href="https://github.com/gnames/gnfinder"><strong>GNfinder</strong></a> as a <a href="https://apidoc.globalnames.org/gnfinder">RESTful API</a>
service as well. For now API can work only with UTF-8 encoded texts, but other
file formats will be available via API as well after completion of a
web-based user interface.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/04/10/gnverifier-0.3.0/">
      GNverifier release v0.3.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNverifier and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      10 Apr 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Very fast scientific name checker is out.</p>

<p>There are millions of checklists in use by scientists and nature enthusiasts.
Very often, such lists contain misspellings or outdated names. To help
researchers clean up their checklists and monitor their quality, we are
releasing <a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a> v0.3.0 written in Go language.</p>

<p><a href="https://github.com/gnames/gnverifier"><strong>GNverivier</strong></a> code follows <a href="https://semver.org">Semantic Versioning</a> practices.  So
users need to be aware that for versiong 0.x.x backward incompatible changes
might happen.</p>

<p>We released several implementations of name-verification
(reconciliation/resolution) before. All of them did not have enough speed for
verifying massive lists of scientific names. This release provides from 10x to
100x throughput improvements compared to older implementations.</p>

<h2 id="summary">Summary</h2>

<p><a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a> can help to answer the following questions:</p>

<ul>
  <li>Is a name-string a real name?</li>
  <li>Is it spelled correctly, and if not, what might be the correct spelling?</li>
  <li>Is the name currently in use?</li>
  <li>If it is a synonym, what data sources consider to be currently accepted names?</li>
  <li>Is a name a homonym?</li>
  <li>What is a taxon that the name points to, and where is it placed in various
classifications?</li>
</ul>

<p>Name verification and reconciliation involves several steps.</p>

<ul>
  <li>Exact match: input name matches canonical form located in one or more
data-sources.</li>
  <li>Fuzzy match: if no exact match is found, there is a fuzzy matching of
canonical forms</li>
  <li>Partial Exact match: if the previous two steps failed, we remove words from
the end or from the middle of a name and try to match what is left until we
end up with a bare genus.</li>
  <li>Partial Fuzzy match: in case if the partial exact match did not work, and the
remained name is not uninomial, we apply fuzzy matching algorithms.</li>
</ul>

<p>A scoring algorithm then sorts matched results. The <a href="https://verifier.globalnames.org/about">“About”
page</a> contains more detailed information about matching and scoring.</p>

<h2 id="performance">Performance</h2>

<p>We observe speeds of ~2,500 names per second for checklists that are coming
from optical character recognition process and contain many misspellings.</p>

<h2 id="usages">Usages</h2>

<p>The simplest way to use <a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a> is via
<a href="https://verifier.globalnames.org">web-interface</a>.  The online application emits results in HTML,
CSV and JSON formats and can process up to 5000 names per request.</p>

<p>For larger datasets, and as an alternative, there is a <a href="https://github.com/gnames/gnverifier/releases/tag/v0.3.0">command line
application</a> that can be downloaded for Windows, Mac, and Linux.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnverifier file-with-names.txt
</code></pre></div></div>

<p>This version adds an option <code class="language-plaintext highlighter-rouge">-c</code> or <code class="language-plaintext highlighter-rouge">--capitalize</code> to fix name-strings’
capitalization before verification. It is beneficial for web-interface, as it
allows users “to be lazy” when they try to match names.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>gnverifier <span class="s2">"drsophila melanogaster"</span> <span class="nt">-c</span> <span class="nt">-f</span> pretty
INFO[0000] Using config file: /home/dimus/.config/gnverifier.yaml.
<span class="o">{</span>
  <span class="s2">"inputId"</span>: <span class="s2">"b20a7c40-f593-5a68-a048-0a24742b4283"</span>,
  <span class="s2">"input"</span>: <span class="s2">"drsophila melanogaster"</span>,
  <span class="s2">"inputCapitalized"</span>: <span class="nb">true</span>,
  <span class="s2">"matchType"</span>: <span class="s2">"Fuzzy"</span>,
  <span class="s2">"bestResult"</span>: <span class="o">{</span>
    <span class="s2">"dataSourceId"</span>: 1,
    <span class="s2">"dataSourceTitleShort"</span>: <span class="s2">"Catalogue of Life"</span>,
    <span class="s2">"curation"</span>: <span class="s2">"Curated"</span>,
    <span class="s2">"recordId"</span>: <span class="s2">"2586298"</span>,
    <span class="s2">"localId"</span>: <span class="s2">"69bbaee49e7c2f749ee7712f3f168920"</span>,
    <span class="s2">"outlink"</span>: <span class="s2">"http://www.catalogueoflife.org/annual-checklist/2019/details/species/id/69bbaee49e7c2f749ee7712f3f168920"</span>,
    <span class="s2">"entryDate"</span>: <span class="s2">"2020-06-15"</span>,
    <span class="s2">"matchedName"</span>: <span class="s2">"Drosophila melanogaster Meigen, 1830"</span>,
    <span class="s2">"matchedCardinality"</span>: 2,
    <span class="s2">"matchedCanonicalSimple"</span>: <span class="s2">"Drosophila melanogaster"</span>,
    <span class="s2">"matchedCanonicalFull"</span>: <span class="s2">"Drosophila melanogaster"</span>,
    <span class="s2">"currentRecordId"</span>: <span class="s2">"2586298"</span>,
    <span class="s2">"currentName"</span>: <span class="s2">"Drosophila melanogaster Meigen, 1830"</span>,
    <span class="s2">"currentCardinality"</span>: 2,
    <span class="s2">"currentCanonicalSimple"</span>: <span class="s2">"Drosophila melanogaster"</span>,
    <span class="s2">"currentCanonicalFull"</span>: <span class="s2">"Drosophila melanogaster"</span>,
    <span class="s2">"isSynonym"</span>: <span class="nb">false</span>,
    <span class="s2">"classificationPath"</span>: <span class="s2">"Animalia|Arthropoda|Insecta|Diptera|Drosophilidae|Drosophila|Drosophila melanogaster"</span>,
    <span class="s2">"classificationRanks"</span>: <span class="s2">"kingdom|phylum|class|order|family|genus|species"</span>,
    <span class="s2">"classificationIds"</span>: <span class="s2">"3939792|3940206|3940214|3946159|3946225|4031785|2586298"</span>,
    <span class="s2">"editDistance"</span>: 1,
    <span class="s2">"stemEditDistance"</span>: 1,
    <span class="s2">"matchType"</span>: <span class="s2">"Fuzzy"</span>
  <span class="o">}</span>,
  <span class="s2">"dataSourcesNum"</span>: 28,
  <span class="s2">"curation"</span>: <span class="s2">"Curated"</span>
<span class="o">}</span>
</code></pre></div></div>

<p>It is possible to map a checklist to one of 100+ data sources aggregated in
<a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a>.</p>

<p>The following command will match all names from
<code class="language-plaintext highlighter-rouge">file-with-names.txt</code> against <code class="language-plaintext highlighter-rouge">the Catalogue of Life.</code></p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnverifier file-with-names.txt <span class="nt">-s</span> 1 <span class="nt">-o</span> <span class="nt">-f</span> pretty
</code></pre></div></div>

<p>It is also possible to run web-interface locally by running:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnverifier <span class="nt">-p</span> 4000
</code></pre></div></div>

<p>After running the command above, the interface can be accessed by a browser via
<code class="language-plaintext highlighter-rouge">http://localhost:4000</code> URL.</p>

<p>One can find a complete list of <code class="language-plaintext highlighter-rouge">gnverifier</code> by running:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gnverifier <span class="nt">-h</span>
</code></pre></div></div>

<h2 id="application-programming-interface-api">Application Programming Interface (API)</h2>

<p><a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a> does not keep all the data needed for processing
name-strings locally. It uses a <a href="https://app.swaggerhub.com/apis-docs/dimus/gnames/1.0.0">remote API</a> located at
<code class="language-plaintext highlighter-rouge">https://verifier.globalnames.org/api/v1</code>.</p>

<p>The RESTful API is public. It has an <a href="https://app.swaggerhub.com/apis-docs/dimus/gnames/1.0.0">OpenAPI description</a> and is
available for external scripts.</p>

<h2 id="deprecation-of-old-systems">Deprecation of old systems</h2>

<p>There are several older approaches to solve the same problem:</p>

<ul>
  <li><a href="https://gni.globalnames.org">GNI</a></li>
  <li><a href="https://resolver.globalnames.org">GNresolver</a></li>
  <li><a href="https://index.globalnames.org">GNI Scala</a></li>
</ul>

<p>If you use any of these, consider switching to <a href="https://github.com/gnames/gnverifier"><strong>GNverifier</strong></a>
because older systems will eventually be deprecated and stopped.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/04/10/gnarser-1.2.0/">
      GNparser (Go language) release 1.2.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      10 Apr 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="https://github.com/gnames/gnparser/releases/tag/v1.2.0">Version 1.2.0</a> of <a href="https://github.com/gnames/gnparser">GNparser</a> is out. It adds an option to parse low-case names
in case if a checklist does not follow nomenclatural standards.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>gnparser <span class="s2">"plantago major"</span> <span class="nt">--capitalize</span>
Id,Verbatim,Cardinality,CanonicalStem,CanonicalSimple,CanonicalFull,Authorship,Year,Quality
085e38af-e19b-56e5-9fec-5d81a467a656,plantago major,2,Plantago maior,Plantago major,Plantago major,,,4
</code></pre></div></div>

<p>Capitalization does not apply if for named hybrids</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>gnparser <span class="s2">"xAus bus"</span> <span class="nt">-c</span>
Id,Verbatim,Cardinality,CanonicalStem,CanonicalSimple,CanonicalFull,Authorship,Year,Quality
9b24b828-88a6-58b7-ac76-1342c8ac135d,xAus bus,2,Aus bus,Aus bus,× Aus bus,,,3
</code></pre></div></div>

<p>GNparser assigns Quality=4 (the worst) and issues a warning.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>gnparser <span class="s2">"plantago major"</span> <span class="nt">-c</span> <span class="nt">-f</span> pretty
<span class="o">{</span>
  <span class="s2">"parsed"</span>: <span class="nb">true</span>,
  <span class="s2">"quality"</span>: 4,
  <span class="s2">"qualityWarnings"</span>: <span class="o">[</span>
    <span class="o">{</span>
      <span class="s2">"quality"</span>: 4,
      <span class="s2">"warning"</span>: <span class="s2">"Name starts with low-case character"</span>
    <span class="o">}</span>
  <span class="o">]</span>,
  <span class="s2">"verbatim"</span>: <span class="s2">"plantago major"</span>,
  <span class="s2">"normalized"</span>: <span class="s2">"Plantago major"</span>,
  <span class="s2">"canonical"</span>: <span class="o">{</span>
    <span class="s2">"stemmed"</span>: <span class="s2">"Plantago maior"</span>,
    <span class="s2">"simple"</span>: <span class="s2">"Plantago major"</span>,
    <span class="s2">"full"</span>: <span class="s2">"Plantago major"</span>
  <span class="o">}</span>,
  <span class="s2">"cardinality"</span>: 2,
  <span class="s2">"id"</span>: <span class="s2">"085e38af-e19b-56e5-9fec-5d81a467a656"</span>,
  <span class="s2">"parserVersion"</span>: <span class="s2">"nightly"</span>
<span class="o">}</span>
</code></pre></div></div>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2021/03/23/gnparser/">
      GNparser (Go language) release 1.1.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      23 Mar 2021
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Scientific name parsing allows to determine a <em>canonical form</em>, the
<em>authorship</em> of a name, and receive other meta-information. Canonical forms are
crucial for comparing names from different data sources.</p>

<p>We are releasing <a href="https://github.com/gnames/gnparser"><strong>GNparser</strong></a> v1.1.0 written in Go language. We
support <a href="https://semver.org/">Semantic Versioning</a>, therefore it is a stable version. Output format,
functions, and settings are going to be backward compatible for many years
(until v2).</p>

<p>This is the 3rd implementation of name-parsing for Global Names Architecture
project. First one, written in Ruby, <a href="https://github.com/GlobalNamesArchitecture/biodiversity"><strong>biodiversity gem</strong></a>,
uses now the Go code of <a href="https://github.com/gnames/gnparser"><strong>GNparser</strong></a>. Second one, written in Scala
<a href="https://github.com/GlobalNamesArchitecture/gnparser">is archived</a>, and awaits for a new maintainer.</p>

<h2 id="summary">Summary</h2>

<p><a href="https://github.com/gnames/gnparser"><strong>GNparser</strong></a> is a sophisticated software, it is able to parse the
most complex scientific names. It is also very fast, and able to parse more
than 200 million names in an hour. The parser is a core component of many other
Global Names Architecture projects.</p>

<p>It can be used via:</p>

<ul>
  <li><a href="https://parser.globalnames.org">Web page</a></li>
  <li><a href="https://app.swaggerhub.com/apis-docs/dimus/gnparser/1.0.0">REST API</a></li>
  <li><a href="https://github.com/gnames/gnparser#command-line">Command line application</a></li>
  <li><a href="https://rubygems.org/gems/biodiversity">Ruby Gem</a></li>
  <li><a href="https://github.com/ropensci/rgnparser">R package</a></li>
  <li><a href="https://github.com/gnames/gnparser#use-as-a-library-in-go">Go Library</a></li>
</ul>

<p>We also provide
<a href="https://github.com/gnames/gnparser/blob/master/binding/main.go">C-binding</a> to
its code. This approach allows to incorporate <a href="https://github.com/gnames/gnparser"><strong>GNparser</strong></a> natively
into all languages that support C-binding (such as Java, Python, Ruby etc)</p>

<h2 id="improvements-since-the-last-scala-based-release-of-gnparser">Improvements since the last Scala-based release of GNparser</h2>

<ul>
  <li>
    <p>Speed — about 2 times faster than Scala-based version for CSV output,
             and about 8 times faster for JSON output.</p>
  </li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/27">Issue #27</a> — support for <code class="language-plaintext highlighter-rouge">agamosp. agamossp. agamovar.</code> ranks.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/28">Issue #28</a> — support for non-ASCII apostrophes.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/36">Issue #36</a> — support <code class="language-plaintext highlighter-rouge">_</code> as a space for files in Newick format.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/40">Issue #40</a> — support names where one of parentheses is missing.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/43">Issue #43</a> — support for <code class="language-plaintext highlighter-rouge">notho-</code> (hybrid) ranks.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/45">Issue #45</a> — support for <code class="language-plaintext highlighter-rouge">natio</code> rank.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/46">Issue #46</a> — support for <code class="language-plaintext highlighter-rouge">subg.</code> rank.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/48">Issue #48</a> — improve transliteration of diactritical characters.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/49">Issue #49</a> — support for outdated names with several hyphens in specific
                epithet.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/51">Issue #51</a> — distinguish between <code class="language-plaintext highlighter-rouge">Aus (Bus) cus</code> in botany and zoology
                (author or subgenus).</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/52">Issue #52</a> — support hyphen in outdated genus names.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/57">Issue #57</a> — warn when <code class="language-plaintext highlighter-rouge">f.</code> might mean either <code class="language-plaintext highlighter-rouge">filius</code> or <code class="language-plaintext highlighter-rouge">forma</code>.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/58">Issue #58</a> — distinguish between <code class="language-plaintext highlighter-rouge">Aus (Bus)</code> in ICN and ICZN
                (author or subgenus).</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/63">Issue #63</a> — normalize <code class="language-plaintext highlighter-rouge">format</code> to <code class="language-plaintext highlighter-rouge">f.</code> instead of <code class="language-plaintext highlighter-rouge">fm.</code>.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/60">Issue #60</a> — allow outdated ranks in form of Greek letters.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/61">Issue #61</a> — support authors’ names with <code class="language-plaintext highlighter-rouge">bis</code> suffix.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/66">Issue #66</a> — remove HTML tags from names, unless asked otherwise.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/67">Issue #67</a> — add name’s authorship to the “root” of JSON structure.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/68">Issue #68</a> — provide stemmed canonical form.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/69">Issue #69</a> — provide shared C library to bind <strong>GNparser</strong> to
                other languages.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/72">Issue #72</a> — parse surrogate names from BOLD project.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/75">Issue #75</a> — normalize subspecies to subsp.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/74">Issue #74</a> — support CSV output.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/78">Issue #78</a> — parse virus-like non-virus names correctly.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/79">Issue #79</a> — make CSV as a default output.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/80">Issue #80</a> — add cardinality to output.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/81">Issue #81</a> — support year ranges like ‘1778/79’.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/82">Issue #82</a> — parse authors with prefix <code class="language-plaintext highlighter-rouge">zu</code>.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/89">Issue #89</a> — allow <code class="language-plaintext highlighter-rouge">subspec.</code> as a rank.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/90">Issue #90</a> — allow <code class="language-plaintext highlighter-rouge">ß</code> in names.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/93">Issue #93</a> — parse <code class="language-plaintext highlighter-rouge">y</code> from Spanish papers as an author separator.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/127">Issue #127</a> — release a stable 1.0.0 version.</li>
  <li>
<a href="https://github.com/gnames/gnparser/issues/162">Issue #162</a> — support bacterial <code class="language-plaintext highlighter-rouge">Candidatus</code> names.</li>
</ul>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2019/12/11/gnfinder/">
      gnfinder release 0.9.1 -- bug fixes
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnfinder and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      11 Dec 2019
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>We are releasing a bug-fixing version (v0.9.1) of
<a href="https://github.com/gnames/gnfinder">gnfinder</a>, a written in Go project that
provides ability to search for scientific names in plain UTF-8 encoded texts.
It now has a better version and build timestamp report, its black list
dictionary expanded by 5 more words, and a bug fixed that prevents client
programs to break if name-verification returned an error instead of result.
We also started experimenting with making gnfinder available for other
languages through a C-shared library.</p>

<p>More can be found in gnfinder’s
<a href="https://github.com/gnames/gnfinder/blob/master/CHANGELOG.md">CHANGELOG file</a>.</p>

<p>The gnfinder code is used to create scientific name indices
in Biodiversity Heritage Library, HathiTrust Digital Library, and serves as an
engine for <a href="https://gnrd.globalnames.org">GNRD</a></p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2019/12/10/gnrd/">
      GNRD release (v0.9.0) switched to gnfinder from TaxonFinder/NetiNeti and became 25 times faster.
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      GNRD and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      10 Dec 2019
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Big change came to GNRD, a program David Shorthouse and Dmitry Mozzherin
released back in 2012. <a href="https://gnrd.globalnames.org">GNRD</a> is a web
application that is able to find scientific names in UTF-8 encoded plain
texts, in PDFs, MS Word and MS Excel documents, and even images.</p>

<p>For a long time it used two name-finding libraries TaxonFinder (developed by
Patrick Leary) and NetiNeti (developed by Lakshmi Manohar Akella). Both
projects served us well all these years using complementary heuristic and
natural language processing algorithms. <a href="https://www.biodiversitylibrary.org/">Biodiversity
Heritage Library</a>,
<a href="https://biostor.org/">BioStor</a> and many others used GNRD for detection of
scientific names for many eyars with success. However the speed for
large-scale name-finding was not satisfactory. To make large-scale
name-detection possible we developed
<a href="https://github/gnames/gnfinder">gnfinder</a> that also uses both heuristic and
NLP algorithms. With this new release of GNRD we substitute TasonFinder and
Netineti engines with gnfinder.</p>

<p>We tried hard to keep <a href="http://gnrd.globalnames.org/api">API</a> as close as
possible to how it was before, however there are a few changes, especially at
the name-verification (reconciliation and resolution) part. This change made
both name-finding and name-verification much faster with increased
quality. For example, it used to take 15 seconds to find names in a 1000-page
biological book. Now it takes only 0.5 seconds. GNRD tries to get names with
OCR errors as well, as a result you might get false positives. We do
recommend to use name-verification option to weed out such false results.</p>

<p>If you need to cite GNRD in a paper, v0.9.0 has a DOI attached:
<a href="https://zenodo.org/record/3569619"><code class="language-plaintext highlighter-rouge">10.5281/zenodo.3569619</code></a></p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2019/11/21/nsf1645959-outcomes/">
      Outcomes draft (Final Report for NSF award 1645959)
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      NSF
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      21 Nov 2019
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>We developed an ability to discover scientific names in millions of books in
a matter of hours. We are now able to finish name-finding for Biodiversity
Heritage Library (200 000 volumes) in 12 hours on a single laptop, and
HathiTrust Digital Library (18 million volumes, estimated 10% of all
published books) in 9 hours, using 50 servers. Scientific name-strings are
often represented differently by different sources. For example <em>Homo
sapiens</em> can be written as <em>Homo sapiens</em> Linnaeus, 1758, <em>Homo sapiens
sapiens</em>, <em>Homo sapiens</em> Linn. etc. To understand that all these variations
talk about the same species they need to be normalized. We built a tool that
normalizes 300 million name-strings per hour on a single computer. Also we
created a service that is able to verify scientific names in all existing
variations. The service validates names against well-known curated data
sources with a rate of 800,000 per hour. As a result of all these new
developments we have an opportunity to create a truly global index of
scientific names and mine biological information using scientific names as
anchors.</p>

<p>Modern biology depends heavily on scientific names. They were introduced in
18th century by Carl Linnaeus. For more than a quarter of a millennium they
are the basis of sharing information about biological taxa. Scientific names
are heavily used by researches, physicians, students, governmental agencies,
citizen scientists etc. Scientific names create a common basis for
biology and give us an ability to connect millions of medical and
biological facts together into a graph of knowledge.</p>

<p>However scientific names change quite often. According to our observations
there are about 3 different scientific names per species on average. On top
of that, scientific names usually are not rendered exactly the same way in
different publications, as we showed above, and such lexical variants of
names have to be reconciled. To collect a complete history of studies about
species we have to be able to take in account of these differences between
name variations. To provide an access to globally accumulated biodiversity
knowledge we have to scientific names in literature published in the last 250
years.</p>

<p>Our group have been participating in Global Names Architecture project since
2007 and tried to find solutions for these three major questions:</p>

<ul>
  <li>How to find scientific names in texts?</li>
  <li>How to “normalize” lexical and spelling inconsistencies in names belonging to
the same lexical group?</li>
  <li>How to connect a given name to all names ever given to its taxon, and find
which one of these names is currently accepted as valid?</li>
</ul>

<p>Former ABI innovation grant from NSF from 2011 (#1062387) gave us an
opportunity to develop functional prototypes of programs that were answering
these questions. However, these prototypes had been too slow to be applied
to whole corpus of biological and medical human knowledge.</p>

<p>We understood that for such programs to be useful on a global scale they must
be dramatically improved in speed, quality and scalability. They must be able
to use all resources of a multi-processor computer, and scale to multiple
machines with ease. Current ABI Development funding allowed us to achieve
speed and scalability goals.</p>

<p>We developed 3 major software tools:</p>

<h2 id="parser-of-scientific-names-gnparser">Parser of Scientific Names: gnparser</h2>

<p>This tool allows to dissect vast majority of scientific names into their
semantic elements. For example it is able to find out that “Carex scirpoidea
ssp. convoluta (Kük.) Dunlop” and “Carex scirpoidea Michaux var. convoluta
Kükenthal” both have the same conservative element “Cares scirpoidea
convoluta”, but vary in rank (variety vs subspecies) “Carex scirpoidea var.
convoluta” and “Carex scirpoidea subsp. convoluta”. Such matching is very
important for aggregation of biodiversity data from a variety of sources. The
gnparser program is designed to be small, easy to install and be blazingly
fast. It is currently used by several prominent biodiversity aggregators,
such as Encyclopedia of Life and Catalogue of Life, as well as Global Names
Index. It is able to process 300,000,000 names in an hour on one computer.</p>

<h2 id="global-names-index-gnindex">Global Names Index: gnindex</h2>

<p>The gnindex service aggregates scientific names from a large number of
biodiversity data sources and uses them to verify incoming names. A user is
able to send a list of name-strings and find out if they are known names, if
they contain spelling mistakes, if they are currently accepted as valid
names, etc. Such service is important for many use cases, for example as a
verification tool for species lists collected by citizen scientists, as a
curation tool for museum collections, for mapping country species lists of
government agencies to standardized data set, such as Catalogue of Life etc.
This tool relies heavily on gnparser to align data from many sources. It is
also indispensable for verifying (reconciling/resolving) name candidates
found in texts by our algorithms. This service currently has a throughput of
800,000 names per hour.</p>

<h2 id="global-names-finder-gnfinder">Global Names Finder: gnfinder</h2>

<p>The gnfinder tool uses heuristic and natural language processing algorithms
to detect scientific names in texts. In machine learning terms it belongs to
named entity recognition class of tools. It marks parts of texts that
potentially can be scientific names, and then sends them for remote
verification to gnindex. Average book in digital libraries is about 300 pages
long. On a single machine we are able to reach a rate of name-detection to
45,000 books an hour.</p>

<p>To summarize, these new tools give us ability to create indexes of scientific
names in a matter of hours for gigantic collections of literature and make
a global index for all digitized literature possible.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2019/11/20/col-gbif-bhl/">
      Biodiversity Heritage Library (BHL) link to Catalogue of Life Plus (CoL+)
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      meeting
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      20 Nov 2019
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Peter Schalk and Olaf Bánki (GBIF and CoL+) came to our group today. We had a
short meeting to start connecting names from Catalogue of Life to original
nomenclatural events collected in Biodiversity Heritage Library. We had a
remote connection to folks from BHL (Joel Richard and Mike Lichtenberg), and
CoL+ (Dave Remsen, Markus Doering).</p>

<p>Markus said that he will create a list of references for proto/basionyms in
CoL+, and starting December I will start trying to connect the names from BHL
to these references. If things work well, it will enhance usability of
Col/Col+ and BHL as well.</p>

<p>Markus created today a <a href="https://gitter.im/CatalogueOfLife/colbhl">gitter
account</a> that we can use for this
project.</p>

<p>Olaf will set a weekly meetings through December-January to keep us all in
sync for this project. Will be interesting to learn from IPNI about their
approach of doing pretty much the same task.</p>

<p>We also had a good conversation with Peter, Ed Devalt and Olaf about
importance of different biodiversity projects working together instead of
trying to re-implement each other ‘wheels’. I would say there is definitely a
progress in cooperation between different initiatives and it is good news for
biodiversity informatics.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2019/11/19/gnparser-clib/">
      gnparser release (v0.12.0) can be used in most modern languages
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      19 Nov 2019
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>A few days ago we released v0.12.0 of <a href="https://gitlab.com/gogna/gnparser/-/releases">gnparser (Go
version)</a>. This version made it
possible to compile gnparser algorithms into <a href="https://gitlab.com/gogna/gnparser/blob/master/binding/main.go">C-compatible
library</a>. Such
library makes it possible to use gnparser with its native speeds in any
language that supports binding to C. Such languages inlude Python, Ruby, Java
(via JNI), Rust, C, C++ and many others.</p>

<p>We already updated <a href="https://github.com/GlobalNamesArchitecture/biodiversity">Ruby’s biodiversity parser
gem</a> to take benefit
of a dramatic speed increase and parsing quality of <code class="language-plaintext highlighter-rouge">gnparser</code>.</p>

<p>Here are a quick benchmarks that compare how <code class="language-plaintext highlighter-rouge">biodiversity</code> performs before
and now:</p>

<table>
  <thead>
    <tr>
      <th>Program</th>
      <th>Version</th>
      <th>Full/Simple</th>
      <th style="text-align: right">Names/min</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>gnparser</td>
      <td>0.12.0</td>
      <td>Simple</td>
      <td style="text-align: right">3,000,000</td>
    </tr>
    <tr>
      <td>biodiversity</td>
      <td>4.0.1</td>
      <td>Simple</td>
      <td style="text-align: right">2,000,000</td>
    </tr>
    <tr>
      <td>biodiversity</td>
      <td>4.0.1</td>
      <td>Full JSON</td>
      <td style="text-align: right">800,000</td>
    </tr>
    <tr>
      <td>biodiversity</td>
      <td>3.5.1</td>
      <td>n/a</td>
      <td style="text-align: right">40,000</td>
    </tr>
  </tbody>
</table>

<p>With this improved speed Encyclopedia of Life, which is written in Ruby,
can process all their names using Ruby in less than 15 minutes.</p>

<p>README file of <code class="language-plaintext highlighter-rouge">gnparser</code> contains <a href="https://gitlab.com/gogna/gnparser#use-as-a-shared-c-library">instructions</a> how to make such c-shared
library and <code class="language-plaintext highlighter-rouge">biodiversity</code> code is a good example of connecting the library to
other languages.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2019/11/13/biodiversity-next/">
      Biodiversity-Next Conference
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      Meetings/Workshops
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      13 Nov 2019
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>For the last couple of years I have been working on making scientific-name
detection possible on a massive scale. The result of this work was creation
of several tools that increased the speed of name-finding dramatically:</p>

<ul>
  <li>name-parsing: <a href="https://parser.globalnames.org">gnparser</a>
</li>
  <li>name-detection: <a href="https://github.com/gnames/gnfinder">gnfinder</a>
</li>
  <li>name-curation: <a href="https://github.com/gnames/gntagger">gntagger</a>
</li>
  <li>name-verification (built by Alex Myltsev): <a href="https://index.globalnames.org">gnindex</a>
</li>
</ul>

<p>As a result we are able to scan large corpora of biodiversity
literature, such as Biodhiversity Heritage Library — BHL (200 000 volumes) and
HathiTrust Digital Library (16 000 000 volumes) in a matter of hours.</p>

<p>Recently, we presented our achievements at Biodiversity Next conference, an
expanded version of traditional yearly TDWG meeting. I had a talk at a
symposium <a href="https://biss.pensoft.net/browse_user_collection_documents?collection_id=125">Improving access to hidden scientific data in the Biodiversity
Heritage
Library</a>.
You can read about it in more details at a <a href="https://blog.biodiversitylibrary.org/2019/11/bhl-at-biodiversity-next.html">BHL blog
post</a>.</p>

<p>Such a significant collection of biodiversity literature as BHL gains
dramatically in usability from data mining efforts. For many years they use
the scientific names index that had been generated by our project: Global
Names Architecture. Several years ago generation of such index was a very
slow and laboreous task, that could not be repeated easily. With our recent
developments we are able to index BHL repeatedly with ease. It gives us
an opportunity to listen for feedback from BHL users, make incremental improvements
in our algorithms and increase the quality of scientific names index
continuously.</p>

<p>We are very interested to work with other people who try to enhance usability
of literature aggregators like BHL by developing natural language processing
and machine learning algorithms. During the conference several researches in
the field agreed to participate in a brain-storming workshop at Illinois
Natural History Survay, Champaign/Urbana to develop new technologies for
data-mining in BHL and other similar corpora. We are planning to organize
such a workshop in April/May 2020.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2017/05/19/blackout/">
      Power down
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      announcement
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      19 May 2017
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>On May 20th all our servers are going to be down because of a power maintenance
at our computer center. Sorry for inconvenience! We are going to be back as soon
as possible.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2016/11/15/names-in-november/">
      Names in November meeting
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      names
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      15 Nov 2016
    </span>
    <a href="https://github.com/deepreef" class="post-author">
      <img src="https://github.com/deepreef.png" class="avatar" alt="deepreef avatar" width="24" height="24">
      deepreef
    </a>
  </div>
  <div class="post-content">
    <p>Global Names developers Dmitry Mozzherin and Richard Pyle were invited to
attend a workshop called “Names in November”, organized by the Catalog of Life
and GBIF and hosted in Leiden.  The three-day meeting involved more than twenty
people from key taxonomic and nomenclatural organizations, and focused on
discussing ways that a global information system of taxonomy, including both
names data and accepted species information, could be designed to more
seamlessly interconnect biodiversity data through organism names.  Although the
theme of the meeting was certainly not new (many participants in this meeting
had attended similar meetings going back decades discussing essentially the
same idea), the tone of the discussion was refreshing in that it focused
comparatively little on politics and technical details, and instead
concentrated on identifying whether such a shared taxonomic infrastructure was
even possible (given the political, financial, and technical circumstances
currently existing within the main likely partners), and what conditions would
need to be met.</p>

<p>Many of the points that participants agreed on in terms of needs and services
very closely matched the fundamental goals and infrastructure we have developed
(and continue to develop) within the context of Global Names.  Now that GN is
much more closely coordinating with the Catalog of Life, GN data indexes and
services will likely play an important role in implementing the shared global
taxonomy resource envisioned during the meeting. Following this meeting, we
have a renewed sense of focus within GN development to finish harmonizing
integration of GNI and GNUB services, and especially to rapidly increase the
effort to bulk-populate GNUB from existing data.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2016/10/06/gnparser-0-3-3/">
      gnparser release v0.3.3
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      06 Oct 2016
    </span>
    <a href="https://github.com/alexander-myltsev" class="post-author">
      <img src="https://github.com/alexander-myltsev.png" class="avatar" alt="alexander-myltsev avatar" width="24" height="24">
      alexander-myltsev
    </a>
  </div>
  <div class="post-content">
    <p>We are happy to announce the release of <a href="https://github.com/GlobalNamesArchitecture/gnparser"><code class="language-plaintext highlighter-rouge">gnparser</code></a>.
Changes in the <a href="https://github.com/GlobalNamesArchitecture/gnparser/releases/tag/release-0.3.3">v0.3.3 release</a>:</p>

<ul>
  <li>add optionally showing canonical name UUID</li>
</ul>

<figure class="highlight"><pre><code class="language-json" data-lang="json"><span class="err">scala&gt;</span><span class="w"> </span><span class="err">fromString(</span><span class="s2">"Homo sapiens"</span><span class="err">).render(compact=</span><span class="kc">false</span><span class="err">,</span><span class="w"> </span><span class="err">showCanonicalUuid=</span><span class="kc">true</span><span class="err">)</span><span class="w">
</span><span class="err">res</span><span class="mi">0</span><span class="err">:</span><span class="w"> </span><span class="err">String</span><span class="w"> </span><span class="err">=</span><span class="w"> 
</span><span class="err">//</span><span class="w"> </span><span class="err">...</span><span class="w">
  </span><span class="nl">"canonical_name"</span><span class="w"> </span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
    </span><span class="nl">"id"</span><span class="w"> </span><span class="p">:</span><span class="w"> </span><span class="s2">"16f235a0-e4a3-529c-9b83-bd15fe722110"</span><span class="p">,</span><span class="w">
    </span><span class="nl">"value"</span><span class="w"> </span><span class="p">:</span><span class="w"> </span><span class="s2">"Homo sapiens"</span><span class="w">
  </span><span class="p">}</span><span class="err">,</span><span class="w">
</span><span class="err">//</span><span class="w"> </span><span class="err">...</span></code></pre></figure>

<ul>
  <li>
    <p>add year’s range to AST node that is encoded with <code class="language-plaintext highlighter-rouge">Year</code>’s field
<code class="language-plaintext highlighter-rouge">rangeEnd: Option[CapturePosition]</code></p>
  </li>
  <li>
    <p>parse names ending on hybrid sign (<a href="https://github.com/GlobalNamesArchitecture/gnparser/issues/88">#88</a>)</p>
  </li>
  <li>
    <p>support hybrid abbreviation expansions (<a href="https://github.com/GlobalNamesArchitecture/gnparser/issues/310">#310</a>)</p>
  </li>
  <li>
    <p>support raw hybrid formula parsing (<a href="https://github.com/GlobalNamesArchitecture/gnparser/issues/311">#311</a>)</p>
  </li>
  <li>
    <p>continuous build is moved to <a href="https://circleci.com/gh/GlobalNamesArchitecture/gnparser">CircleCI</a></p>
  </li>
  <li>
    <p>and many structural changes, bug-fixes and quality improvements.
They are described in the <a href="https://github.com/GlobalNamesArchitecture/gnparser/releases/tag/release-0.3.3">release</a> documenation.</p>
  </li>
</ul>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2016/05/23/gnparser-0-3-0/">
      Scala-based gnparser v.0.3.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      23 May 2016
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>To avoid confusion – gnparser is a new project, different from the formerly
released biodiversity parser.</p>

<p>We are happy to announce the second public release of Scala-based
<a href="https://github.com/GlobalNamesArchitecture/gnparser">Global Names Parser</a> or <code class="language-plaintext highlighter-rouge">gnparser</code>. There are many significant
changes in the <a href="https://github.com/GlobalNamesArchitecture/gnparser/releases/tag/release-0.3.0">v. 0.3.0 release</a>.</p>

<ul>
  <li>
    <p>Speed improvements. Parser is about 50% faster than already quite fast 0.2.0
version. We were able to parse 30 million names per CPU per hour with this
release.</p>
  </li>
  <li>
    <p>Compatibility with Scala 2.10.6: it was important for us to make the parser
backward compatible with this older version of Scala, because we wanted
to support Spark project.</p>
  </li>
  <li>
    <p>Compatibility with Spark v. 1.6.1. Now the parser can be used in BigData
projects running on Spark and massively parallelize parsing process using
Spark platform. We added <a href="https://github.com/GlobalNamesArchitecture/gnparser/blob/master/examples/spark/README.rst">documenation</a> describing how to use
the parser with either Scala or Python natively on Spark.</p>
  </li>
  <li>
    <p>Simplified parsing output in addition to “Standard output”: It analyzes the
name-strings and returns its id, canonical form, canonical form with
infraspecific ranks, authorship and a year.</p>
  </li>
  <li>
    <p>Improved and stabilized JSON fields. You can find complete description of
the parser JSON output in its <a href="http://globalnames.org/schemas/gnparser.json">JSON schema</a>. We based names of
fields on <a href="http://www.tdwg.org/standards/117">TDWG’s Taxon Concept Schema</a>, and we indend to keep
JSON format stable from now on.</p>
  </li>
  <li>
    <p>There were many structural changes, bug-fixes and quality improvements.
They are described in the <a href="https://github.com/GlobalNamesArchitecture/gnparser/releases/tag/release-0.3.0">release</a> documenation.</p>
  </li>
</ul>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/11/30/ubio-is-back/">
      uBio and Nomenclator Zoologicus are back
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      uBio
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      30 Nov 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="http://ubio.org">uBio</a> and <a href="http://ubio.org/NomenclatorZoologicus">Nomenclator Zoologicus online</a> experienced difficulties this year and had been down a
lot lately, mostly because there is no system administrator to look for them at
Marine Biological Laboratory anymore.</p>

<p>I moved uBio from old hardware at Marine Biological Laboratory to Google
Container Engine, and it is running again. Some functionality is not back yet,
mostly due to some hard-coded configuration parameters in files.  I hope
problems with the code will be fixed eventually by interested parties (I do
not plan to rewrite the code). I’ll coordinate my efforts with Dave Remsen and
Patrick Leary, and hopefully together we will preserve uBio for the community.</p>

<p>Please note that being a system administrator for uBio is not part of my job.
I like the project, I consider it to be a ‘precursor’ of GN, I will try my
best to keep it running on my spare time. MBL/WHOI library pays for the cloud.</p>

<p>Docker containers to run uBio are located at
<a href="https://hub.docker.com/u/mblab/dashboard/">dockerhub</a>. We use Docker and
Kubernetes at Google Container Engine to keep it alive.</p>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/11/30/gnparser-0-2-0/">
      Scala-based gnparser v.0.2.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnparser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      30 Nov 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>(Please note that gnparser is a new project, different from the formerly
released biodiversity parser.)</p>

<p>We are happy to announce a public release of <a href="https://github.com/GlobalNamesArchitecture/gnparser">Global Names Parser</a>
or <code class="language-plaintext highlighter-rouge">gnparser</code> – the first project that marks transition of Global Names
reconciliation and resolution services from “prototype” to “production”. The
<code class="language-plaintext highlighter-rouge">gnparser</code> project is developed by <a href="https://github.com/alexander-myltsev" class="user-mention">@alexander-myltsev</a> and <a href="https://github.com/dimus" class="user-mention">@dimus</a> in Scala
language. GNParser can be used as a library, a command line tool, a socket
server, a web-program and RESTful-service. It is easiest to try it at
<a href="http://parser.globalnames.org">parser.globalnames.org</a></p>

<p>Scientific names might be expressed by quite different name strings. Sometimes
the difference is just one comma, sometimes authors are included or excluded,
sometimes ranks are omitted. With all this variability “in the wild” we need to
figure out how to group all these different spelling variants.  Name parsing is
an unexpectedly complex and absolutely necessary step for connecting biological
information via scientific names.</p>

<p>In 2008 Global Names released <a href="https://github.com/GlobalNamesArchitecture/biodiversity">Biodiversity Gem</a> – a
scientific name parser written in Ruby for these purposes. The library in its
3 variants enjoyed a significant success – about <a href="https://rubygems.org/search?utf8=%E2%9C%93&amp;query=biodiversity">150 000
downloads</a> and a notion as the <a href="http://biogems.info/">most popular bio-library</a>
for Ruby language. It allowed to parse about 2-3 million names an hour, and
had been the basis of name reconciliation for many projects from the moment
of its publication.</p>

<p>GNParser is a direct descendant of the biodiversity gem. It serves the same
purpose and input/output format of both projects are similar. It also marks
eventual discontinuation of ‘biodiversity gem project’ and migration of all
Global Names code to the new <code class="language-plaintext highlighter-rouge">gnparser</code> library.</p>

<p>Why did we go through the pain of making a completely new parser from scratch?
The short answer is <em>scalability and portability</em>. We want to be able to
remove parsing step from being a bottleneck for any number of names thrown at
resolution services. For example finding all names in <a href="http://www.biodiversitylibrary.org/">Biodiversity Heritage
Library</a> took us 43 days 3 years ago. Parsing step alone took more than 1
day. If we want to improve algorithms of finding names in BHL – we cannot
wait 40 days. We want to be able to do it within one day and improve whole
BHL index every time our algorithms are enhanced significantly.</p>

<p>We have an ambitious goal of making time spent on sending names to resolution
services over internet and then time spent on transferring the answers back to
be the bottlenecks of our name matching services. For such speeds we need a
very fast parsing. Scala allows us to dramatically improve speed and
scalability of the parsing step.</p>

<p>Having a parser running in Java Virtual Machine environment allows us to give
biodiversity community a much more portable parsing tool. <a href="https://github.com/GlobalNamesArchitecture/gnparser/tree/master/examples">Out of the
box</a> parser library will work with Scala, Java, R, Jython and JRuby
directly. We hope that it will speedup and simplify many biodiversity projects.</p>

<p>This is the first public release of the library. Please download it, run
it, test it and give us your feedback, so we can improve it further. Happy
parsing <img class="emoji" title=":smile:" alt=":smile:" src="https://github.githubassets.com/images/icons/emoji/unicode/1f604.png" height="20" width="20"></p>

<p><strong>WARNING:</strong> JSON output format might change slightly or dramatically, as we
are in the process of refining it. The JSON format should be finalized for
version 0.3.0</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/30/open-source-papers2/">
      Writing Papers as Open Source -- Solutions
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      openscience
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      30 Oct 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>I decided to figure out how can I write scientific papers in truly Open Source
fashion. And here are practical decisions that allowed me to do it:</p>

<h2 id="criteria">Criteria</h2>

<ul>
  <li>Paper draft is under true revision control system</li>
  <li>Open access from the very beginning</li>
  <li>Open tools/standards</li>
</ul>

<h2 id="solutions">Solutions</h2>

<h3 id="revision-control">Revision control</h3>

<p>To use full power of revision control system a project should be mostly in text
format of some sort. We currently keep practically all our code on GitHub, so
Git was a natural choice.</p>

<h3 id="open-toolsformats">Open tools/formats</h3>

<p>I decided to go with LaTeX, as it is tried and powerful markup, very well
suited for scientific writing. It allows to work with plain text, so we can
easily keep revisions in Git.</p>

<p>Vim is my editor of choice, but nothing prevents me or co-authors to use any
other modern text editor for LaTeX.</p>

<h3 id="open-access">Open access</h3>

<p>With LaTeX and Git it is easy to provide early access to the work in progress,
especially with 2 commercial products that give free access for open projects
– GitHub and OverLeaf. Overleaf supports git, although not as well as GitHub
does. So currently is it better to have GitHub as the main repository, and keep
Overleaf as a glorified viewer and use it only as a secondary repository.
Another useful tool is Mendeley for finding and organazing bibliography.</p>

<h2 id="final-result">Final Result</h2>

<p>I am still learning my ropes, but excited about the progress. And the paper
about <a href="https://github.com/GlobalNamesArchitecture/gnparser">Global Names Parser</a> is now on <a href="https://github.com/gnpapers/gnparser">GitHub</a>
and <a href="https://www.overleaf.com/read/zqsnwhrmpxrm">OverLeaf</a>! Overleaf allows anybody interested to
see the paper in user-friendly PDF format. It also simplifies submission of
papers to a large variety of open access journals.</p>

<p>I created a post on my personal blog describing how did I <a href="http://mozzherin.org/2015/11/01/open-paper.html">set up my
system</a> with LaTeX, vim and tmux.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/19/open-source-papers/">
      Writing Papers as Open Source
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      openscience
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      19 Oct 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Does a culture exist out there which considers a process of writing scientific
papers to be akin to writing an open source code?</p>

<p>For the last 8 years I had been blessed being paid for doing open source
development. It means that for that long about everything I do is almost
instantly available publicly. This model fits my way of thinking, my values,
and I see an advantage in making all I do available for public to see, comment,
and enhance.</p>

<p>Now I am writing a paper, and I feel being thrown into “dark ages”. Whole paper
writing and publishing culture was one of the reasons I left molecular biology
and went to programming. I assume the following is usually true when people
write a scientific paper –</p>

<ul>
  <li>People normally do not share publicly what they work on until it is published.</li>
  <li>People normally use proprietary software to write papers</li>
  <li>People often loose their copyright or ability to share their work when their paper is accepted by a journal</li>
</ul>

<p>Obviously there is a progress with the last point, but what about other 2?</p>

<ul>
  <li>Can I use public revision control system when I am writing a paper?</li>
  <li>Can I publish using a revision control system from the very first paragraph for all to see?</li>
  <li>What open standards/tools can I use (LaTeX, or even markdown?) for writing a paper</li>
  <li>Can I consider publishing paper to be a ‘release’, like for a program?</li>
  <li>Should electronic version be frozen? Can it evolve after publication?</li>
</ul>

  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/14/bishop-museum-workshop/">
      Honolulu Workshop
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gna
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      14 Oct 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Last week (October 5-9 2015) <a href="https://github.com/deepreef" class="user-mention">@deepreef</a>, <a href="https://github.com/dimus" class="user-mention">@dimus</a>, <a href="https://github.com/alexander-myltsev" class="user-mention">@alexander-myltsev</a> had a
workshop in Honolulu at <a href="http://www.bishopmuseum.org">Bishop Museum</a> to sync ideas, learn more about each
other work, and design new generation of services. The meeting had been
productive and I think in the end our two GN groups get integrated. We
are moving all our code under one roof at GitHub now.</p>

<p>We had an interesting meeting with <a href="https://github.com/jar398" class="user-mention">@jar398</a> from <a href="http://opentreeoflife.org/">Open Tree of Life</a> trying to
figure out how can we connect OTOL with all the other resources on the web, and
<a href="https://github.com/deepreef" class="user-mention">@deepreef</a> suggested to use his <a href="http://bioguid.org/">BioGuid</a> project for these purposes. We moved
BioGuid to github and added all the GitHubbish bells and whistles like a
<a href="http://blog.bioguid.org">blog</a> and [gitter][bioguid-gitter] for example to the project.
I think it is pretty cool that we will have a <a href="https://github.com/GlobalNamesArchitecture/BioGUID/issues/10">downloadable csv</a>
file from all the IDs <a href="https://github.com/deepreef" class="user-mention">@deepreef</a> collected that can be used by all other
projects in new exiting ways.</p>

<p>Another interesting conversation was with <a href="http://phylotastic.org/">Phylotastic</a> project. We worked on
an idea of making an application which will allow convert pictures of
scientific names people take at museums or from pages of research papers into
texts, extract names appeared there and build phylo-trees from these names
using Open Tree. Also the app would show pictures from Encyclopedia of Life and
pages from Wikipedia. Such app will mash up interfaces of Global Names to find
and reconcile names, Open Tree to build trees, and EOL to get information
about species.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/13/BioGUID-Wins-Award/">
      BioGUID Wins Award!
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      bioguid
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      13 Oct 2015
    </span>
    <a href="https://github.com/deepreef" class="post-author">
      <img src="https://github.com/deepreef.png" class="avatar" alt="deepreef avatar" width="24" height="24">
      deepreef
    </a>
  </div>
  <div class="post-content">
    <p>As we <a href="http://globalnames.org/news/2015/10/08/bioguid-moved/">annouced previously</a>, <a href="http://bioguid.globalnames.org">BioGUID.org</a> has been incorporated into the Global Names suite of indexes and services. Within two days of this happening, we recieved some wonderful news: BioGUID.org <a href="http://www.gbif.org/newsroom/news/2015-ebbe-nielsen-challenge-winners">won second place</a> in the GBIF Ebbe Nielsen Challenge! We’re very excited about this recognition, and it re-inforces our decision to incorporate BioGUID into the GNA system. You can follow continuing developments on the new <a href="http://blog.globalnames.org">BioGUID Blog</a>.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/08/BioGUID-Moved/">
      BioGUID merges with GN
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnub
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      08 Oct 2015
    </span>
    <a href="https://github.com/deepreef" class="post-author">
      <img src="https://github.com/deepreef.png" class="avatar" alt="deepreef avatar" width="24" height="24">
      deepreef
    </a>
  </div>
  <div class="post-content">
    <p><a href="http://bioguid.globalnames.org">BioGUID.org</a>, an indexing service that cross-links identifiers assigned to data objects in the biodiversity information universe, has now been incorporated into the Global Names suite of indexes and services. BioGUID.org represents the third major data component of Global Names (along side <a href="http://gni.globalnames.org">GNI</a> and <a href="http://gnub.globalnames.org">GNUB</a>), and replaces a less robust identifier linking service that had previously been included within GNUB.  In addition to the crucial role of cross-linking identifiers within the general GN architecture itself, the broader function of BioGUID.org falls within the scope of Global Names in the sense that identifiers can be thought of as names, and names play the same functional role as identifiers.</p>

<p>We are currently in the process of porting BioGUID into the <a href="https://github.com/GlobalNamesArchitecture">GN GitHub</a>, and you can follow developments on the new <a href="http://blog.bioguid.org">BioGUID Blog</a>.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/06/gnub-proposal/">
      GNUB proposal submitted to NSF
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gnub and nsf
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      06 Oct 2015
    </span>
    <a href="https://github.com/deepreef" class="post-author">
      <img src="https://github.com/deepreef.png" class="avatar" alt="deepreef avatar" width="24" height="24">
      deepreef
    </a>
  </div>
  <div class="post-content">
    <p><a href="http://www.bishopmuseum.org">Bishop Museum</a>, in partnership with the <a href="http://www.catalogueoflife.org/">Catalogue of Life</a>, <a href="https://www.idigbio.org/">iDigBio</a>,
<a href="http://www.gbif.org">GBIF</a>, <a href="http://marinespecies.org/">WoRMS</a>, <a href="http://plazi.org/">PLAZI</a>, <a href="http://biodiversitylibrary.org/">BHL</a>, the <a href="http://www.nadsdiptera.org/ICD/ICDhome.htm">International Congresses of Dipterology</a>,
and <a href="http://www.pensoft.net/">Pensoft Publishers</a>, submitted a proposal to the <a href="http://www.nsf.gov">U.S. National Science
Foundation</a>’s <a href="http://www.nsf.gov/funding/pgm_summ.jsp?pims_id=5444">Advances in Biological Informatics (ABI)</a> program, to develop
the Global Names Usage Bank.  This proposed project will dramatically improve
the core infrastructure behind GNUB in particular, and globalnames in general.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/10/05/ui-mid-november/">
      Global Names Gains Stable Funding
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      speciesfilegroup
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      05 Oct 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Moving Global Names to a stable ground… On October 1st I did sign a job offer
from <a href="http://software.speciesfile.org/HomePage/Software/SoftwareHomePage.aspx">Species File Group</a> and going to start at the new position on
November 16th. This position is supported by a fund created by <a href="http://wwx.inhs.illinois.edu/directory/show/dceades">David
Eades</a> (thank you David) and allows to think of grants not like a
vehicle for GN’s survival, but means of further enhancement of the system. I am
honored and touched that David, and <a href="http://software.speciesfile.org/HomePage/Software/SoftwareHomePage.aspx">Species File Group</a> created this
amazing opportunity for Global Names. This move should also lead to a tight
integration between <a href="http://catalogueoflife.com">Catalogue of Life</a> and Global Names, which in my view
is a win/win situation for both projects and for biodiversity.</p>

<p>Our next steps are releasing JVM-based <a href="https://github.com/GlobalNamesArchitecture/gnparser">scientific name parser</a>,
building scalable, reliable and fast name resolution service and integratation
of  name resolution with the key Global Names project developed by <a href="https://www.ted.com/talks/richard_pyle_dives_the_twilight_zone?language=en">Rich
Pyle</a> and <a href="https://github.com/whittonr">Rob Whitton</a> – <a href="http://globalnames.org/docs/gnub/">Global Names Usage
Bank</a> for which Rich just submitted a grant proposal.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/09/22/otol-eol/">
      Crossmapping Names by IDs
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      crossmap
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      22 Sep 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Yesterday <a href="https://github.com/hyanwong" class="user-mention">@hyanwong</a> and <a href="https://github.com/jhpoelen" class="user-mention">@jhpoelen</a> mentioned on <a href="https://gitter.im/EOL/eol">eol</a> and <a href="https://gitter.im/GlobalNamesArchitecture/GlobalNames">global
names</a> gitter chats that it would be great to be able to crossmap
names from different sources by IDs.</p>

<p>What would be a use for such a crossmap? It would allow to quickly mashup data
from various projects in interesting ways. For example to show images of
species from <a href="http://eol.org/api">EOL API</a> on phylo-trees generated using <a href="https://github.com/OpenTreeOfLife/opentree/wiki/Open-Tree-of-Life-APIs">Open Tree of Life API</a>.</p>

<p><a href="http://www.youtube.com/watch?v=VuWxlsw-ftQ"><img src="http://img.youtube.com/vi/VuWxlsw-ftQ/0.jpg" alt="TreeOfLife"></a></p>

<p><a href="https://github.com/OpenTreeOfLife/opentree/wiki/Open-Tree-Taxonomy">OpenTree Taxonomy</a> has mapping of OTT IDs to <a href="http://www.ncbi.nlm.nih.gov/">NCBI</a> IDs.
<a href="http://eol.org">Encyclopdia of Life</a> also has mapping of NCBI IDs to EOL IDs. So if
someone wants to map NCBI names to EOL using the same algorithm as EOL used –
they only need to query data about IDs. Even better – it would create a very
fast connection of one aggregator (Open Tree), to another aggregator (EOL)
through IDs of other sources without doing explicit name resolution.</p>

<p>Such queries would be much faster, as they would be just about comparing
indexed columns in a table. However the quality of the results from such
approach would depend on the quality of name resolution used by the
aggregators.</p>

<p>I am thinking about trying just that. As a pilot – we can generate Darwin Core
Archive files from OTT and EOL which would contain information about IDs from
other sources. Then we will need to add an API that would make it possible to
run queries from such information.</p>

<p>Another good suggestion from <a href="https://github.com/jhpoelen" class="user-mention">@jhpoelen</a> – is to publish data about this kind of
crossmap as a csv file, that can be easily put in some kind of a database and
used separately on its own.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/09/09/alex-myltsev/">
      Alexander Myltsev
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      globalnames
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      09 Sep 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Global Names is happy to welcome a new member – Alexander Myltsev
(<a href="https://github.com/alexander-myltsev" class="user-mention">@alexander-myltsev</a> on GitHub). Alexander is of <a href="https://github.com/sirthias/parboiled2">parboiled2</a> fame. Parboiled2
is a <a href="https://en.wikipedia.org/wiki/Parsing_expression_grammar">Parsing Expressed Grammar</a> parser for Scala, and it did originate
from Alex’ code which he wrote as a Google Summer of Code participant in 2013.</p>

<p><img src="/img/blog/amyltsev.jpg" alt="Alexander Myltsev"></p>

<p>Alexander lives in Moscow and currently works on a port of biodiversity parser
to Scala; the project is called <a href="https://github.com/GlobalNamesArchitecture/gnparser">gnparser</a>. The new parser is compatible with
Java, JRuby, Jython and everything else written for Java virtual machine
environment. When the parser is ready it will be the basis of a new Scala-based
collection of GN tools.</p>

<p>Alexander had been working with us for a few months now, but I had been waiting
with the announcement until major paperwork hurdles were solved.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/09/03/sysopia/">
      Sysopia release v0.5.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      sysopia and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      03 Sep 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Last week was the end of the Google Summer or Code season. Out of two projects
that we had been mentoring one was not really about biology. It was a project
for system administrators. A <a href="https://github.com/EOL/sysopia">visualization tool</a> for staistics about
cpu usage, memory, disk space etc.</p>

<p><img src="/img/blog/sysopia.png" alt="Sysopia"></p>

<p>Everybody who runs involved biodiversity informatics projects knows how
important it is to monitor your systems. There are several open source tools
for that – Nagios, Sensu, Graphite, Systemd, Collectd…</p>

<p>Our monitoring system of choice is <a href="https://sensuapp.org/">Sensu</a>. It is very flexible and powerful
tool, well designed and suitable for large number of tasks. One of these is
collecting statistics from computers and store them in about any kind of
database. As a result Sensu can be used for monitoring critical events and for
collecting data about systems. The question is however how to visualize all the
collected data.</p>

<p>We designed <a href="https://github.com/EOL/sysopia">Sysopia</a> to do exactly that. During the summer <a href="https://github.com/vpowerrc" class="user-mention">@vpowerrc</a>
expanded original prototype and created powerful and flexible visualization
tool which is capable to give system administrator an understanding what is
happening with 2-20 computers at a glance, receive life updates, and compare
today’s statistics with up to one year of data. We already use Sysopia in
production, and we are going to deploy it for Global Names as soon as our
new computers are in place.</p>

<p>You can read more about sysopia on its <a href="https://eol.github.io/sysopia">help</a> page</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/09/03/globalnames-site/">
      Site globalnames.org merged with GN blog
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      globalnames
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      03 Sep 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>For quite a while we used to have a Drupal-based site for GlobalNames. As we do
have now a <a href="http://github.org/jekyll/jekyll">Jekyll</a>-based blog, it was logical to move our static site as well.
And now it did happen – both of them are accessible via <a href="http://globalnames.org">globalnames.org</a></p>

<p>This new site will continue to be an ‘official’ blog for news about GNA, we
will publish information about new releases of software here, documents and
discussions about scientific names.</p>

<p>One great thing about this move – it is possible for anybody with account at
<a href="https://github.com">github</a> to participate – if you want to add a document, or a blog item –
just fork the <a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io">repository</a>, add a post to the <a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io/tree/master/_posts">_posts</a> directory and
send a pull request. At some point we will add detailed instructions how to do
that.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/08/27/parser-3-4-1/">
      GN Parser v.3.4.1
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      parser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      27 Aug 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>New version 3.4.1 of GlobalNames Parser gem <a href="http://rubygems.org/gems/biodiversity">biodiversity</a> is out.
It adds ability to parse authors names starting with <code class="language-plaintext highlighter-rouge">d'</code> like
<code class="language-plaintext highlighter-rouge">Cirsium creticum d'Urv. subsp. creticum</code></p>

<p>which now is parsed correctly</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/08/20/parser-3-4-0/">
      GN Parser v.3.4.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      parser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      20 Aug 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>New version 3.4.0 of GlobalNames Parser gem <a href="http://rubygems.org/gems/biodiversity">biodiversity</a> is out. It adds new
method that allows to add infraspecific ranks to canonical forms after the
fact.</p>

<p>It was possible to add ranks in canonical forms before using the following
code:</p>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="nb">require</span> <span class="s2">"biodiversity"</span>
<span class="n">parser</span> <span class="o">=</span> <span class="no">ScientificNameParser</span><span class="p">.</span><span class="nf">new</span><span class="p">(</span><span class="ss">canonical_with_rank: </span><span class="kp">true</span><span class="p">)</span>
<span class="n">parsed</span> <span class="o">=</span> <span class="n">parser</span><span class="p">.</span><span class="nf">parse</span><span class="p">(</span><span class="s2">"Carex scirpoidea subsp. convoluta (Kük.)"</span><span class="p">)</span>
<span class="n">parsed</span><span class="p">[</span><span class="ss">:scientificName</span><span class="p">][</span><span class="ss">:canonical</span><span class="p">]</span>
<span class="c1">#output: Carex scirpoidea subsp. convoluta</span></code></pre></figure>

<p>Now it is also possible to add ranks to canonical forms after the fact using
static method <code class="language-plaintext highlighter-rouge">ScientificNameParser.add_rank_to_canonical</code></p>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="nb">require</span> <span class="s2">"biodiversity"</span>
<span class="n">parser</span> <span class="o">=</span> <span class="no">ScientificNameParser</span><span class="p">.</span><span class="nf">new</span>
<span class="n">parsed</span> <span class="o">=</span> <span class="n">parser</span><span class="p">.</span><span class="nf">parse</span><span class="p">(</span><span class="s2">"Carex scirpoidea subsp. convoluta (Kük.)"</span><span class="p">)</span>
<span class="n">parsed</span><span class="p">[</span><span class="ss">:scientificName</span><span class="p">][</span><span class="ss">:canonical</span><span class="p">]</span>
<span class="c1">#output: Carex scirpoidea convoluta</span>
<span class="no">ScientificNameParser</span><span class="p">.</span><span class="nf">add_rank_to_canonical</span><span class="p">(</span><span class="n">parsed</span><span class="p">)</span>
<span class="n">parsed</span><span class="p">[</span><span class="ss">:scientificName</span><span class="p">][</span><span class="ss">:canonical</span><span class="p">]</span>
<span class="c1">#output: Carex scirpoidea subsp. convoluta</span></code></pre></figure>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/08/14/parser-3-3-0/">
      GlobalNames Parser v.3.3.0 adds host option for socket server
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      parser and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      14 Aug 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>New version 3.3.0 of GlobalNames Parser gem biodiversity is out. It adds new
option to socket server <code class="language-plaintext highlighter-rouge">-H</code> or <code class="language-plaintext highlighter-rouge">--host</code> to change default 127.0.0.1
host setting.  To see full set of options run</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">gem <span class="nb">install </span>biodiversity
parserver <span class="nt">-h</span></code></pre></figure>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/08/13/col-meeting/">
      Catalogue of Life Meeting in Champaign/Urbana
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      catalogue-of-life
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      13 Aug 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>There was a short meeting about <a href="http://www.catalogueoflife.org/">Catalogue of Life</a> future directions
organized at <a href="http://software.speciesfile.org/HomePage/Software/SoftwareHomePage.aspx">Species File Group</a> at Champaign/Urbana. Concerning Global
Names it was a very productive meeting. It was great to understand the current
state of Catalogue of Life, to see that CoL is not loosing momentum inspite of
financial problems of biodiversity informatics in general. There was
definitely interest in creating more bridges between various projects.</p>

<p>Yuri Roskov did present a ‘pilot’ project of cooperation between Encyclopeia of
Life species pages group and CoL. Data about <a href="http://www.ntnu.no/ub/scorpion-files/">~2000 species of
scorpions</a> had been harvested from html-based site to be used in
both projects. I think it was a great exercise and I do hope it will be just
the first example of such cooperation.</p>

<p>From the point of Global Names there were good news too. I think it was
everybody’s feeling that Global Names resolution is an important complementary
service for Catalogue of Life. Cooperation between various biodiversity
projects was brought up again and again, and organizing biodiversity
infrastructure as a mix of several projects where GBIF, EOL, CoL, GN etc
work as modules of a bigger puzzle, complement and enhance each other.</p>

<p>One thing that was brought up is lack of nomenclatural component in GN. I
talked about our plans to integrate <a href="http://www.gbif.org/dataset/34a96ebe-e51c-4222-9d08-5c2043c39dec">GN Usage Bank</a> and
<a href="http://resolver.globalnames.org/">GN Resolver</a> and demonstrate the flow of nomenclatural data into
resolution/reconciliation process. We will try to make such connection by
November and demonstrate the workflow on upcoming GBIF/CoL workshop.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/08/12/gn-resolver-vernaculars/">
      Global Names Resolver now shows vernacular names
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      resolver
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      12 Aug 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>If you use <a href="http://resolver.globalnames.org">Global Names Resolver</a> and you are interested in
vernacular names associated with scientific names you can add
<a href="http://resolver.globalnames.org/name_resolvers.json?names=Bubo%20bubo&amp;with_vernaculars=true">with_vernaculars=true</a> flag to see them.  If you are most
interested in particular <a href="http://resolver.globalnames.org/data_sources">data sources</a> you can add
<a href="http://resolver.globalnames.org/name_resolvers.json?names=Bubo%20bubo&amp;with_vernaculars=true&amp;data_source_ids=11%7C4">data_source_ids</a> setting.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/06/01/biodiversity-3-2-1/">
      Global Names Parser v3.2.1
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      parser, uuid, and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      01 Jun 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <h1 id="new-version-of-scientific-name-parser-is-out">New version of <a href="https://github.com/GlobalNamesArchitecture/biodiversity">Scientific Name Parser</a> is <a href="https://rubygems.org/gems/biodiversity">out</a>
</h1>

<p>This release has some backward compatibility issues with output.</p>

<h2 id="field-verbatim-is-not-preprocessed-in-any-way">Field “verbatim” is not preprocessed in any way</h2>

<p>In previous versions we did strip empty spaces and new line characters around
the name to generate “verbatim” field. Now name stays the way it was entered
into the parser.</p>

<p>Old behavior:</p>

<p>“Homo sapiens    “ -&gt; …“verbatim”: “Homo sapiens”</p>

<p>“Homo sapiens\r\n” -&gt; …“verbatim”: “Homo sapiens”</p>

<p>New behavior:</p>

<p>“Homo sapiens    “ -&gt; …“verbatim”: “Homo sapiens    “</p>

<p>“Homo sapiens\r\n” -&gt; …“verbatim”: “Homo sapiens\r\n”</p>

<h2 id="global-names-uuid-v5-is-added-to-the-output-as-id-field">
<a href="http://globalnamesarchitecture.github.io/gna/uuid/2015/05/31/gn-uuid-0-5-0.html">Global Names UUID v5</a> is added to the output as “id” field</h2>

<figure class="highlight"><pre><code class="language-json" data-lang="json"><span class="p">{</span><span class="w">
    </span><span class="nl">"scientificName"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
        </span><span class="nl">"id"</span><span class="p">:</span><span class="w"> </span><span class="s2">"16f235a0-e4a3-529c-9b83-bd15fe722110"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"parsed"</span><span class="p">:</span><span class="w"> </span><span class="kc">true</span><span class="p">,</span><span class="w">
        </span><span class="nl">"parser_version"</span><span class="p">:</span><span class="w"> </span><span class="s2">"3.2.1"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"verbatim"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Homo sapiens"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"normalized"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Homo sapiens"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"canonical"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Homo sapiens"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"hybrid"</span><span class="p">:</span><span class="w"> </span><span class="kc">false</span><span class="p">,</span><span class="w">
        </span><span class="nl">"details"</span><span class="p">:</span><span class="w"> </span><span class="p">[{</span><span class="w">
            </span><span class="nl">"genus"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
                </span><span class="nl">"string"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Homo"</span><span class="w">
            </span><span class="p">},</span><span class="w">
            </span><span class="nl">"species"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
                </span><span class="nl">"string"</span><span class="p">:</span><span class="w"> </span><span class="s2">"sapiens"</span><span class="w">
            </span><span class="p">}</span><span class="w">
        </span><span class="p">}],</span><span class="w">
        </span><span class="nl">"parser_run"</span><span class="p">:</span><span class="w"> </span><span class="mi">1</span><span class="p">,</span><span class="w">
        </span><span class="nl">"positions"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
            </span><span class="nl">"0"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">"genus"</span><span class="p">,</span><span class="w"> </span><span class="mi">4</span><span class="p">],</span><span class="w">
            </span><span class="nl">"5"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">"species"</span><span class="p">,</span><span class="w"> </span><span class="mi">12</span><span class="p">]</span><span class="w">
        </span><span class="p">}</span><span class="w">
    </span><span class="p">}</span><span class="w">
</span><span class="p">}</span></code></pre></figure>

<p>Read more about UUID v5 in another <a href="http://globalnamesarchitecture.github.io/gna/uuid/2015/05/31/gn-uuid-0-5-0.html">blog post</a></p>

<h2 id="names-with-underscores-instead-of-spaces-are-supported">Names with underscores instead of spaces are supported</h2>

<p>Such names are often used in representations of phyo-trees. Parser now
substitutes underscores to spaces during normalization phase</p>

<h2 id="normalized-canonical-forms-do-not-have-apostrophes-anymore">Normalized canonical forms do not have apostrophes anymore</h2>

<p>I am removing behavior introduced in v3.1.10 which would preserve apostrophes
in normalized version of names like “Arca m’coyi Tenison-Woods”. Apostrophes
are not code compliant.</p>

<p>New behavior:</p>

<figure class="highlight"><pre><code class="language-json" data-lang="json"><span class="p">{</span><span class="w">
    </span><span class="nl">"scientificName"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
        </span><span class="nl">"id"</span><span class="p">:</span><span class="w"> </span><span class="s2">"b3a9b1a3-f73c-5333-8194-a84c6583d130"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"parsed"</span><span class="p">:</span><span class="w"> </span><span class="kc">true</span><span class="p">,</span><span class="w">
        </span><span class="nl">"parser_version"</span><span class="p">:</span><span class="w"> </span><span class="s2">"3.2.1"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"verbatim"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Arca m'coyi Tenison-Woods"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"normalized"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Arca mcoyi Tenison-Woods"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"canonical"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Arca mcoyi"</span><span class="p">,</span><span class="w">
        </span><span class="nl">"hybrid"</span><span class="p">:</span><span class="w"> </span><span class="kc">false</span><span class="p">,</span><span class="w">
        </span><span class="nl">"details"</span><span class="p">:</span><span class="w"> </span><span class="p">[{</span><span class="w">
            </span><span class="nl">"genus"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
                </span><span class="nl">"string"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Arca"</span><span class="w">
            </span><span class="p">},</span><span class="w">
            </span><span class="nl">"species"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
                </span><span class="nl">"string"</span><span class="p">:</span><span class="w"> </span><span class="s2">"mcoyi"</span><span class="p">,</span><span class="w">
                </span><span class="nl">"authorship"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Tenison-Woods"</span><span class="p">,</span><span class="w">
                </span><span class="nl">"basionymAuthorTeam"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
                    </span><span class="nl">"authorTeam"</span><span class="p">:</span><span class="w"> </span><span class="s2">"Tenison-Woods"</span><span class="p">,</span><span class="w">
                    </span><span class="nl">"author"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">"Tenison-Woods"</span><span class="p">]</span><span class="w">
                </span><span class="p">}</span><span class="w">
            </span><span class="p">}</span><span class="w">
        </span><span class="p">}],</span><span class="w">
        </span><span class="nl">"parser_run"</span><span class="p">:</span><span class="w"> </span><span class="mi">1</span><span class="p">,</span><span class="w">
        </span><span class="nl">"positions"</span><span class="p">:</span><span class="w"> </span><span class="p">{</span><span class="w">
            </span><span class="nl">"0"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">"genus"</span><span class="p">,</span><span class="w"> </span><span class="mi">4</span><span class="p">],</span><span class="w">
            </span><span class="nl">"5"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">"species"</span><span class="p">,</span><span class="w"> </span><span class="mi">11</span><span class="p">],</span><span class="w">
            </span><span class="nl">"12"</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">"author_word"</span><span class="p">,</span><span class="w"> </span><span class="mi">25</span><span class="p">]</span><span class="w">
        </span><span class="p">}</span><span class="w">
    </span><span class="p">}</span><span class="w">
</span><span class="p">}</span></code></pre></figure>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/31/gn-uuid-0-5-0/">
      New UUID v5 Generation Tool -- gn_uuid v0.5.0
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      uuid and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      31 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>We are releasing a new tool - <a href="https://github.com/GlobalNamesArchitecture/gn_uuid">gn_uuid</a> to simplify creation of UUID
version 5 identifiers for scientific name strings.  UUID v5 has features which
are particular useful for the biodiversity community.</p>

<h2 id="uuid-version-5-description">UUID version 5: Description</h2>

<p><a href="http://en.wikipedia.org/wiki/Universally_unique_identifier">Universally unique identifiers</a> are very popular because for all
practical purposes they <a href="http://tools.ietf.org/html/rfc4122#section-2_">guarantee globally unique IDs</a> without any
negotiation between different entities. There are several ways how UUIDs can be
created:</p>

<table>
  <thead>
    <tr>
      <th>UUID version</th>
      <th>Uniqueness is achieved by</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>version 1</td>
      <td>Using computer’s MAC address and time</td>
    </tr>
    <tr>
      <td>version 2</td>
      <td>like v1 plus adding info about user and local domain</td>
    </tr>
    <tr>
      <td>version 3</td>
      <td>Using MD5 hash of a string in combination with a name space</td>
    </tr>
    <tr>
      <td>version 4</td>
      <td>Using pseudo-random algorithms</td>
    </tr>
    <tr>
      <td>version 5</td>
      <td>Using SHA1 hash of a string in combination with a name space</td>
    </tr>
  </tbody>
</table>

<p>UUID v5 is generated using information from a string, so everyone who uses this
method will generate exactly same ID out of the same string. Interested parties
do need to agree on generation of a name space, and after that no matter which
programming language they use – they will be able to exchange data about a
string using their identifiers.</p>

<p>This gem already has a DNS domain “globalnames.org” defined as a name space, so
generation of the UUID v5 becomes simpler.</p>

<p>I believe UUID v5 creates very exciting opportunities for biodiversity
community.  For example if one expert annotates a string or attaches data to it
– this information can be linked globally and then harvested by anybody,
without any preliminary negotiation.</p>

<p>Quite often researches make an argument that a scientific name is an identifier
on its own and there is no need for another level of indirection like UUID.
They are right, scientific name string can be an identifier, however,
scientific names have severe shortcomings in such a role.</p>

<h2 id="why-scientific-names-are-bad-identifiers-for-computers">Why Scientific Names are Bad Identifiers for Computers</h2>

<h3 id="scientific-name-strings-have-different-length">Scientific name strings have different length</h3>

<p>More often than not identifiers end up in databases and used as a primary index
to sort, connect and search data. Scientific name strings vary from 2 bytes to
more than  500 bytes in length. So if they are used as keys in database they
are inefficient, they waste a lot of space, become less efficient for finding
or sorting information – indexes key size is usually determined by the
the largest key.</p>

<p>UUIDs have always the same, rather small size – 16 bytes.  Even when UUIDs are
used in their “standard” string representation – they are still reasonably
small – 36 characters. Storing them in a database as a number is obviously more
efficient.</p>

<h3 id="it-is-hard-to-spot-differences-in-name-strings">It is hard to spot differences in name strings</h3>

<p>It is very hard for human eye to spot the difference between strings like this</p>

<ul>
  <li>
    <p>Corchoropsis tomentosa var. psilocarpa (Harms &amp; Loes.) C.Y.Wu &amp; Y.Tang</p>
  </li>
  <li>
    <p>Corchoropsis tomentosa var. psilocanpa (Harms &amp; Loes.) C.Y.Wu &amp; Y.Tang</p>
  </li>
</ul>

<p>Much easier for their corresponding UUIDs</p>

<ul>
  <li>
    <p>5edecb2b-903f-54f1-a087-b47b3b021fcd</p>
  </li>
  <li>
    <p>833c664b-7d00-5c3b-97a4-98b0ab7d0a9a</p>
  </li>
</ul>

<h3 id="name-strings-come-in-different-encodings">Name strings come in different encodings.</h3>

<p>Currently Latin1, UTF-8 and UTF-16 are most popular encodings used in
biodiversity. If authorship or name itself has characters outside of the
128bits of ASCII code – identically looking names will be quite different for
computers.</p>

<h3 id="name-strings-are-less-stable-because-of-their-encoding">Name strings are less stable because of their encoding</h3>

<p>When names are moved from one database to another, from one paper to another
sometimes they do not survive the trip. If you spent any time looking at
scientific names in electronic form you did see something like this:</p>

<ul>
  <li>
    <p>Acacia ampliceps ? Acacia bivenosa</p>
  </li>
  <li>
    <p>Absidia macrospora V�nov� 1968</p>
  </li>
  <li>
    <p>Absidia sphaerosporangioides Man&lt;acute&gt;ka &amp; Truszk., 1958</p>
  </li>
  <li>
    <p>Cnemisus kaszabi Endr?di 1964</p>
  </li>
</ul>

<p>Usually names like these had been submitted in a “wrong” encoding and some
characters in them were misinterpreted. UUID on the other hand is just a
hexadecimal number, which can be transitioned between various encodings more
safely.</p>

<h3 id="name-strings-might-look-the-same-in-print-or-on-screen-but-be-different">Name strings might look the same in print or on screen, but be different</h3>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">Homo sapiens</code></p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">Homo sаpiens</code></p>
  </li>
</ul>

<p>These two strings might look exactly the same on a screen or printed on paper,
but in reality they are different. Here are their UUIDs:</p>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">16f235a0-e4a3-529c-9b83-bd15fe722110</code></p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">093dc7f7-5915-56a5-87de-033e20310b14</code></p>
  </li>
</ul>

<p>The difference is that the second name has a Cyrillic <code class="language-plaintext highlighter-rouge">а</code> character, which in
most cases will look exactly the same as Latin <code class="language-plaintext highlighter-rouge">a</code> character. And when the
names are printed on paper there is absolutely no way to tell the difference.
UUID will tell us that these two name strings are not the same.</p>

<h3 id="nothing-prevents-to-continue-to-use-name-strings-for-human-interaction">Nothing prevents to continue to use name strings for human interaction</h3>

<p>One argument that people often give – it is much easier for users to type</p>

<p>http://biosite.org/Parus_major</p>

<p>than</p>

<p>http://biosite.org/47d61c81-5a0f-5448-964a-34bbfb54ce8b</p>

<p>For most of us it is definitely true and nothing prevents developers
to create links of the first type, while still using UUIDs behind the scene.</p>

<h2 id="why-uuids-v5-are-better-than-any-other-uuids-as-a-scientific-name-identifier">Why UUIDs v5 are better than any other UUIDs as a scientific name identifier</h2>

<ul>
  <li>
    <p>They can be generated independently by anybody and still be the same to the
same name string</p>
  </li>
  <li>
    <p>They use SHA1 algorithm which does not have (extremely rare) collisions
found for MD5 algorithm</p>
  </li>
  <li>
    <p>Same ID can be generated in <a href="https://github.com/GlobalNamesArchitecture/gn_uuid_examples">any popular language</a> following
<a href="http://tools.ietf.org/html/rfc4122#section-4.3">well-defined algorithm</a></p>
  </li>
</ul>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/31/gn-crossmap-0-1-6/">
      Crossmap Tool v0.1.6
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      crossmap and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      31 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Version 0.1.6 of <code class="language-plaintext highlighter-rouge">gn_crossmap</code> tool is out.</p>

<p><a href="https://github.com/GlobalNamesArchitecture/gn_crossmap">Global Names Crossmap</a> tool allows to match names from a a
spreadsheet to names from <a href="http://resolver.globalnames.org/data_sources">any data source</a> in Global Names
Resolver.</p>

<p>Fixes a bug in rank inferring by <a href="https://github.com/GlobalNamesArchitecture/biodiversity">Global Names Parser</a>.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/28/gn-crossmap-0-1-5/">
      Crossmap Tool v0.1.5
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      crossmap and release
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      28 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>New version of gn_crossmap tool is out.</p>

<p><a href="https://github.com/GlobalNamesArchitecture/gn_crossmap">Global Names Crossmap</a> tool allows to match names from a a
spreadsheet to names from <a href="http://resolver.globalnames.org/data_sources">any data source</a> in Global Names
Resolver.</p>

<p>The main change in this version – output file with crossmap data now contains
all fields from original input document and it allows to filter and sort data
using any field from the input.</p>

<p>Other changes are</p>

<ul>
  <li>
    <p><a href="https://github.com/dimus" class="user-mention">@dimus</a> - #5 - All original fields are now preserved in the output file.</p>
  </li>
  <li>
    <p><a href="https://github.com/dimus" class="user-mention">@dimus</a> - #3 - If ingest has more than 10K rows – user will see logging events</p>
  </li>
  <li>
    <p><a href="https://github.com/dimus" class="user-mention">@dimus</a> - #4 Bug - Add error messages if headers don’t have necessary fields</p>
  </li>
  <li>
    <p><a href="https://github.com/dimus" class="user-mention">@dimus</a> - #2 - Header fields are now allowed to have trailing spaces</p>
  </li>
  <li>
    <p><a href="https://github.com/dimus" class="user-mention">@dimus</a> - #7 Bug - Empty rank does not break crossmapping anymore</p>
  </li>
  <li>
    <p><a href="https://github.com/dimus" class="user-mention">@dimus</a> - #1 Bug - Add missing rest-client gem</p>
  </li>
</ul>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/20/idigbio-client/">
      iDigBio API Client v0.1.1
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      idigbio and api
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      20 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>In a few weeks there will be an <a href="https://github.com/idigbio-api-hackathon/HackathonCentral/wiki">iDigBio API hackathon</a>. As I
menioned <a href="https://globalnamesarchitecture.github.io/idigbio/hakathon/2015/04/28/idigbio-api-hackathon-meet.html">earlier</a> we decided to add another API client written in
Ruby before the hackathon starts. And <a href="https://github.com/gete76">Greg Traub</a> and I are releasing
<a href="https://github.com/GlobalNamesArchitecture/idigbio_client">iDigBio API Client</a> written in Ruby today.</p>

<p>Greg started to make a Ruby client at <a href="https://github.com/iDigBio/idigbio-ruby-client">iDigBio</a>. I took his code and
refactored it into a <a href="https://github.com/GlobalNamesArchitecture/idigbio_client">Ruby gem</a>. So now instead of 0 we have 2 Ruby
clients <img class="emoji" title=":smile:" alt=":smile:" src="https://github.githubassets.com/images/icons/emoji/unicode/1f604.png" height="20" width="20"></p>

<p>This is the very first release, so if you will start using it and find
something is wrong/missing please submit an <a href="https://github.com/GlobalNamesArchitecture/idigbio_client/issues">issue</a>. The gem uses <a href="https://beta-search.idigbio.org/v2/">beta
API</a>, so sometimes it might get ‘stuck’. This problem will go away when
beta API will move to production.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/14/parser-3-1-10/">
      Scientific Name Parser v3.1.10
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      parser
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      14 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <h1 id="new-version-of-scientific-name-parser-is-out">New version of <a href="https://github.com/GlobalNamesArchitecture/biodiversity">Scientific Name Parser</a> is <a href="https://rubygems.org/gems/biodiversity">out</a>
</h1>

<p>Addressing most of <a href="https://github.com/GlobalNamesArchitecture/biodiversity/issues/7">Issue #7</a></p>

<h3 id="do-not-parse-non-virus-names-containing-rna">Do not parse non-virus names containing RNA</h3>

<p>If a name was not detected as a virus but contains RNA word it will not be
parsed anymore. It is a problem for some surrogate names, like
<strong>Candida albicans RNA_CTR0-3</strong> but they are very rare.</p>

<table>
  <thead>
    <tr>
      <th>Name</th>
      <th>Action</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Candida albicans RNA_CTR0-3</td>
      <td>Not parsed</td>
    </tr>
    <tr>
      <td>Alpha proteobacterium RNA12</td>
      <td>Not parsed</td>
    </tr>
    <tr>
      <td>Ustilaginoidea virens RNA virus</td>
      <td>Not parsed, marked as virus</td>
    </tr>
    <tr>
      <td>Calathus (Lindrothius) KURNAKOV 1961</td>
      <td>Parsed as before</td>
    </tr>
  </tbody>
</table>

<h3 id="better-detection-of-virus-names">Better detection of virus names</h3>

<p>Names containing <code class="language-plaintext highlighter-rouge">virophage</code> <code class="language-plaintext highlighter-rouge">*NPV</code>, <code class="language-plaintext highlighter-rouge">*sattelite</code>, <code class="language-plaintext highlighter-rouge">*particle</code> are marked as ‘viruses’ and
not parsed</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Gossypium mustilinum symptomless alphasatellite
Okra leaf curl Mali alphasatellites-Cameroon
Bemisia betasatellite LW-2014
Tomato leaf curl Bangladesh betasatellites [India/Patna/Chilli/2008]
Intracisternal A-particles
Saccharomyces cerevisiae killer particle M1
Uranotaenia sapphirina NPV
Spodoptera exigua nuclear polyhedrosis virus SeMNPV
Spodoptera frugiperda MNPV
Rachiplusia ou MNPV (strain R1)
Orgyia pseudotsugata nuclear polyhedrosis virus OpMNPV
Mamestra configurata NPV-A
Helicoverpa armigera SNPV NNg1
Zamilon virophage
Sputnik virophage 3
</code></pre></div></div>

<h3 id="better-handling-of-speciesinfraspecies-epithets-with-apostrophe">Better handling of species/infraspecies epithets with apostrophe</h3>

<p>Names like below are now parsed correctly. Their normalized/canonical forms
preserve apostrophe</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Junellia o'donelli Moldenke, 1946
Trophon d'orbignyi Carcelles, 1946
Arca m'coyi Tenison-Woods, 1878
Nucula m'andrewii Hanley, 1860
Eristalis l'herminierii Macquart
Odynerus o'neili Cameron
Serjania meridionalis Cambess. var. o'donelli F.A. Barkley
</code></pre></div></div>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/12/gn-moves-to-illinois/">
      Global Names Grant Goes to Illinois
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      speciesfilegroup
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      12 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>I did some soul-searching, advise-gathering, thinking, planning, and crystal
ball gazing.  And it seems that moving <a href="http://www.nsf.gov/awardsearch/showAward?AWD_ID=1356347">Global Names grant</a> and
myself to <a href="http://software.speciesfile.org/HomePage/Software/SoftwareHomePage.aspx">Species File Group</a> is a right decision.  Why? Because <a href="http://mbl.edu">Marine
Biological Laboratory</a> is a hard core research institute, which completely
depends on grants and as such is not well-suited for <em>infrastructure</em>
projects. Global Names is definitely an infrastructure project and I know very
well how bad it is to be responsible for an infrastructure project and not
being able to work on it. It is just not a good way to do business.</p>

<p>It is my 8th year at MBL. I enjoy MBL, I love living on Cape Cod. I love an
immense energy of MBL collective mind.  I met really amazing people, amazing
scientists here. I worked with great people at <a href="http://eol.org">Encyclopedia of Life</a>
project.  And also I was never sure if I will be there next year, or sometimes
next month.  I had weeks and months when I had no ability to move forward with
Global Names, because it had no financial support at that time.</p>

<p>Species Files Group is long term financed, allowing a long term commitment.
They are interested in Global Names, they do want me to continue to develop it,
integrate it with Catalogue of Life. And these are my goals too. David Eades
understands that Global Names will need a long-term investment in hardware, and
he provides a generous annual fund for that. It means no more 7 year old
computers running Global Names services. I also hope it will help to integrate
Global Names Usage Bank, a crucial GN component developed by
<a href="https://www.ted.com/talks/richard_pyle_dives_the_twilight_zone?language=en">Rich Pyle</a> and <a href="https://github.com/whittonr">Rob Whitton</a>.</p>

<p>Another big factor is ability to work closely with programmers and taxonomists
of the SFG group. At MBL I am now the only one on EOL project (<a href="https://github.com/jrice">Jeremy</a> is
remote), and I feel I am getting stale without nomenclators/taxonomists around.</p>

<p>Of course we need to figure out how to move current GN computers without
shutting down services for a few weeks. I imagine I would have to rent an
expensive cloud setup for a month or two, and run GN from there while machines
are in transit. We will have to figure out how to transfer grant, make a new
hire for the project etc. But all of these are good problems to solve. I
believe GN suddenly got a brighter future ahead.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/11/gn-crossmap-gem/">
      New tool to crossmap checklists
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      resolver and checklist
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      11 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Yesterday I released a new command line tool for name resolution called
<a href="https://github.com/GlobalNamesArchitecture/gn_crossmap">gn_crossmap</a>. It is designed for people who work with checklists
of scientific names using a spreadsheet software (MS Excel, Apple Numbers, Open
Office, Libre Office, Google Sheets etc.) and want to compare names that they
have with another reference source. The program takes a spreadsheet saved as
csv file as input and generates another csv-based spreadsheet with resolution
data.  <a href="https://github.com/GlobalNamesArchitecture/gn_crossmap/tree/master/spec/files">Examples of input and output</a> are included into the code.
<a href="https://github.com/GlobalNamesArchitecture/gn_crossmap/blob/master/README.md">README</a> file describes how to use the project from a command line or
as a Ruby library.</p>

<p>This program requires internet connection, Ruby &gt;= 2.1 installed on the machine.</p>

<p>Basic usage is:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ gem install gn_crossmap
$ crossmap -i input.csv -o ouput.csv -d 1
</code></pre></div></div>

<p>where</p>

<hr>

<table>
  <thead>
    <tr>
      <th>short</th>
      <th>long attr</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>-i</td>
      <td>–input</td>
      <td>checklist’s spreadsheet saved as csv file</td>
    </tr>
    <tr>
      <td>-o</td>
      <td>–output</td>
      <td>path to the output file. Default is output.csv in the current directory</td>
    </tr>
    <tr>
      <td>-d</td>
      <td>–data-source-id</td>
      <td>ID of one of the GN Resolver <a href="http://resolver.globalnames.org/data_sources">data sources</a>, Catalogue of Life id (1) is default</td>
    </tr>
  </tbody>
</table>

<hr>

<p>Web interface to this program is also <a href="https://github.com/GlobalNamesArchitecture/checklist">in works</a></p>

<p>This project started at the <a href="https://drive.google.com/file/d/0B-WDk-H2QjRKaWJUUkpMSEdta1k/view">Catalogue of Life workshop</a> in Leiden,
which happened in March 2015. The main focus of the hackathon was to figure out
how to help national checklist teams to create, maintain and compare data in their
data.  We determined 3 main approaches</p>

<ol>
  <li>
<strong>Crossmapping</strong> checklists against other checklists and/or reference sources</li>
  <li>
<strong>Annotation</strong> of crossmapped data – ability to share metadata, report mistakes</li>
  <li>
<strong>Distribution of species</strong> – how to fix occurance errors for a country</li>
</ol>

<p>A hackathon group which worked on crossmapping produced a
<a href="https://github.com/Sp2000/hackathon_group1">code</a> which would compare checklists against Catalogue of
Life. The <code class="language-plaintext highlighter-rouge">gn_crossmap</code> program I am releasing is based heavily on what we
learned during the hackathon. Crossmaping code is mostly based on use cases
from <a href="https://github.com/rpfigueira">Rui Figueira</a> and <a href="https://github.com/WouterKoch">Wouter Koch</a>. During the hackathon we
also determined ways to improve quality of name resolution further by:</p>

<ul>
  <li>Using infraspecies’ rank (var., f. subsp. etc) in the matching and penalize
score if ranks are different</li>
  <li>Taking in account if matching authors are basyonym or combination authors</li>
  <li>Using meta-information attached to names via sensu…, not … etc.
to distinguish name usages</li>
</ul>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/06/phylo-trees/">
      Kickoff  Meeting for Disseminating Phylogenetic Knowledge Project
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      phylogeny, trees, and nsf
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      06 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Yesterday <a href="https://github.com/arlin">Arlin Stoltzfus</a> organized a kickoff meeting for the project
that got funded by NSF this year - “<a href="http://nsf.gov/awardsearch/showAward?AWD_ID=1458572">Collaborative Research: ABI Development: An
open infrastructure to disseminate phylogenetic knowledge</a>”. Global Names is
participating in project and I believe it will be an interesting ride.</p>

<p>The idea behind is pretty cool. Imagine that someone works on a group of
organisms.  They submit names of the organisms to a service and the service
builds a phylogenetic tree out of the names. When tree is created it will start
its own life similar to a repo on Github. People will be able to reuse it,
annotate it for their own purposes, create derivative trees. It would be a
pretty nice feature for <a href="http://eol.org">Encyclopedia of Life</a> to see how species
belonging to a particular clade are related to each other through phylogeny.
One problem with creation of such trees is name normalization. Scientific names
can have many alternative spellings, so to find phylo-information we will need
to be able to map names from user list to names which are recognized by the
service.</p>

<p>I suspect that a crossmapping tool I am working on this week might be adjusted
for this particular task, but as usual – the devil is in details and we will
find out the requirements during the design process.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/03/higher-level-classification/">
      New Higher Level Classification for Catalogue of Life
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      col
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      03 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="https://about.me/bobcorrigan">Bob Corrigan</a> sent around an email pointing at a <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0119248">paper in PLOS</a>
which describes new classification adopted by <a href="http://www.catalogueoflife.org/">Catalogue of Life</a>. After
looking through the paper my understanding is that it is a step forward and at
the same time business as usual for <code class="language-plaintext highlighter-rouge">CoL</code>.</p>

<p><code class="language-plaintext highlighter-rouge">Catalogue of Life</code> needs a solid managerial classification for their data and
according to article the goal is achieved:</p>

<blockquote>
  <p>Our goal, therefore, is to provide a hierarchical classification for the CoL
and its contributors that (a) is ranked to encompass ordinal-level taxa to
facilitate a seamless import of contributing databases; (b) serves the needs
of the diverse public-domain user community, most of whom are familiar with
the Linnaean conceptual system of ordering taxon relationships; and (c) is
likely to be more or less stable for the next five years. Such a modern
comprehensive hierarchy did not previously exist at this level of
specificity.</p>
</blockquote>

<p>Classifications are a dirty business so as usual –</p>

<blockquote>
  <p>These actual complexities of phylogenetic history emphasize that
classification is a practical human enterprise where compromises must be made</p>
</blockquote>

<p>Altogether looks like CoL gets a new hierarchical face.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/03/sysopia-starts/">
      Starting GSOC 2015 Sysopia Project
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      sysopia and gsoc
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      03 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Our Google Summer of Code student – <a href="https://github.com/vpowerrc">Viduranga Wijesooriya</a> and I had
our firts meeting today to start <a href="https://www.google-melange.com/gsoc/project/details/google/gsoc2015/vpowerrc/5665117697998848">Google Summer of Code</a> project –
<a href="https://github.com/EOL/sysopia">Sysopia</a>. The purpose of the project is not names, however I do
consider it to be important for EOL and for GN as it allows to spend less time
on administration of computers and more time on writing code.</p>

<p>The idea behind the project is to create a dashboard that allows us to see what
is going on with all computers in a system with one glance. The system shows
several metrics graphs, each of which shows information about all machines at
the same time.  By default it shows data for 24 hours, so if everything works
well it is enough for sysadmin to check out sysopia once a day to have a very
good idea about what is happening with the system from the moment Sysopia is
installed. We did install it for EOL and I find it very useful.</p>

<p><a href="https://github.com/EOL/sysopia"><img src="/images/blog/sysopia.png" alt="sysopia"></a></p>

<p>Not much functionality is there yet, but graphs show well, and it is possible
to get a point data by hovering over a line, and highlight a particular machine
when hovering over the machine name in the dialog box.</p>

<p>Currently the only backend for sysopia is <a href="http://sensuapp.org/">Sensu</a> but we are going to
expand it to other backends after we nail down the user interface.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/05/02/species-file-group/">
      Species File Group -- new home for GN?
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      col, speciesfile, and taxonworks
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      02 May 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Trying to find a permanent home for GN I travelled to Champaign-Urbana on
Thursday-Friday to visit <a href="http://software.speciesfile.org/HomePage/Software/SoftwareHomePage.aspx">Species File Group</a> at the University of
Illinois. I do know this group rather well, as <a href="https://github.com/lwalley">Lisa Walley</a> and I went
there more than a year ago for a hackathon organized by <a href="https://github.com/mjy">Matt Yoder</a>.</p>

<p>I am quite impressed with work this group does and when Matt suggested
me to join them – my first thought was – this might be a way to make
Global Names financial situation more reliable!</p>

<p>Currently Global Names completely depends on grants, and grants come and go. It
is a pretty bad way to finance an infrastructure project, as you do not want to
have roads or electricity depend on unstable funding. We always want to be able
to drive to a store or a concert, and we want to always be able to have lights
in our homes.  Same goes with projects like GN. If people start using them –
they start depend on them and it is really bad situation when funding dries out
and service deteriorates as a result.</p>

<p>The visit went well. I had a great opportunity to talk to David Eades, Matt,
Dmitry Dmitriev, and Yuri Roscov. It was especially great to talk to Yuri, as
he is the main person behind the <a href="http://www.catalogueoflife.org/">Catalogue of Life</a> content. I consider
Catalogue of Life to be one of the most important use cases and partners for
GN, and talking to Yuri for an extended amount of time was extremely
helpful.</p>

<p>Originally position was about helping Yuri to automate his work-flow, however
when Matt and I talked on Skype the accent started to shift towards supporting
Global Names. David Eades, Matt, and Yuri all believe that GN is a missing link
in Catalogue of Life functionality and as such by working closely with Yuri,
and figuring out what GN can do for Catalogue of Life actually does help to
automate some of hard parts of Yuri’s work.</p>

<p>The meeting was very encouraging and inspiring. Now I have to think hard and
make a decision.  I would love be able to keep my house on Cape and I would
love to be able to come in summer and work with <a href="http://mbl.edu">MBL</a> and <a href="http://www.risd.edu/">RISD</a>.
And it seems nothing prevents me to spend 3 months on Cape in summer if I move.
On Monady I am going to talk about my trip at my work at MBL.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/04/29/good-bad-names/">
      Good names, bad names?
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gni, resolver, and globi
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      29 Apr 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Just had a great conversation with <a href="https://github.com/jhpoelen">Jorrit Poelen</a> from <a href="http://www.globalbioticinteractions.org/about.html">GloBI</a>.
Jorrit uses <a href="http://resolver.globalnames.org">GN Resolver</a> to clean up names for Globi, and on top of
knowing that name exists in GN Resolver he also needs to know if the name is
actually valid.</p>

<p>Resolver by design contains ‘good’ and ‘bad’ names. We do need to know what
kind of misspellings exist in the wild and map information associated with them
to good names. These misspellings and outright wrong names make Jorrit’s
life much harder, as we do not have a tool that clearly marks ‘good’ names as
good.  There are ways to be more or less sure if a name is good:</p>

<ul>
  <li>If names belong to a specific clade – use only highly curated sources</li>
  <li>Count how many sources know about a name</li>
  <li>See if a name appears at least in one curated source</li>
  <li>Check if name got parsed</li>
</ul>

<p>But all these approaches are not universal, and do not give a clear answer.  So
what would be a solution?</p>

<p>It seems that a good solution would be to write a classifier which takes in
account all relevant features and meta-features of a name, considers them and
then puts the name into ‘good’ or ‘bad’ bucket.  Every name has several
features associated with it and we can train a Bayes classifier to make a
decision if name is ‘good’ or ‘bad’ using these features. When it is done
running through our ~20 million names – each of them will be marked as trusted
or not.</p>

<p>I am pretty much sure that such classifier, especially at its first iteration
will make mistakes. How can we deal with them? Here is an idea – when API
returns back data back to a user – data will have two new fields – ‘trusted’
as yes/no and a URL to complain about this decision something like:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>http:/resolver.globalnames.org/trusted?name_id=123&amp;wrong_value=1
</code></pre></div></div>

<p>People can just copy and paste this URL into browser, or set it as a “Report a
mistake” button for every name in their results html.  If this button is pushed
GN Resolver will register human curation event and data from this event will be
used to improve performance of the classifier algorithm. Human curations will
trump computer algorithm and they can be collected in a new data source for
feedbacks…</p>

<p>Details of the interface can be decided later when we build the classifier. I
know that the problem of separating trusted names from untrusted is a task that
about everybody who uses resolver actively asked me about one time or another.
So who and when can build it? And now I am thinking that our Google Summer of
Code student might be interested in making it happen instead of improving
NetiNeti.  I personally think automatic curation of names is more important.</p>

<p>Jorrit submitted an <a href="https://github.com/GlobalNamesArchitecture/gni/issues/38">issue</a> about this idea
Globit <a href="https://github.com/jhpoelen/eol-globi-data/issues/132">issue</a></p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/04/28/idigbio-api-hackathon-meet/">
      iDigBio hackathon preparation
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      idigbio and hakathon
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      28 Apr 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>On June 3rd I am going to <a href="https://github.com/idigbio-api-hackathon/HackathonCentral/wiki">iDigBio hackathon</a> meeting which will be about finding
ways to enhance their API. Today there was a pre-hackathon meeting where
iDigBio folks explained how did they implement their API, it’s backend, and
how do they use their own API for their GUIs.</p>

<p>I had been very impress with what they have done. Backend is based on Elastic
search, API is RESTful and json based. What was a surprise for me – the API
calls are often take pure json as arguments. It was also great to see how did
they simplified Elastic search queries for API, keeping their API queries
simple and powerful at the same time.</p>

<p>They also made Python and R clients for the API. So I will try to make Ruby
version of the API client before the hackathon.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/04/28/gsoc-netineti/">
      Launching NetiNeti Google Summer of Code Project
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      netineti and gsoc
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      28 Apr 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Today we had our first meeting to start NetiNeti enhancement project funded by
Google Summer of Code. The studend who was selected do do the job is a 4th
year graduate from University of Philadelphia <a href="https://github.com/wencanluo">Wencan Luo</a>.</p>

<p>The purpose of the project is to improve performance of our NLP-based
scientific name finding tool – NetiNeti, developed 5 years ago by [Lakshmi
Manohar Acella][lakshmi]. Lets see how it will go…</p>

<p>Official coding time starts on May 27th – for now we are going through a design
phase – figuring out who are the users of the application, then we will try
to do idealized design of its features, find implementaion paths and limitations
existing to implement the features and then Wencan is going to do exploration
of features.</p>

<p>For the process we are going to use <code class="language-plaintext highlighter-rouge">ZenHub</code> to manage issues, obviosusly
<code class="language-plaintext highlighter-rouge">GitHub</code> for the code and ability to have a project related blog with Github
and <code class="language-plaintext highlighter-rouge">Jekyll</code>.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/04/27/gsoc-2015-starts/">
      Google Summer of Code 2015 starts
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      eol and gsoc
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      27 Apr 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Today is the official start of Google Summer of Code work. Up to now
organizations submitted their ideas, organizations had been chosen by Goggle,
students decided which ideas and organizations they like and submitted their
proposals, Google decided on how many projects from every organizations they
are willing to fund, and finally – the best proposals from students were
matched to the funded ideas. Encyclopedia of Life and Global Names submitted 4
proposals and we got 3 of them funded, so congratulations to us and to students
:)</p>

<h3 id="enhancements-in-eol-semantic-search"><a href="https://www.google-melange.com/gsoc/project/details/google/gsoc2015/avinashda/5707702298738688">Enhancements in EOL Semantic Search</a></h3>

<table>
  <tbody>
    <tr>
      <td>Student:</td>
      <td>Avinash Daiict</td>
    </tr>
    <tr>
      <td>Mentor:</td>
      <td><a href="https://github.com/AmrMMorad">Amr Morad</a></td>
    </tr>
  </tbody>
</table>

<h3 id="devops-dashboard-sysopia"><a href="https://www.google-melange.com/gsoc/project/details/google/gsoc2015/vpowerrc/5665117697998848">DevOps Dashboard Sysopia</a></h3>

<table>
  <tbody>
    <tr>
      <td>Student:</td>
      <td><a href="https://github.com/vpowerrc">Viduranga Wijesooriya</a></td>
    </tr>
    <tr>
      <td>Mentor:</td>
      <td><a href="https://github.com/dimus">dimus</a></td>
    </tr>
  </tbody>
</table>

<h3 id="finding-scientific-names"><a href="https://www.google-melange.com/gsoc/project/details/google/gsoc2015/wcluo/5661458385862656">Finding Scientific Names</a></h3>

<table>
  <tbody>
    <tr>
      <td>Student:</td>
      <td><a href="https://github.com/wencanluo">Wencan Luo</a></td>
    </tr>
    <tr>
      <td>Mentor:</td>
      <td><a href="https://github.com/dimus">dimus</a></td>
    </tr>
  </tbody>
</table>

<hr>

<p>I had been very happy to see how many people were interested in the idea of
finding scientific names in texts – we had 12 proposals, so competition was
fierce this year! I think we got great students and I am looking forward
to the Google Summer of Code 2015.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/04/22/thank_you_eol/">
      Gracious gift from Encylopedia of Life
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      eol
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      22 Apr 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p><a href="http://eol.org">Encyclopedia of Life</a> folks made a truly glorious present to Global Names
consisting of several Dell 710 servers which were used for running EOL at
Harvard. Now, when site moved to Smithsoninan EOL donates some of these
computers to GN, and 10 of them are already at <a href="http://mbl.edu">Marine Biological
Laboratory</a>, waiting to be plugged to internet and electricity. Another
truly amazing gift from EOL is 14 100gb hard drives which will run GN
databases.</p>

<p>I feel happy warm and fuzzy – thank you EOL! I hope with this new
hardware I will be able to increase GN capacity about 5x using current code!
Next step – installing Chef, Docker, and GN applications to serve the
biodiversity community.</p>


  </div>
</article>


  <article>
  <h2>
    <a href="/news/2015/04/17/gna_is_back/">
      GNA work continues
    </a>
  </h2>
  <span class="post-category">
    <span class="label">
      gna and eol
    </span>
  </span>
  <div class="post-meta">
    <span class="post-date">
      17 Apr 2015
    </span>
    <a href="https://github.com/dimus" class="post-author">
      <img src="https://github.com/dimus.png" class="avatar" alt="dimus avatar" width="24" height="24">
      dimus
    </a>
  </div>
  <div class="post-content">
    <p>Last year we did get the second round of funding for Global Names Architecture
development. Our <a href="http://nsf.gov/awardsearch/showAward?AWD_ID=1062387">first grant</a> had been about exploration of how to find
scientific names in texts, how to crossmap different spelling variants of the
same names to each other, how to connect names to literature collected at
<a href="http://www.biodiversitylibrary.org/">Biodiversity Heritage Library</a>, how to organize scientific name usages,
how to register new zoological scientific names electronically. Several
intersting projects spanned out of this effort and you can read about them
at <a href="http://globalnames.org">Global Names site</a>.</p>

<p>It was a hard year for <a href="http://eol.org">Encyclopedia of Life</a> where I work, and for Global
Names. I did have to spend most of the first 8 months since our second NSF
grant got funding helping EOL with system administration support and transfer
EOL site from Harvard to Smithsonian Museum of Natural History. It is done now,
and I am happy to be able to work on Global Names project again!</p>

<p>What kind of resources do we have now? 2 months of Paddy (David Patterson)
time, 2 months of Rich Pyle time, About a 1.5 years of mine,
and 1 year of another developer. We also got 2 excellent participants for
Google Summer of Code this year, so it is 6 months of their time as well.
And a quest for further funding continues as I write.</p>

<p>Encyclopedia of Life kindly donated a lot of <a href="http://globalnamesarchitecture.github.io/gna/eol/2015/04/22/thank_you_eol.html">hardware</a>, and
<a href="http://mbl.edu">Marine Biological Laboratory</a> provides us with a whole rack of space,
fast internet connection. So we are set for an exciting year ahead!</p>

<p>What are the plans? This grant covers work on name finding and name resolution.
We try to find major use-cases (Arctos, EOL, iDigBio, Catalogue of Life, GBIF)
and satisfy their needs. We expect it will cover needs of 90% of other users,
and the remaining 10% of functionality will trickle by means of going through
github issues, fixing bugs, adding features, thinking about new ideas.</p>


  </div>
</article>



      </div>

      <div class="unit one-fifth hide-on-mobiles">
  <aside>
    <ul>
      <li class="current">
        <a href="/news/">All News</a>
      </li>
    </ul>
    <h4>Recent Releases</h4>
    <ul>
      
      <li class="">
        <a href="/news/2021/12/12/gnverifier-0.6.2/">GNverifier v0.6.2</a>
      </li>
      
      <li class="">
        <a href="/news/2021/10/28/gnparser-1.5.0/">GNparser v1.5.0</a>
      </li>
      
      <li class="">
        <a href="/news/2021/08/13/gnparser-1.3.3/">GNparser v1.3.3</a>
      </li>
      
      <li class="">
        <a href="/news/2021/07/05/gnparser-1.3.0/">GNparser v1.3.0</a>
      </li>
      
      <li class="">
        <a href="/news/2021/06/19/gnfinder-0.14.1/">GNfinder v0.14.1</a>
      </li>
      
    </ul>
    <h4>Other News</h4>
    <ul>
        
        <li class="">
          <a href="/news/2021/12/12/gnverifier-0.6.2/">GNverifier release v0.6.2, Advanced Search</a>
        </li>
        
        <li class="">
          <a href="/news/2021/10/28/gnparser-1.5.0/">GNparser (Go language) release 1.5.0</a>
        </li>
        
        <li class="">
          <a href="/news/2021/08/13/gnparser-1.3.3/">GNparser (Go language) release 1.3.3</a>
        </li>
        
        <li class="">
          <a href="/news/2021/07/14/go/">Why Go is a great language for biodiversity informatics</a>
        </li>
        
        <li class="">
          <a href="/news/2021/07/05/gnparser-1.3.0/">GNparser (Go language) release 1.3.0</a>
        </li>
        
        <li class="">
          <a href="/news/2021/06/19/gnfinder-0.14.1/">GNfinder release v0.14.1</a>
        </li>
        
        <li class="">
          <a href="/news/2021/04/10/gnverifier-0.3.0/">GNverifier release v0.3.0</a>
        </li>
        
        <li class="">
          <a href="/news/2021/04/10/gnarser-1.2.0/">GNparser (Go language) release 1.2.0</a>
        </li>
        
        <li class="">
          <a href="/news/2021/03/23/gnparser/">GNparser (Go language) release 1.1.0</a>
        </li>
        
        <li class="">
          <a href="/news/2019/12/11/gnfinder/">gnfinder release 0.9.1 -- bug fixes</a>
        </li>
        
        <li class="">
          <a href="/news/2019/12/10/gnrd/">GNRD release (v0.9.0) switched to gnfinder from TaxonFinder/NetiNeti and became 25 times faster.</a>
        </li>
        
        <li class="">
          <a href="/news/2019/11/21/nsf1645959-outcomes/">Outcomes draft (Final Report for NSF award 1645959)</a>
        </li>
        
        <li class="">
          <a href="/news/2019/11/20/col-gbif-bhl/">Biodiversity Heritage Library (BHL) link to Catalogue of Life Plus (CoL+)</a>
        </li>
        
        <li class="">
          <a href="/news/2019/11/19/gnparser-clib/">gnparser release (v0.12.0) can be used in most modern languages</a>
        </li>
        
        <li class="">
          <a href="/news/2019/11/13/biodiversity-next/">Biodiversity-Next Conference</a>
        </li>
        
        <li class="">
          <a href="/news/2017/05/19/blackout/">Power down</a>
        </li>
        
        <li class="">
          <a href="/news/2016/11/15/names-in-november/">Names in November meeting</a>
        </li>
        
        <li class="">
          <a href="/news/2016/10/06/gnparser-0-3-3/">gnparser release v0.3.3</a>
        </li>
        
        <li class="">
          <a href="/news/2016/05/23/gnparser-0-3-0/">Scala-based gnparser v.0.3.0</a>
        </li>
        
        <li class="">
          <a href="/news/2015/11/30/ubio-is-back/">uBio and Nomenclator Zoologicus are back</a>
        </li>
        
        <li class="">
          <a href="/news/2015/11/30/gnparser-0-2-0/">Scala-based gnparser v.0.2.0</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/30/open-source-papers2/">Writing Papers as Open Source -- Solutions</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/19/open-source-papers/">Writing Papers as Open Source</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/14/bishop-museum-workshop/">Honolulu Workshop</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/13/BioGUID-Wins-Award/">BioGUID Wins Award!</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/08/BioGUID-Moved/">BioGUID merges with GN</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/06/gnub-proposal/">GNUB proposal submitted to NSF</a>
        </li>
        
        <li class="">
          <a href="/news/2015/10/05/ui-mid-november/">Global Names Gains Stable Funding</a>
        </li>
        
        <li class="">
          <a href="/news/2015/09/22/otol-eol/">Crossmapping Names by IDs</a>
        </li>
        
        <li class="">
          <a href="/news/2015/09/09/alex-myltsev/">Alexander Myltsev</a>
        </li>
        
        <li class="">
          <a href="/news/2015/09/03/sysopia/">Sysopia release v0.5.0</a>
        </li>
        
        <li class="">
          <a href="/news/2015/09/03/globalnames-site/">Site globalnames.org merged with GN blog</a>
        </li>
        
        <li class="">
          <a href="/news/2015/08/27/parser-3-4-1/">GN Parser v.3.4.1</a>
        </li>
        
        <li class="">
          <a href="/news/2015/08/20/parser-3-4-0/">GN Parser v.3.4.0</a>
        </li>
        
        <li class="">
          <a href="/news/2015/08/14/parser-3-3-0/">GlobalNames Parser v.3.3.0 adds host option for socket server</a>
        </li>
        
        <li class="">
          <a href="/news/2015/08/13/col-meeting/">Catalogue of Life Meeting in Champaign/Urbana</a>
        </li>
        
        <li class="">
          <a href="/news/2015/08/12/gn-resolver-vernaculars/">Global Names Resolver now shows vernacular names</a>
        </li>
        
        <li class="">
          <a href="/news/2015/06/01/biodiversity-3-2-1/">Global Names Parser v3.2.1</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/31/gn-uuid-0-5-0/">New UUID v5 Generation Tool -- gn_uuid v0.5.0</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/31/gn-crossmap-0-1-6/">Crossmap Tool v0.1.6</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/28/gn-crossmap-0-1-5/">Crossmap Tool v0.1.5</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/20/idigbio-client/">iDigBio API Client v0.1.1</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/14/parser-3-1-10/">Scientific Name Parser v3.1.10</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/12/gn-moves-to-illinois/">Global Names Grant Goes to Illinois</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/11/gn-crossmap-gem/">New tool to crossmap checklists</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/06/phylo-trees/">Kickoff  Meeting for Disseminating Phylogenetic Knowledge Project</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/03/higher-level-classification/">New Higher Level Classification for Catalogue of Life</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/03/sysopia-starts/">Starting GSOC 2015 Sysopia Project</a>
        </li>
        
        <li class="">
          <a href="/news/2015/05/02/species-file-group/">Species File Group -- new home for GN?</a>
        </li>
        
        <li class="">
          <a href="/news/2015/04/29/good-bad-names/">Good names, bad names?</a>
        </li>
        
        <li class="">
          <a href="/news/2015/04/28/idigbio-api-hackathon-meet/">iDigBio hackathon preparation</a>
        </li>
        
        <li class="">
          <a href="/news/2015/04/28/gsoc-netineti/">Launching NetiNeti Google Summer of Code Project</a>
        </li>
        
        <li class="">
          <a href="/news/2015/04/27/gsoc-2015-starts/">Google Summer of Code 2015 starts</a>
        </li>
        
        <li class="">
          <a href="/news/2015/04/22/thank_you_eol/">Gracious gift from Encylopedia of Life</a>
        </li>
        
        <li class="">
          <a href="/news/2015/04/17/gna_is_back/">GNA work continues</a>
        </li>
        
    </ul>
  </aside>
</div>


      <div class="clear"></div>

    </div>
  </section>


  <footer role="contentinfo">
  <div class="grid" style="vertical-align: middle">
    <div class="unit one-third center-on-mobiles">
      <p>
        <a href="https://creativecommons.org/publicdomain/zero/1.0/">
          <img src="/img/cc0.png" width="80" alt="Creative Commons 0">
        </a>
        The content of thiswebsite is in public domain
      </p>
    </div>
    <div class="unit one-third center-on-mobiles">
      <p>
      Powered by <a href="http://github.com/jekyll/jekyll">Jekyll</a>
      </p>
    </div>
    <div class="unit one-third center-on-mobiles">
      <p>
        Hosted on 
        <a href="https://github.com/GlobalNamesArchitecture">
          <img src="/img/footer-logo.png" width="100" height="30" alt="GitHub • Social coding">
        </a>
      </p>
    </div>
  </div>
</footer>

  <script>
  var anchorForId = function (id) {
    var anchor = document.createElement("a");
    anchor.className = "header-link";
    anchor.href      = "#" + id;
    anchor.innerHTML = "<span class=\"sr-only\">Permalink</span><i class=\"fa fa-link\"></i>";
    anchor.title = "Permalink";
    return anchor;
  };

  var linkifyAnchors = function (level, containingElement) {
    var headers = containingElement.getElementsByTagName("h" + level);
    for (var h = 0; h < headers.length; h++) {
      var header = headers[h];

      if (typeof header.id !== "undefined" && header.id !== "") {
        header.appendChild(anchorForId(header.id));
      }
    }
  };

  document.onreadystatechange = function () {
    if (this.readyState === "complete") {
      var contentBlock = document.getElementsByClassName("docs")[0] || document.getElementsByClassName("news")[0];
      if (!contentBlock) {
        return;
      }
      for (var level = 1; level <= 6; level++) {
        linkifyAnchors(level, contentBlock);
      }
    }
  };
</script>

  


</body>
</html>
