<!DOCTYPE HTML>
<html lang="en-US">
<head>
  <meta charset="UTF-8">
  <title>Glossary</title>
  <meta name="viewport" content="width=device-width,initial-scale=1">
  <meta name="generator" content="Jekyll v4.3.2">
  <link rel="alternate" type="application/rss+xml" title="Global Names Architecture" href="/feed.xml">
  <link rel="alternate" type="application/atom+xml" title="Recent commits to the site" href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io/commits/master.atom">
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic,900">
  <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">
  <link rel="stylesheet" href="/css/screen.css">
  <link rel="icon" type="image/x-icon" href="/favicon.ico">
  <!--[if lt IE 9]>
  <script src="/js/html5shiv.min.js"></script>
  <script src="/js/respond.min.js"></script>
  <![endif]-->
</head>


<body class="wrap">
  <header role="banner">
  <nav class="mobile-nav show-on-mobiles">
    <ul>
  <li class="">
    <a href="/">Home</a>
  </li>
  <li class="">
    <a href="/news/">News</a>
  </li>
  <li class="">
    <a href="/apps/">Apps</a>
  </li>
  <li class="">
    <a href="/workshops/">Workshops</a>
  </li>
  <li class="">
    <a href="/bibliography/">Papers</a>
  </li>
  <li class="current">
    <a href="/docs/home/">Docs</a>
  </li>
  <li>
    <a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io"><span class="hide-on-mobiles"></span>Source</a>
  </li>
</ul>

  </nav>
  <div class="grid">
    <div class="unit one-quarter center-on-mobiles">
      <div class="logo">
        <a href="/">
          <span class="sr-only">GlobalNames</span>
          <img src="/img/gna-logo.svg" width="100" alt="GNA Logo">
        </a>
      </div>
    </div>
    <nav class="main-nav unit three-quarters hide-on-mobiles">
      <ul>
  <li class="">
    <a href="/">Home</a>
  </li>
  <li class="">
    <a href="/news/">News</a>
  </li>
  <li class="">
    <a href="/apps/">Apps</a>
  </li>
  <li class="">
    <a href="/workshops/">Workshops</a>
  </li>
  <li class="">
    <a href="/bibliography/">Papers</a>
  </li>
  <li class="current">
    <a href="/docs/home/">Docs</a>
  </li>
  <li>
    <a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io"><span class="hide-on-mobiles"></span>Source</a>
  </li>
</ul>

    </nav>
  </div>
</header>


    <section class="docs">
    <div class="grid">

      

<div class="docs-nav-mobile unit whole show-on-mobiles">
  <select onchange="if (this.value) window.location.href=this.value">
    <option value="">Navigate the docs…</option>
    
    <optgroup label="Getting Started">
      





  

  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
      <option value="/docs/home/">Introduction</option>
    
  

  

  
    
  
    
  
    
  
    
      <option value="/docs/glossary/">Glossary</option>
    
  
    
  
    
  
    
  
    
  


    </optgroup>
    
    <optgroup label="Tutorials">
      





  

  
    
  
    
  
    
  
    
  
    
  
    
      <option value="/docs/tut-xsv-gnparser/">Parse names into a CSV file</option>
    
  
    
  
    
  

  

  
    
  
    
  
    
  
    
  
    
  
    
  
    
      <option value="/docs/tut-gnfinder/">Extract names from many PDF files</option>
    
  
    
  


    </optgroup>
    
    <optgroup label="GN docs">
      





  

  
    
  
    
  
    
      <option value="/docs/gnub/">Global Names Usage Bank (GNUB)</option>
    
  
    
  
    
  
    
  
    
  
    
  


    </optgroup>
    
    <optgroup label="Names">
      





  

  
    
      <option value="/docs/good-bad-names/">Good and Bad Names</option>
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
  

  

  
    
  
    
      <option value="/docs/curious-names/">Odd Names</option>
    
  
    
  
    
  
    
  
    
  
    
  
    
  


    </optgroup>
    
  </select>
</div>


      <div class="unit four-fifths">
        <article>
          <h2>
          Glossary
          <a href="/docs/glossary/" class="permalink" title="Permalink">∞</a>
          </h2>
    <div>
   
   <span class="doc-date">Edited on: 24 Sep 2015</span>
   
   
   <a href="https://github.com/djpmapleferryman" class="post-author">
      <img src="https://github.com/djpmapleferryman.png" class="avatar" alt="djpmapleferryman avatar" width="24" height="24">
      djpmapleferryman
    </a>
   
    </div>
          <blockquote>
  <p><a href="https://github.com/GlobalNamesArchitecture/GlobalNamesArchitecture.github.io/issues">Requests and feedback</a> are appreciated. Terms in <strong>bold</strong> indicate that a term is being used as a name-string</p>
</blockquote>

<h4 id="ambiregnal">Ambiregnal</h4>
<p>Organisms which may be subject to more than one code of nomenclature. Examples include (but are not limited to) the cyanobacteria which, while bacteria and therefore logically subject to the International Code of Nomenclature of
Bacteria, are studied by the same community as studies eukaryotic algae and who apply the code of botanical nomenclature(Inyternational Code of Nomenclatore of Algae, Fungi and Plants  - ICNAFP); to organisms that have been treatedas protozoa (and to which International Code of Zoological Nomenclature - ICZN has been applied) and as algae (to which the ICBN has been applied – such as dinoflagellates, euglenids, chrysophytes; ororganisms whose perceived affinities have greatly changed as a result of newknowledge (such as microsporidia, once regarded as type of protozoon and therefore subject to ICZN, but which are a much modified type of fungus and to which ISBN applies. Such cross-code taxa present an array of special problems, in some cases having more than one legitimate name.</p>

<h4 id="annotation">Annotation</h4>
<p><strong>1.</strong> Refers to the inclusion within a name string of additional information that clarifies the meaning of the name string (e.g. <strong>Indigofera argentea Burm.f., 1768 non L., 1771</strong>, or <strong>Aegilops triaristata Willd., nom. superfl.</strong>.
<strong>2.</strong> A mechanism to add comments to digital data elements, allowing users and other stakeholders to identify errors, caps,or to make suggestions.  The annotations may be attached to UUIDs that represent the data element.</p>

<h4 id="atlas-of-living-australia-ala">Atlas of Living Australia (ALA)</h4>
<p>A large project in Australia that provides the infrastructure to improve exchange, collaboration and other forms of interoperability among Australian government departments and organizations with interests in biodiversity. Data management framework nowunderpins other national biodiversity sites, such as for France.</p>

<h4 id="appearance">Appearance</h4>
<p>The occurrence of a name in a source.  An alternative term, ‘Usage’ refers to one or more appearances in a source, and is preferred.</p>

<h4 id="authorauthority">Author/Authority</h4>
<p>The name of the scientist or group of scientists who introduce a new code-compliant name.  The author is usually included with the date of publication (e.g. <em>Pangasius larnaudii</em> Bocourt, 1866); and these usually correspond with the author and date of the including publication, but this is not universally so. Some name strings such as <strong>Justicia carthagenensis Willd. ex Nees</strong> include the term ‘ex’ to indicate the intellectual author of the name. If a name is moved to a new genus, the author of the basionym may be included in parentheses after the appropriate canonical element of scientific name (<em>Bulbinella cauda-felis</em> (L.f.) T.Durand &amp; Schinz, where L.f. refers to Linnaeus the younger).  <em>Halopeplis pygmaea</em> (Pall.) Bunge ex Ung.-Sternb.  tells us that Pallas was the author of the basionym, and that his species name was moved to <em>Haplopeplis</em> by Bunge in a publication by Ungern-Sternberg.</p>

<h4 id="available-name">Available name</h4>
<p>A scientific name that complies with the provisions of the of the zoological code. Botanists use ‘validly published” for this.  ‘Code Compliant’ is alternative term.</p>

<h4 id="basionym">Basionym</h4>
<p>Refers to binomen which is the first instance of the use of a species epithet. The species epithet may be moved to a different genus to create a ‘new combination’.   <em>Gerardia purpurea</em> Linnaeus is the basionym of <em>Agalinis purpurea (L.)
Pennell</em>. ‘Protonym’ is used in a similar way, but more precisely refers to the first ‘appearance’ of a name.</p>

<h4 id="binomial-nomenclature">Binomial nomenclature</h4>
<p>Refers to the naming of species in which the species name has two parts, a genus name and a species epithet (e.g. <em>Cryptomonas paramecium</em>).</p>

<h4 id="biocode">Biocode</h4>
<p>A code that was intended to apply to all types of organisms, and so overcomingsome of the problems that arise for the existence of independent codes for animals, plants (and algae and fungi), bacteria, etc. It was published as: <em>Greuter W; Hawksworth DL; McNeill J; Mayo MA; Minelli A; Sneath PHA; Tindall BJ; Trehane P; Tubbs P (eds) (1998) Draft
BioCode (1997): the prospective international rules for the scientific names of organisms. Taxon 47: 127-150.</em></p>

<h4 id="biodiversity-information-standards-tdwg">Biodiversity Information Standards (TDWG)</h4>
<p>An <a href="http://tdwg.org">international organization</a> formerly known as <em>Taxonomic Databases Working Group</em>. It stands behind of many adopted formats and standards for biodiversity information transfer. TDWG was formed to establish international
collaboration to promote the wider and more effective dissemination of information about the World’s heritage of biological organisms for the benefit of the world at large.</p>

<h4 id="candidatuscandidate">Candidatus/Candidate</h4>
<p>A term in nomenclature, used mostly by bacteriologists and virologists, to refer to taxa that have not yet met all of the requirements of the prevailing code of nomenclature. Typically the term is written out: <strong>Candidatus Pelagibacter ubique</strong>.</p>

<h4 id="canonical-name">Canonical name</h4>
<p>The canonical name is comprised of the latinized elements of a scientific name. The canonical version of <em>Cucurbita pepo</em> is <strong>Cucurbita pepo</strong>; the canonical version of <em>Zophosis quadrilineata</em> (Olivier 1795) is <strong>Zophosis quadrilineata</strong>; the canonical version of <em>Rhododendron weyrichii</em> Maxim. f. <em>albiflorum</em> T.Yamaz is <strong>Rhododendron weyrichii albiflorum</strong>; the canonical form of <em>Choriozopella trägårdhi</em> Lawrence, 1947  is <strong>Choriozopella tragardhi</strong>; of  <em>Amphiprora pseudoduplex</em> (Osada &amp; Kobayasi, 1990) comb. nov. is <strong>Amphiprora pseudoduplex</strong>, and of the hybrid <em>Coeloglossum viride</em> (L.) Hartman x <em>Dactylorhiza majalis</em> (Rchb. f.) P.F. Hunt &amp; Summerhayes ssp. <em>praetermissa</em> (Druce) D.M. Moore &amp; Soó is <strong>Coeloglossum viride  × Dactylorhiza majalis praetermissa</strong>.</p>

<h4 id="catalogue-of-life">Catalogue of Life</h4>
<p>A <a href="http://catalogueoflife.org">project</a> which aims to collate all species into a curated single checklist in the context of a managerial taxanomical hierarchy.</p>

<h4 id="chresonym">Chresonym</h4>
<p>A reference to the use of a name. The sperm whale <em>Physeter catodon</em> was first formally described by Linnaeus in the 1758 edition of Systema Naturae, so the correct name with authority information is <em>Physeter catodon</em> Linnaeus, 1758. <strong>Physeter catodon<em>: Harmer, 1928** on the other hand refers to the use of the name by Harmer. Chresonyms may be  similar in
appearance to the scientific name plus authority (the example above may be given as **Physeter catodon</em> Harmer, 1928</strong>). Chresonyms may be distinguished by the presence of an additional element of punctuation.  A chresonym may simply regfer to a usage, or may be used to indzicate the taxonomic concept that occurred in the source. In some taxonomic areas, it is routine to include chresonyms among the synonymy lists. This is wrong as the chresonym is NOT based on a nomenclatural act.</p>

<h4 id="clade-identifiable">Clade identifiable.</h4>
<p>A term applied to a name-string which contains sufficient information as to allow the clade to be identified.  Most usually this is because the name includes the scientific name (<strong>Girardinia diversifolia</strong> or <strong>Balaenidae</strong>), but it also includes negated names (<strong>Nitella aff. cristata KGK0128</strong>  and <strong>Heteronema c.f. exaratum</strong>) because the name-string relates to the genus <em>Nitella</em> and <em>Heteronema</em>.  In the latter case, there is sufficient information to distinguish which homonym of <strong>Heteronema</strong> is being referenced). The use of a common or vernacular name-string may be sufficiently explicit (<strong>nemertean</strong> refers to Nemertea, <strong>endophytic ascomycete BCC 8616</strong> identifies the target as an ascomycete fungus. In some cases, a name-string  that contains a scientific name that is clade identifiable (<strong>Osedax sp. yellow-patch</strong> where the term ‘yellow patch’ is a ‘surrogate’ for a species epithet. Other name strings may contain a scientific name, but the clade referred to by the name-string is not identifiable (<strong>endosymbiont ‘Oregon 40’ of Acharax johnsoni</strong>).</p>

<h4 id="classification">Classification</h4>
<p>The arrangement of taxa into a hierarchy that uses scientific names and usually but not necessarily includes ranks. The hierarchical arrtangement may also be referred to as a ‘parent-child’ structure.</p>

<h4 id="code">Code</h4>
<p>Refers to a nomenclatural code that sets rules and guidelines for the formation of scientific names: for Cultivated Plants (Brickell et al., 2009);  phylogenetic clades (Cantino &amp; de Querioz, 2010), animals (International Commission on Zoological Nomenclature, 1999), viruses (International Committee on Taxonomy of Viruses 2011), prokaryotes  (Lapage, et al 1992), and algae, fungi and plants (McNeill et al., 2012). There are some less familiar codes such as the ‘Biocode’and the ‘Code of Phytosocial Nomenclature’.</p>

<h4 id="code-compliant-name">Code compliant name</h4>
<p>A scientific name that complies with the provisions of the appropriate  code.  Other terms with similar meanings include ‘available name’and  ‘validly published” name.</p>

<h4 id="code-of-phytosociologial-nomenclature">Code of Phytosociologial Nomenclature</h4>
<p>One of the less well known codes of nomenclature which issues the rules of <strong>syntaxa</strong> – vegetaton types composed of many species. This code is out of scope of Global Names</p>

<h4 id="common-name">Common name</h4>
<p>Also referred to as colloquial or vernacular name for a taxon. See Vernacular name.</p>

<h4 id="committee-on-phylogenetic-nomenclature-cpn">Committee on Phylogenetic Nomenclature (CPN)</h4>
<p>A committee of the International Society for Phylogenetic Nomenclature from which the Phylocode is derived.</p>

<h4 id="colloquial-name">Colloquial name</h4>
<p>Also referred to as common or vernacular name for a taxon. See Vernacular name.</p>

<h4 id="crossmapping">Crossmapping</h4>
<p>The exercise of comparing compilations of names.  The Global Names Cross Mapper (GNXM) is a prototype tool developed by GN in collaboration with CoL.  The process may or may not involve processing of names, such as ‘parsing’ or reducing them to ‘canonical names’.</p>

<h4 id="date">Date</h4>
<p>The date of publication of a nomenclatural act.  This is usually the same date of the publication that includes the nomenclatural act, but not always.</p>

<h4 id="diagnosis">Diagnosis</h4>
<p>An element of some nomenclatural codes, being a succinct statement of the attributes of a taxon when first published, some codes require the diagnosis to be in Latin. Minimally, may refer to the parent taxon and the apomorphic feature(s) that distinguishes the taxon in question.</p>

<h4 id="disambiguation">Disambiguation</h4>
<p>The act of providing clarity of meaning where a term may mean more than one thing.  In the case of names management, the primary use of disambiguation is to discriminate among the meanings of ‘homonyns’, names of different organisms but that are spelled identically.</p>

<h4 id="doi">DOI</h4>
<p>DOIs are persistent identifiers used, for example, by publishers for articles. An explanation of the benefits was  given on the <a href="http://mailman.nhm.ku.edu/pipermail/taxacom/2010-February/068205.html">Taxacom ListServe by Rod Page and Donat Agosti</a>.</p>

<h4 id="darwin-core-archive-dwca">Darwin Core Archive (DwCA)</h4>
<p><a href="https://github.com/tdwg/dwc/blob/dwc_cleanup/guides/text/index.md">Darwin Core Archive</a> is a draft standard that allows the transfer of biodiversity information from one machine to another. It takes the form of a set of text files with a simple descriptor to provide information as to how the file is organized. Data files are arranged in a star-like manner, with one core data file surrounded by any number of “extensions” (http://rs.tdwg.org/dwc/terms/guides/text/).. Each extension record (or “extension file row”) points to a record in the core file; in this way a DWCA package might contain a list of species as one file, to which is attached
another file with occurrence information, or further files dealing with other types of information. This is GBIF’s preferred data exchange standard and is integrated with their IPT environment.</p>

<h4 id="dryad">Dryad</h4>
<p>According to their web-site,  “DataDryad.org is a curated general-purpose repositorythat makes the data underlying scientific publicationsdiscoverable, freely reusable, and citable. Dryad has integrated data submission for a growing list of journals; submission of data from other publications is also welcome”.</p>

<h4 id="duplicated">Duplicated</h4>
<p>Records within a field or database that are identical to one or more other records.  As opposed to Unique. Duplicated records mayy be verbatim duplicates (all characters are the same), or may include the same information in slightly different ways (such as when the same data are associated with lexical variants of the name).</p>

<h4 id="dwca">DWCA</h4>
<p>See ‘Darwin Core Archive’.</p>

<h4 id="the-european-distributed-institute-of-taxonomy-edit">The European Distributed Institute of Taxonomy (EDIT)</h4>
<p><a href="http://www.e-taxonomy.eu/">EDIT</a> is a collective of 29 leading European, North American and Russian institutions that responded to a call of the European Commission, issued in 2004, for a network in “Taxonomy for Biodiversity and Ecosystem Research”. The EDIT
consortium agreement has started on the 1st of March 2006 and lasted until 2011.</p>

<h4 id="entity">Entity</h4>
<p>A group of organisms, such as a taxon, but entity can also be used for groups that may lack a formal name or rank, such as trees, microbes, creepy-crawlies.</p>

<h4 id="encyclopedia-of-life-eol">Encyclopedia Of Life (EOL)</h4>
<p>A <a href="http://eol.org">project</a> to create a web page for every known species on Earth.</p>

<h4 id="epithet">Epithet</h4>
<p>The second part of the latin ‘binomial’ for species - the “<strong>robur</strong>” of <em>Banskia robur</em>, it is mostly used in the context of botanical names.</p>

<h4 id="fossils">Fossils</h4>
<p>Organisms, parts of organisms, or the works of organisms that are no longer living, are subject to gravity and are associated with sediments.  The best estimate of the total number of described fossil species is about 250,000</p>

<h4 id="fuzzy-matching">Fuzzy matching</h4>
<p>A technique of comparing name strings that allows for controlled degrees (one character, two characters) or mis-match. The intention is to identify different name strings as lexical variants - caused through typographical, transliteration, or OCR errors - for the same name. Global Names FuzzyMatching (GNFM) is based on an algorithm developed  Tony Rees (CSIRO, Australia).   Fuzzy matching identifies <strong>Dorsophila melanogaster</strong> as being a variant of <strong>Drosophila melanogaster</strong>.  The tolerance can be adjusted, but at tolerances of two characters and above, there is a high probability that (canonical) name-strings representing different names may be matched together.  This is also possible with a tolerance of one character (<strong>Mumia</strong> matches <strong>Cumia</strong>,
<strong>Mimia</strong>, <strong>Mucia</strong>, <strong>Mukia</strong>, <strong>Numia</strong>, and <strong>Rumia</strong>.</p>

<h4 id="genbank">GenBank</h4>
<p>The US node of the International Nucleotide Database Sequence collection, the purpose of which is to collect and disseminate sequence information, other molecular data, and associated metadata.</p>

<h4 id="gn">GN</h4>
<p>A distributed group of institutions and individuals based within the USA who promote the assembly of the GNA. At the Time of writing, includes the Bishop Museum (Hawaii), the MArine Biological Laboratory (massachusetts), and the University of Illinois at Urbana-Champaign.</p>

<h4 id="global-biodiversity-information-facility-gbif">Global Biodiversity Information Facility (GBIF)</h4>
<p>An <a href="http://gbif.org">international organization</a> supported via governments worldwide and co-ordinated through the United Nations to be a leading clearing house of biodiversity information.</p>

<h4 id="global-names-architecture-gna">Global Names Architecture (GNA)</h4>
<p>Conceived as an open-ended names-based cyberinfrastructure, GNA  will provides data and services to interconnect and add to expert sources and services to meet the needs of taxonomists, nomenclaturalists, and biodiversity informaticians. This is envisioned as a semantic environment comprised of modules interconnected by standards-compliant web services. This site provides information about GNA and its development.</p>

<h4 id="global-names-classification-and-list-respository-gnaclr">Global Names Classification and List Respository GNACLR</h4>
<p>GNACLR is a repository of classifications and lists of names, and metadata relating to source,  The name-strings within the sources are compiled in GlobalNames Index. GNACLR is accessible at http://gnaclr.globalnames.org/classifications</p>

<h4 id="global-names-index-gni">Global Names Index (GNI)</h4>
<p>An index of name strings in the broadest sense (including code-compliant scientific names, vernacular names, surrogates, identifiers and erroneous versions of names), with links to sources that have the names and to the associated data or
metadata associated with names. GNI is a core element of the GNA. This is referred to as a ‘dirty bucket’ because it is a raw list; but every item in it has a scientific cotext or indexes scientific content. One role of GNA is to manage the name strings to provide biologically meaningful access to the content.  In order to be useful, alternative names for the same entity need to be linked together in a reconciliation group, disambiguated so that identically spelled names for different entities can be distinguished, classified for navigational and data aggregation purposes and to serve the needs of taxonomists, and flagged to indicate the status of the name.</p>

<h4 id="global-names-interface-for-taxonomic-editing-gnite">Global Names Interface for Taxonomic Editing (GNITE)</h4>
<p>An interface that will allow individual taxonomists to improve the underlying names-based infrastructure. Through this interface (under development @ gnite.org) taxonomists will be able to review the names that are known to GNA and add
missing ones;  edit, correct and annotate names:;  merge, split and edit reconciliation groups; edit and improve disambiguation of homonyms and chresonyms; and they will be able to merge, edit classifications,or create classifications.</p>

<h4 id="global-names-recognition-and-discovery-gnrd">Global Names Recognition and Discovery (GNRD)</h4>
<p>GNRD is a toolkit that is used to find names in sources.  It combines names recognition and names discovery software.  The Names Recognition software is TaxonFinder b  The Name Discovery component is based on the distinctive characteristics of scientific names (such as the latinized nature of the words, inclusion of authors and dates between the mid 1700’s and today).  The algorithm currently used is NetiNeti developed by Lakshmi Akella. The GNRD package is under continuous improvement, , does paresr and fuzzy matcher get used)</p>

<h4 id="global-names-usage-bank-gnub">Global Names Usage Bank (GNUB)</h4>
<p>An data management environment to record the occurrence of names in documents, databases, notes or other records. A  central component of GNA. The usages will include all nomenclatural acts, making GNUB critical to nomenclators and
registration environments such as Index Fungorum, IPNI, and Zoobank. GNUB will index usages that help to clarify the meaning of each name and so contains resources that will be
developed into taxonomic tools and services.</p>

<h4 id="good-names">Good names</h4>
<p>A name may be deemed to be good if it is a well-formed scientific name, complies with the relevant nomenclatural code, and is endorsed by one or more informed taxonomists, and ideally by a consensus of informed taxonomists.</p>

<h4 id="gsd">GSD</h4>
<p>A Global Species Database is a provider of taxonomic content relating to one or more specified clades to the Catalogue of Life.  According to the Catalogue of Web Site, a GSD is  a  taxonomic database, which aspires to cover one taxon worldwide, contain a taxonomic checklist of all species within that taxon, deal with species as taxa, and contain synonymy and taxonomic opinion, have an explicit mechanism for seeking at least one responsible/consensus taxonomy, and for applying it consistently and cross-index significant alternative taxonomies in their synonymy.</p>

<h4 id="heterotypic-synonym-also-taxonomic-synonym--subjective-synonym">Heterotypic synonym also Taxonomic synonym,  Subjective synonym</h4>
<p>Used for two code-compliant scientific names that are thought to refer to the same species. To be heterotypic (as opposed to homotypic) the names will have been created independently and used different type material. Homotypic synonyms are created when one worker is not aware of the work of others, when stages of a life cycle or genders are described as different species, or when a “lumper” feels that a “splitter” has divided a species too finely.</p>

<h4 id="holotype">Holotype</h4>
<p>A single specimen (or part or work of a specimen) that is used to anchor ataxonomic concept and its associated name. A holotype will usually be a dead organism that is preserved in a museum or herbarium. The holotype may be the only type material that is available or may be part of an array of type material. The role of the type is to unambiguously establish the intended
meaning of a scientific name.</p>

<h4 id="homonym">Homonym</h4>
<p>A scientific name  that is spelled in the same way but refers to more than one group of organisms different organisms. For example <a href="https://en.wikipedia.org/wiki/Aotus"><em>Aotus</em></a> describes a group of monkeys and a
group of peas. Homonyms can occur within codes or between codes (when some people refer to them as hemihomonyms. The meanings of homonyms needs to bedisambiguated if we are to understand what the name refers to. Most homonyms occur with the names of genera, those involving species are few and far between, but one is Acanthocystis and <em>Pieris japonica</em>.  The author information or context can often be sufficient to disambiguate homonyms. However the occurrence of chresonyms within databases creates many entries with the same genus (or species) names, and hence creates a confusing situation that needs to be addressed.</p>

<h4 id="homotypic-or-objective-synonyms">Homotypic or Objective synonyms</h4>
<p>Refers to different names for the same species, and have the same basionym. An example is: <em>Leucocryptos remigera</em> Vørs, 1992 and <em>Kathablepharis remigera</em> (Vørs, 1992) Clay &amp; Krugens, 1999, or with an alternative spelling, <em>Katablepharis remigera</em>. Homotypic synonyms are also referred to as objective or nomenclatural synonyms.</p>

<h4 id="identifier">Identifier</h4>
<p>See identifier (molecular) and identifier (data)</p>

<h4 id="identifier-data">Identifier (data)</h4>
<p>A data identifier is an alphanumeric string that is used to refer to a data record.  Identifiers may be local in the sense that they distinguish between data records in a narrow context 9such as within a given project or research group. In a federated big data world they should be universally unique, and it should be straightforward to use them to gain access to the data associated with the identifier.  Examples are LSIDs and UUIDs, with UUIDs preferred by Global Names.</p>

<h4 id="identifier-molecular">Identifier (molecular)</h4>
<p>Molecular identifiers are also referred to as BarCodes, and are typically the sequence of a small part of the genome that can be used to distinguish among taxa.  The part of the genome that is used differs for different clades, and is selected if it  does not show (much) variation within species, but exhibits differences among species.  Cox-1 used for many animals, 16S subunit ribosomal RNA for various protists, and so on.</p>

<h4 id="index-bergeyana">Index Bergeyana</h4>
<p>A catalogue of bacterial names published in 1966; mostly rendered obsolete with subsequent revisions of the relevant code of nomenclature.</p>

<h4 id="index-fungorum-if">Index Fungorum (IF)</h4>
<p>The global fungal nomenclator coordinated and supported by the Index Fungorum Partnership, which aims to include all names proposed for fungi (including yeasts, lichens, chromistan fungi, protozoan fungi and fossil forms) since 1753 at species level and below</p>

<h4 id="international-code-of-nomenclature-for-algae-fungi-and-plants-icnafp-or-icn">International Code of Nomenclature for algae, fungi, and plants (ICNAFP or ICN)</h4>
<p>Previously known as International Code of Botanical Nomenclature (ICBN), a nomenclatural code that governs the formulation of the names of plants, algae and fungi</p>

<h4 id="international-code-of-nomenclature-of-prokaryotes-icnp">International Code of Nomenclature of Prokaryotes (ICNP)</h4>
<p>Previously known as International Code of Nomenclature of Bacteria (ICNB), a nomenclatural code to govern formulation of names of procariots</p>

<h4 id="international-committee-on-taxonomy-of-viruses-ictv">International Committee on Taxonomy of Viruses (ICTV)</h4>
<p>It produces the International Code of Virus Classification and Nomenclature of ICTV</p>

<h4 id="international-code-of-virus-classification-and-nomenclature-of-ictv">International Code of Virus Classification and Nomenclature of ICTV</h4>
<p>The code governing formulation of names of viruses. The nomenclatural practices are unlike the other major codes, and the official language is English instead of Latin.</p>

<h4 id="international-code-of-zoological-nomenclature-iczn">International Code of Zoological Nomenclature (ICZN)</h4>
<p>The code governing formulation of zoological names.</p>

<h4 id="international-commission-on-the-nomenclature-of-cultivated-plants-icncp">International Commission on the Nomenclature of Cultivated Plants (ICNCP)</h4>
<p>International Code of (and Commission for the) Nomenclature of Cultivagted Plants</p>

<h4 id="international-commission-on-zoological-nomenclature">International Commission on Zoological Nomenclature</h4>
<p>the editorial and regulatory body for zoological nomenclature, responsible for preparing the International Code of Zoological Nomenclature.</p>

<h4 id="international-committee-for-the-taxonomy-of-fungi">International Committee for the Taxonomy of Fungi</h4>
<p>A committee appointed to consider issues related to the taxonomy fungi, not concerned with formal proposals for the conservation and rejection of names.</p>

<h4 id="international-committee-on-bionomenclature">International Committee on Bionomenclature</h4>
<p>A committee originally established in 1994, and a scientific member of IUBS from 2009, comprising representatives of the different Codes and providing a forum for discussion and action on matters of common concern.</p>

<h4 id="international-committee-on-systematics-of-prokaryotes">International Committee on Systematics of Prokaryotes</h4>
<p>A body appointed by the IUMS Division of Bacteriology at its international congresses. The ICSP, formerly the International Committee on Systematic Bacteriology (ICSB), is the body that oversees the nomenclature of prokaryotes, determines the rules by which prokaryotes are named and whose Judicial Commission issues Opinions concerning taxonomic matters, revisions to the
Bacteriological Code, etc.</p>

<h4 id="international-plant-names-index-ipni">International Plant Names Index (IPNI)</h4>
<p>A database of the names and associated bibliographic information of all seed plants, ferns, and fern allies (http://www.ipni.org) produced through collaboration of the Australian National Herbarium, Harvard University Herbaria, and the Royal Botanic Gardens Kew</p>

<h4 id="international-trust-for-zoological-nomenclature">International Trust for Zoological Nomenclature</h4>
<p>A not for profit company founded in 1947 which acts on behalf of the International Commission on Zoological Nomenclature to raise and administer funds.</p>

<h4 id="interim-register-of-marine-and-non-marine-genera-irmng">Interim register of Marine and Non Marine genera (IRMNG)</h4>
<p>While actively maintained, was the most complete and organized list of genera of organisms - with (at the time of writing) 446,000 of the estimated 470,000 generic names; with annotations as to whether the taxa the names refer to are marine or non-marine, and whether they are still living or extinct.  Current status is uncertain.</p>

<h4 id="integrated-taxonomic-information-system-itis">Integrated Taxonomic Information System (ITIS)</h4>
<p>An United States Government <a href="http://itis.gov">organization</a> with a mission to create a scientifically credible database of taxonomic information, placing primary focus on taxa of interest to North America, with world treatments included, as available.</p>

<h4 id="italics">Italics</h4>
<p>By convention, the scientific names of organisms, especially at the rank of Genus and lower, are presented in italics - <em>Zophosis quadrilineata</em> (Olivier 1795). In GN, we use bold formatting to indicate that the example is to be considered as a name-string (e.g.<strong>Pangasius larnaudii Bocourt, 1866</strong>).</p>

<h4 id="junior">Junior</h4>
<p>Used in relation to homonyms and synonyms, and identifies those names that, according to the appropriate code of nomenclature, should not prevail. The senior homonym / synonym is almost always the one that was formed first (i.e. the principle of priority is served). The term ‘junior’ is most widely used in the context of zoological nomenclature.</p>

<h4 id="junior-synonym">Junior Synonym</h4>
<p>Synonyms are two or more code-compliant names that are applied to the same taxon. Only one will be correct.  The junior synonyms are those that should not prevail. The senior synonym is almost always the one that was formed first (i.e. the principle of priority is served). The term ‘junior’ is most widely used in the content of zoological nomenclature.</p>

<h4 id="legitimate-name">Legitimate name</h4>
<p>A botanical term that refers to a scientific name that was formulated in agreement with the rules of the ICN. May also be referred to as ‘code-compliant’ or ‘valid’.</p>

<h4 id="lexical-variant">Lexical variant</h4>
<p>Alternative spellings and formulations of a name string. For example <strong>Cyclotrachelus sodalis (LeConte)</strong> and <strong>Cyclotrachelus (Evarthrus) sodalis (Le Conte 1848)</strong> are Lexical variants of the same name-string. In a broader sense lexical variants can include misspellings - such as <strong>Cyclotrachelis sodalis (Le Conte)</strong>.</p>

<h4 id="name">Name</h4>
<p>In the context of GNA, a name is a Awriteable term used as a label, token, flag or other device to refer to a taxon. There are many classes of names, such as scientific names, common names, surrogates, and identifiers. A name is presented in the form of a ‘name-string’. See canonical name, common name, scientific name, and surrogates. The scientific components of names are usually in Latin and at the rank of genus and below, are written in italics.</p>

<h4 id="name-object">Name object</h4>
<p>A component of a canonical name-string. <strong>Homo sapiens</strong> has two name objects: “<strong>Homo</strong>” and “<strong>sapiens</strong>”, a trinomial has 3. Annotations, punctuation, and authority information may be elements in a name string but are not ‘name objects’</p>

<h4 id="name-scientific">Name: scientific</h4>
<p>A latinized name that is used to refer to a taxon. Some scientific names are regulated by the codes of nomenclature and may be referred to as code-compliant if they comply with the relevant Code. Others (such as names of taxa above the rank of family) are not subject to the codes. Some scientific names are comprised of a single latinized word, but in the case
of species, the scientific name is  made up of two words (<em>Rattus norvegicus</em>).  Trinomials are formed when a subgenus or a subspecies is added to the species name. Codes of nomenclature differ in how they define  ‘name”and the
‘scientific name’ of species. The english glossary in the International Code of Zoological Nomenclature refers to the species binomen as being comprised of two names; whereas the botanical convention is that the binomen is a name, and the
species component of it is referred to as an epithet.</p>

<h4 id="name-parsing">Name-parsing.</h4>
<p>See Parser.</p>

<h4 id="name-string">Name-string</h4>
<p>The combination of characters, numbers, punctuation, spaces, symbols, and abbreviations that is used to represent a name.  We currently identify the following classes of name-strings: (1) scientific names; (2) common or vernacular names; (3) surrogates and (4) identifiers. The same name may be represented by many lexical variants of the same name-string. Examples are <strong>Entoloma fragilium</strong> (basic canonical form of a name), <strong>Cortinarius brunneovernus Niskanen, Liimat. &amp; Ammirati, 2013</strong> (canonical form of name with author information),  <strong>H. felis Patton, 1908</strong> (an abbreviated name, the identity of the taxon is not clear), <strong>Bynoe’s prickly gecko</strong> (a common or vernacular name) and <strong>2,4-D degrading transconjugant WD3</strong> (a surrogate for a taxonomic name). We use the convention of <strong>bolding</strong> to distinguish name-strings in this glossary and elsewhere.</p>

<h4 id="name-surrogate">Name: surrogate</h4>
<p>A string of alphanumeric characters that refers to a taxon, but is neither a scientific name nor a vernacular name. Often a form of coding that is used to refer to a sample, a specimen, or a concept before it becomes well established. SAR11 was the name given to a bacterial clade and much was published on it,
until later it was given a scientific name <em>Pelagibacter ubique</em> but as this was  not code-compliant it is currently referred to as <em>Candidatus Pelagibacter ubique</em>.</p>

<h4 id="negated-names">Negated names</h4>
<p>Name-strings may contain annotations that negate some or all of the canonical elements in the name-string. These include aff., cf, nr. non, not, ab., ?, sensu auctt. etc.).  In essence, they alert us to the fact that the name string in use does not refer to the taxon that is usually refers to. Each annotation had a different taxonomic meaning not, similar to, in the sense of authors to indicate that the taxon is not as originally proposed. Examples include: <strong>Limax cf. graecus sensu Wiktor, 2001</strong>, and <strong>Labiotermes nr labralis</strong>, they do not refer to <em>Limax graecus</em> (but to a species that is similar to Limax graecus in the sense of Wiktor; nor to <em>Labiotermes labralis</em> but that the species in question is similar (near) to <em>Labiotermes labralis</em></p>

<h4 id="nom-nov">Nom. nov.</h4>
<p>A new name. A term used when for some reason an existing name can no longer be used – for example it is discovered to be a junior homonym of a name that was introduced earlier for a different species.</p>

<h4 id="nomenclator">Nomenclator</h4>
<p>A compiler of names and opinions that provide information on the code-compliancy of the names. A definitive source of “correct” names. The “Index Fungorum” is the nomenclator for Fungi.  Registries are nomenclators in a digital world.</p>

<h4 id="nomenclatural-act">Nomenclatural act</h4>
<p>A statement in a publication that affects the nomenclatural status of ascientific name or its typification. Such an act may be the introduction of a new name, a correction of prior nomenclatural act, or the creation of a new combination of genus and species names (in which a homotypic, objective, or nomenclatural synonyms may be created). Nomenclators and registries compile  nomenclatural acts.</p>

<h4 id="nomenclature">Nomenclature</h4>
<p>That component of taxonomy that relates to the legitimacy, formation, and stability of scientific names. The regulations that apply are included in the various codes of nomenclature (see Codes of Nomenclature), and define what constitutes
publication, the use of type material, the construction of names, and who trumps who when there is more than one code-compliant name for a taxon. The codes typically deal with taxa up to the rank of Family.</p>

<h4 id="nomina">Nomina</h4>
<p>An open-ended and occasional series of workshops initiated by GNA and used used to develop ideas and components of the Global Names Architecture. (Nomina can also be used to  mean “names”).</p>

<h4 id="nominal-taxa">Nominal taxa</h4>
<p>A scientific name that has been introduced for a taxon without evaluation or confirmation that the name is code-compliant, or that it appropriately applies to a distinct taxon.  When a taxonomist reviews a taxonomic area, they will compile a list of all nominal taxa, then determine which names are code-compliant, and then make decisions about the clade to which the name applies along with making synonymy lists.</p>

<h4 id="parser">Parser</h4>
<p>A software tool that  breaks a name into its semantically discrete elements (such as genus, species epithet, infraspecific epithet, annotations, authors, dates of authorship, hybrid marks and so on.  The Global Names parser is accessible at http://gni.globalnames.org/parsers/new.</p>

<h4 id="the-pan-european-species-directories-infrastructure-pesi">The Pan European Species Directories Infrastructure (PESI)</h4>
<p>PESI provides standardised and authoritative taxonomic information by integrating and securing Europe’s taxonomically authoritative species name registers and nomenclators (name databases) and associated exper(tise) networks that underpin the management of biodiversity in Europe.</p>

<h4 id="parent-child">Parent Child</h4>
<p>The arrangement of taxa into a hierarchy or phylogeny by reference to membership - the group is the parent, and the subordinate taxa or clades are the children.</p>

<h4 id="phylocode">Phylocode</h4>
<p>A Code of biological nomenclature that uses phylogenetic principles to define
clades (components of a phylogenetic tree and/or the taxa included within that
component of the phylogenetic tree). The International Committee on
Phylogenetic Nomenclature can be held responsible for this</p>

<h4 id="phylogenetic-tree">Phylogenetic tree</h4>
<p>Representation of evolutionary relationships (phylogeny) as a branching diagram
or tree</p>

<h4 id="phylogeny">Phylogeny</h4>
<p>The evolutionary relationships among taxa; also misleadingly but widely used
for dendrograms that are produced from analyses of data and are visualization
tools that offer an interpretation of what the phylogeny was. The phylogeny
refers to the true ancestory descendent relationships, a phylogeny refers to
a hypothesis of phylogenetic relationships.</p>

<h4 id="polynomial">Polynomial</h4>
<p>A name of a taxon that includes three or more separate words,e.g.<strong>Cucumis melo subsp. agrestis var. conomon</strong></p>

<h4 id="postbox">Postbox</h4>
<p>PostBox is GN’s list and classification submission environment. It accepts various formats of file, converts them to DWCA form, places the name-strings in GNI and the list/classification in GNACLR. .http://postbox.globalnames.org/</p>

<h4 id="priority">Priority</h4>
<p>Refers to a principle that when scientific names are in competition, normally the oldest (earliest established) name is the one that should be used.</p>

<h4 id="pro-parte-or-pp">pro parte or p.p.</h4>
<p>“in part”. Refers to a part of a concept of an author, usually applied in situations where the author in question is deemed to have included more than one species in their concept, and it is necessary to point out that only a part of the description / scope is applicable.</p>

<h4 id="protologue">Protologue</h4>
<p>A botanical terms that refers to everything associated with a name when it was first published in compliance with the prevailing code – such as the description, diagnosis, phylogenetic definition, registration number, designation of type, illustrations, references, synonymy, geographical data,
specimen citations, and comments.</p>

<h4 id="publication">Publication</h4>
<p>In the broad sense, any readable document or record, in a narrow sense refers to places in which names are “published” (see published). Electronic publication refers to publication in non-paper forms, such as a pdf file accessible via a web-site.</p>

<h4 id="published">Published</h4>
<p>A term that in the context of a nomenclatural code defines the conditions that need to be fulfilled when a name is first presented in order for that name to be recognised as a scientific name.</p>

<h4 id="ranks">Ranks</h4>
<p>The levels within a hierarchical classification of organisms. There are seven principal categories: kingdom, phylum, class, order, family, genus, and species. However, there are many intermediary ranks (such as subclasses, superfamilies, tribes, subspecies, varieties, etc. Over 90 ranks have been used.  Ranks are arbitrary.</p>

<h4 id="reconciliation">Reconciliation</h4>
<p>The GN solution to the “many-names-for-one-taxon” problem, as an alternative to seeking the use of a single preferred name. Reconciliation involves mapping all alternative name-strings for the same taxon into ‘reconciliation groups’ so that a query initiated with a single name can be ‘exploded’ to access data that was published under any name for the taxon.</p>

<h4 id="reconciliation-group">Reconciliation group</h4>
<p>Reconciliation groups are used to address the “many-names-for-one-species”; a reconciliation group seeks to group all name-strings that have been used for a taxon. A query or other action initiates with one name can be extended to
give a response involving some or all names. Names within groups can be annotated to show which are scientific, which are vernacular or surrogate names, to indicate nomenclatural status, etc. Reconciliation includes variant spellings (lexical variants) of all names, homotypic and heterotypic synonyms of scientific names, vernacular names, and surrogate names.</p>

<h4 id="registration">Registration</h4>
<p>Used in the context of scientific names, the act of recording a scientific name with a registration body. It is likely that future taxonomy will require registration of new names through one or more (co-ordinated) registration sites – such as
Zoobank. Registration may be pro-active (i.e. one can obtain a registration number from ZooBank for a new name before the relevant paper is published; but ensures that the paper refers to the registered name.  Most registration is retrospective.</p>

<h4 id="resolution">Resolution</h4>
<p>Resolution extends the process of ‘reconciliation’, presenting the name that is endorsed by a preferred taxonomic authority (his, her, or their view of the ‘right” scientific name for a taxon).</p>

<h4 id="scientific-name">Scientific name</h4>
<p>A name for a taxon that is formed in Latin and complies with a relevant code of nomenclature or otherwise meets the expectations of the nomenclatural and taxonomic communities. A scientific name may or may not include the names of the authors who created the name, basionym, or combination - with or without the dates of those events. The name may be annotated with terms such as subsp. (see examples under canonical name). Taxonomists establish which scientific name is the correct one to apply to a taxon, but different taxonomists may disagree in respect of the correct name.  Latinized names  may include authors, dates of authorship, and various annotations. ‘Parsing’ can reveal all semantic elements, scientific name can be rendered to a ‘canonical’ form of only latinized elements.</p>

<h4 id="scratchpad">ScratchPad</h4>
<p>A Drupal-based content management system for biodiversity that allows taxonomists to create web sites for biodiversity without incurring the cost of developing the software. Supported through the European EDIT / Vibrant programs.</p>

<h4 id="sec">sec</h4>
<p>According to: as in <em>Aus bus</em> sec Smith 1955 means <em>Aus bus</em> as referred to or as interpreted by Smith in her 1955 article.  Sensu is similar, meaning ‘in the sense of’. Derived from ‘secundum’.</p>

<h4 id="senior-homonym">Senior homonym</h4>
<p>Of two homonyms within the context of a code of nomenclature, the senior is the one that should continue to be used – typically the first to be established. Other homonyms needs to be replaced with new names.</p>

<h4 id="senior-synonym">Senior synonym</h4>
<p>Of two synonyms, the one that is to be used for the organism, typically the one that was created first.</p>

<h4 id="sensu">Sensu</h4>
<p>In the sense of; usually used to point to one person’s interpretation of a meaning. See also ‘sec’.</p>

<h4 id="sensu-lato">Sensu lato</h4>
<p>In a broad sense. Often used in the context of lumping and splitting to refer to the more extensive concepts of a given taxon.</p>

<h4 id="sensu-stricto">Sensu stricto</h4>
<p>In a strict sense. Often used in the context of lumping and splitting to refer to  restrictive concepts of a given taxon.</p>

<h4 id="species-epithet">Species epithet</h4>
<p>The second part of the binomial of a species name – the ‘sapiens’ of ‘<em>Homo  sapiens</em>’. Used more commonly in botany.</p>

<h4 id="splitter">Splitter</h4>
<p>Refers to a taxonomist who favors taxa that are very narrowly defined – that is they tend to split taxa up. Splitters create infrasubspecific taxa. Splitters stand in contrast to lumpers</p>

<h4 id="standards">Standards</h4>
<p>Information standards are agreed conventions by which digital information can be organised. Standards are important to machine-to-machine dialog, allowing one machine to release information in a way that other machines can understand.
Semanticization is dependent on standards. The primary body developing standards for biology is TDWG, but many other projects have developed in-house standards.</p>

<h4 id="string">String</h4>
<p>A sequence of characters, spaces, symbols, and punctuation marks that is used in a computing environment. A string may include many words. Strings that are used to represent names are referred to as name-strings.</p>

<h4 id="synonym">Synonym</h4>
<p>One of two or more code-compliant scientific names that refer to the same species. There are objective (homotypic) and subjective (heterotypic) synonyms, and junior and senior synonyms. Only one is correct; and it is usually the first to have been created (the ‘senior synonym’).</p>

<h4 id="systematics">Systematics</h4>
<p>That component of the discipline of taxonomy that relates to the arrangement of taxa , normally within hierarchies or classifications.</p>

<h4 id="surrogate-for-name">Surrogate for name</h4>
<p>An identifier for a taxon but that is not a scientific nor a colloquial name.  It may be an abbreviation, strain number, specimen number, acronym, or other form of string that takes the pace of a name. Widely used surrogates for names should be included within reconciliation groups.</p>

<h4 id="tautonym">Tautonym</h4>
<p>A species name in which the same word is used for both the genus and species name: <em>Bison bison</em>. Permitted for animals but not for plants.</p>

<h4 id="taxamatch">TaxaMatch</h4>
<p>A fuzzy matching algorithm originally developed by Tony Rees to find correct spelling for mispelled name strings</p>

<h4 id="taxon">Taxon</h4>
<p>A Scientifically named group of organisms, with a name and a rank. Ranges from  infrasubspecific taxa, through the usual players of species, genus, family, class, and order and things in between, up to Kingdom and even beyond.</p>

<h4 id="taxon-name-usage-tnu">Taxon Name Usage (TNU)</h4>
<p>Taxon (or taxonomic) name usage (TNU) is an occurrence of a taxon name in a source.Usually they are made in publications, but the concept includes usages databases, personal correspondence, field notebooks, etc. Usages may include
nomenclatural acts that relate to the application of the appropriate code, or taxonomic acts that bear on the identity of the taxon labeled with the name.</p>

<h4 id="taxonomic-act">Taxonomic act</h4>
<p>A published determination by one or more taxonomists that bear on the identify of a taxon.  This may be the discovery of a new taxon, a judgement as to heterotypic (subjective or taxonomic) synonyms, the placement of taxa within hierarchical classifications, clarification of taxonomic concepts, and so on.  Taxonomic acts include nomenclatural acts.</p>

<h4 id="taxonomic-intelligence">Taxonomic intelligence</h4>
<p>The explicit inclusion of taxonomic thinking within the design of data bases, algorithms, and services. Taxonomic thinking includes an understanding of nomenclatural regulations, the sources of the many-names-for-one-species and the one-name-for-many-species problems, the reasons for alternative
classifications and the nature of taxonomic concepts.</p>

<h4 id="taxonomically-endorsed">Taxonomically endorsed</h4>
<p>There may be many scientific names that apply to a taxon. A taxonomist will identify which of the code compliant names that may be applied to a taxon is, in his or her view,  most appropriate .  Such a name is taxonomically endorsed because it is preferred over any other scientific name. Different taxonomists may hold different opinions, and the taxonomically endorsed name may change over time.</p>

<p>####Taxonomy
The discipline that deals with the discovery of taxa, their identification, naming, relationships, specimens, associated literature and other materials that bear on the relationships of taxa, clarification of taxonomic concepts and taxonomic concepts. Some elements of taxonomy rely on judgement, and so different taxonomist may draw different conclusions from the same information.</p>

<h4 id="transregnal">Transregnal</h4>
<p>Issues, such as nomenclatural and taxonomic issues,  that transcend single traditional taxonomic domains of animals, plants, fungi,  bacteria, or viruses.</p>

<h4 id="trinomial-name">Trinomial name</h4>
<p>A scientific name with three parts, such as happens when a subgenus is included within a name, or when a subspecies is included. An example is <em>Bison (Bison) bison</em> (Genus (subgenus) species), or <em>Bison bison bison</em> (Genus, species subspecies). A less confusing example is <em>Chlorella vulgaris var. viridis</em>.</p>

<h4 id="type-material">Type material</h4>
<p>The specimens and other elements used by an author in the original descriptionof a new species or subspecific taxon, and identified as the reference material from which the identity of the taxon is established.</p>

<h4 id="type-specimen">Type specimen</h4>
<p>A single specimen which acts as the defining reference point for the application of a name of a species or an infraspecific taxon. Type specimens are usually retained within Museums and herbaria.</p>

<h4 id="uninomial">Uninomial</h4>
<p>A scientific name comprised of a single single word, in the name-string a <strong>Zyzzyzus Stechow, 1921</strong>, the uninomial is <em>Zyzzyzus</em>).</p>

<p>####Unique
•	A record within a field or database that is distinguishable from all other records.  As opposed to duplicate.</p>

<h4 id="usage">Usage</h4>
<p>The ocurrence or occurrences of elements of a name within a document.  If Smith (1955) published an article on <em>Aus bus</em>, and mentioned the name “Aus bus” 14 times within the article (i.e. there are 14 appearances of the binomial), then there would be three usages - “<em>Aus</em> in Smith 1955”, “<em>bus</em> in Smith 1955”, and (arguably) “<em>Aus bus</em> in Smith 1955”.  The combination of the name and the reference is a chresonym.  In the event that the usages bears on the identity of the taxon, these usages may also be referred to as “<em>Aus</em> sec Smith 1955” or “<em>Aus bus</em> sensu Smith 1955”.</p>

<h4 id="uuid">UUID</h4>
<p>UUID: a  32 alphanumeric string (e.g. 65247ae4-2c16-59ca-bc6d-9ae241202572)  that can be used as a unique identifier for a digital object (in this case it is the GN UUID for <strong>Cafeteria roenbergensis</strong>).  Global Names uses an algorithmic approach to generate the UUIDS  so that other users with the same algorithm will obtain the same UUID for the same name-string.  The UUID may be presented with a pointer to a location by which the identifier can be dereferenced to reveal the information it points to. The ZooBank identifier for the same species is urn:lsid:zoobank.org:act:17665D1C-4022-4A71-8B1F-FE19B01B8F81. Catalogue of Life does not list <em>Cafeteria</em>, but for <em>Cafeteriavirus</em> the identifier is urn:lsid:catalogueoflife.org:taxon:f3b2be98-18fd-11e5-9774-bc764e0806fb:col20150623. As there can be multiple UUIDs for the same data object, such that reconciliation of identifiers has to be part of the overall name reconciliation process.</p>

<h4 id="valid-name">Valid name</h4>
<p>A term used by zoologists and virologists to refer to a code-compliant name of an organism. It does not mean that the name is currently endorsed as the best name for that taxon.</p>

<h4 id="vernacular-name">Vernacular name</h4>
<p>Also referred to as common or collquial name. A name of an organism that is part of natural and local language.  The common name may be in any language, there may be many different common names for the same thing at different locations (<em>Fagus sylvaticus</em>, the European beech tree, is referred to in one source by 15 names such as <strong>Hêtre fouteau</strong> and <strong>Euroopanpyökki</strong>). The same name may be used to refer to different things (<strong>magpie</strong> in Australia refers to <em>Cracticus tibicen</em>, and in England refers to <em>Pica pica</em>). Such names are associated with a language, and often cannot be accuratelly mapped to a scientific name. In a few cases, there are well-governed vernacular names, which correspond exactly to scientific names.</p>

<h4 id="zoobank">ZooBank</h4>
<p>Zoobank.org. Online register for names of animal, part of the activities of the International Commission for Zoological Nomenclature.</p>


          
          





  
  

  
  
    <div class="section-nav">
      <div class="left align-right">
          
            
            
            <a href="/docs/home/" class="prev">Back</a>
          
      </div>
      <div class="right align-left">
          
            
            
            <a href="/docs/tut-xsv-gnparser/" class="next">Next</a>
          
      </div>
    </div>
    <div class="clear"></div>
    
        </article>
      </div>

          
      <div class="unit one-fifth hide-on-mobiles">
  <aside>
    
    
    
    <h4>Getting Started</h4>
    





<ul>

  

  

  
    
  
    
  
    
  
    
  
    
  
    
  
    
  
    
      <li class=""><a href="/docs/home/">Introduction</a></li>
      


  

  

  
    
  
    
  
    
  
    
      <li class="current"><a href="/docs/glossary/">Glossary</a></li>
      


</ul>

    
    <h4>Tutorials</h4>
    





<ul>

  

  

  
    
  
    
  
    
  
    
  
    
  
    
      <li class=""><a href="/docs/tut-xsv-gnparser/">Parse names into a CSV file</a></li>
      


  

  

  
    
  
    
  
    
  
    
  
    
  
    
  
    
      <li class=""><a href="/docs/tut-gnfinder/">Extract names from many PDF files</a></li>
      


</ul>

    
    <h4>GN docs</h4>
    





<ul>

  

  

  
    
  
    
  
    
      <li class=""><a href="/docs/gnub/">Global Names Usage Bank (GNUB)</a></li>
      


</ul>

    
    <h4>Names</h4>
    





<ul>

  

  

  
    
      <li class=""><a href="/docs/good-bad-names/">Good and Bad Names</a></li>
      


  

  

  
    
  
    
      <li class=""><a href="/docs/curious-names/">Odd Names</a></li>
      


</ul>

    
  </aside>
</div>

          
      <div class="clear"></div>

    </div>
  </section>


  <footer role="contentinfo">
  <div class="grid" style="vertical-align: middle">
    <div class="unit one-third center-on-mobiles">
      <p>
        <a href="https://creativecommons.org/publicdomain/zero/1.0/">
          <img src="/img/cc0.png" width="80" alt="Creative Commons 0">
        </a>
        The content of this</br>website is in public domain
      </p>
    </div>
    <div class="unit one-third center-on-mobiles">
      <p>
      Powered by <a href="http://github.com/jekyll/jekyll">Jekyll</a>
      </p>
    </div>
    <div class="unit one-third center-on-mobiles">
      <p>
        Hosted on 
        <a href="https://github.com/GlobalNamesArchitecture">
          <img src="/img/footer-logo.png" width="100" height="30" alt="GitHub • Social coding">
        </a>
      </p>
    </div>
  </div>
</footer>

  <script>
  var anchorForId = function (id) {
    var anchor = document.createElement("a");
    anchor.className = "header-link";
    anchor.href      = "#" + id;
    anchor.innerHTML = "<span class=\"sr-only\">Permalink</span><i class=\"fa fa-link\"></i>";
    anchor.title = "Permalink";
    return anchor;
  };

  var linkifyAnchors = function (level, containingElement) {
    var headers = containingElement.getElementsByTagName("h" + level);
    for (var h = 0; h < headers.length; h++) {
      var header = headers[h];

      if (typeof header.id !== "undefined" && header.id !== "") {
        header.appendChild(anchorForId(header.id));
      }
    }
  };

  document.onreadystatechange = function () {
    if (this.readyState === "complete") {
      var contentBlock = document.getElementsByClassName("docs")[0] || document.getElementsByClassName("news")[0];
      if (!contentBlock) {
        return;
      }
      for (var level = 1; level <= 6; level++) {
        linkifyAnchors(level, contentBlock);
      }
    }
  };
</script>

  


</body>
</html>
